// @antv/l7-editor/lib/components/map-control-group/sam-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/sam-control/index.tsx
var sam_control_exports = {};
__export(sam_control_exports, {
  SamControl: () => SamControl
});
module.exports = __toCommonJS(sam_control_exports);
var import_larkmap = require("@antv/larkmap");
var import_sam = require("@antv/sam");
var import_turf = require("@turf/turf");
var import_antd = require("antd");
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_constants = require("../../../constants");
var import_recoil = require("../../../recoil");
var import_iconfont = require("../../iconfont");
var import_styles = __toESM(require("../styles"));
var import_style = __toESM(require("./style"));
var options = {
  shape: "line",
  size: 3,
  color: "#ff0000",
  state: {
    active: false
  },
  style: {
    opacity: 1,
    lineType: "solid"
  },
  zIndex: 100,
  animate: {
    interval: 0.6,
    trailLength: 1.5,
    duration: 4
  }
};
var SamControl = () => {
  const styles = (0, import_style.default)();
  const style = (0, import_styles.default)();
  const [samModel, setSamModal] = (0, import_react.useState)(null);
  const { scene, features, resetFeatures, revertCoord, bboxAutoFit } = (0, import_recoil.useFeature)();
  const { wasmPath } = (0, import_recoil.useGlobal)();
  const allLayerList = (0, import_larkmap.useLayerList)();
  const [samOpen, setSamOpen] = (0, import_react.useState)(false);
  const [tileLayer, setTileLayer] = (0, import_react.useState)(void 0);
  const [polygonLayer, setPolygonLayer] = (0, import_react.useState)(
    void 0
  );
  const [hoverPolyonLayer, setHoverPolyonLayer] = (0, import_react.useState)(
    void 0
  );
  const [loading, setLoading] = (0, import_react.useState)(false);
  const [marker, setMarker] = (0, import_react.useState)(void 0);
  const [bound, setBound] = (0, import_react.useState)(void 0);
  const [source, setSource] = (0, import_react.useState)({
    data: { type: "FeatureCollection", features: [] }
  });
  const { t } = (0, import_react_i18next.useTranslation)();
  const onMapClick = (0, import_react.useCallback)(
    (event) => {
      const coords = [event.lngLat.lng, event.lngLat.lat];
      if (bound) {
        if ((0, import_turf.booleanPointInPolygon)((0, import_turf.point)(coords), bound)) {
          if (samModel) {
            const px = samModel.lngLat2ImagePixel(coords);
            const newPoint = [
              {
                x: px[0],
                y: px[1],
                clickType: 1
              }
            ];
            const threshold = 1;
            samModel.predict(newPoint).then(async (res) => {
              const fc = await samModel.exportGeoPolygon(res, threshold);
              const image = samModel.exportImageClip(res);
              const newData = {
                feature: fc.features,
                imageUrl: image.src
              };
              if ((0, import_turf.booleanPointInPolygon)((0, import_turf.point)(coords), newData == null ? void 0 : newData.feature[0]) && (newData == null ? void 0 : newData.feature[0].geometry.coordinates[0].length) > 4) {
                const newFeature = revertCoord(newData.feature);
                resetFeatures([...features, ...newFeature]);
              } else {
                import_antd.message.warning(t("map_control_group.sam.tuXingJieXiCuoWu"));
              }
            });
          }
        } else {
          import_antd.message.error(t("map_control_group.sam.qingZaiQuYuNei"));
        }
      }
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [bound, samModel, features, t]
  );
  const generateEmbedding = async () => {
    var _a;
    setLoading(true);
    try {
      if (!tileLayer || !samModel)
        return;
      const tileSource = tileLayer.layer.getSource();
      const tiles = (_a = tileSource == null ? void 0 : tileSource.tileset) == null ? void 0 : _a.currentTiles;
      let minX = Infinity;
      let minY = Infinity;
      let maxX = -Infinity;
      let maxY = -Infinity;
      tiles == null ? void 0 : tiles.forEach((tile) => {
        minX = Math.min(minX, tile.x);
        minY = Math.min(minY, tile.y);
        maxX = Math.max(maxX, tile.x);
        maxY = Math.max(maxY, tile.y);
      });
      const zoom = Math.ceil(scene == null ? void 0 : scene.getZoom());
      const canvas = document.createElement("canvas");
      canvas.width = (maxX - minX + 1) * 256;
      canvas.height = (maxY - minY + 1) * 256;
      const ctx = canvas.getContext("2d");
      const mapHelper = samModel.mapHelper;
      const imageExtent = [
        ...mapHelper.tileToLngLat(minX, maxY + 1, zoom),
        ...mapHelper.tileToLngLat(maxX + 1, minY, zoom)
      ];
      tiles == null ? void 0 : tiles.forEach((tile) => {
        if (tile) {
          ctx == null ? void 0 : ctx.drawImage(
            tile.data,
            (tile.x - minX) * 256,
            (tile.y - minY) * 256,
            256,
            256
          );
        }
      });
      samModel.setGeoImage(canvas.toDataURL(), {
        extent: imageExtent,
        width: canvas.width,
        height: canvas.height
      });
      const base64 = canvas.toDataURL("image/jpeg");
      const index = base64.indexOf(",");
      const strBaseImg = base64 == null ? void 0 : base64.substring(index + 1);
      const action = "https://sam.lvisei.icu/api";
      const formData = new FormData();
      formData.append("image_path", strBaseImg);
      const res = await (await fetch(action, {
        body: formData,
        method: "post"
      })).arrayBuffer();
      try {
        const topRight = mapHelper.tileToLngLat(maxX + 1, minY, zoom);
        const bottomLeft = mapHelper.tileToLngLat(minX, maxY + 1, zoom);
        const topLeft = [bottomLeft[0], topRight[1]];
        const bottomRight = [topRight[0], bottomLeft[1]];
        const bounds = (0, import_turf.polygon)([
          [topRight, topLeft, bottomLeft, bottomRight, topRight]
        ]);
        setMarker(topLeft);
        setBound(bounds);
        setSource({ data: { type: "FeatureCollection", features: [bounds] } });
        bboxAutoFit([bounds]);
        samModel.setEmbedding(res);
        import_antd.message.success(t("map_control_group.sam.jiSuanWanCheng"));
      } catch (error) {
        console.log(error);
      }
    } catch (error) {
      import_antd.message.error(t("map_control_group.sam.jiSuanShiBai"));
      scene == null ? void 0 : scene.off("click", onMapClick);
    } finally {
      setLoading(false);
    }
  };
  (0, import_react.useEffect)(() => {
    const sam = new import_sam.SAMGeo({
      modelUrl: import_sam.MODEL_URL,
      wasmPaths: wasmPath
    });
    sam.initModel().then(() => {
      setSamModal(sam);
    }).catch(() => {
      import_antd.message.error(t("map_control_group.sam.diKuaiShiBieShiBei"));
    });
  }, []);
  (0, import_react.useEffect)(() => {
    if (!(0, import_lodash_es.isEmpty)(allLayerList)) {
      const targetLayer = allLayerList.find(
        (layer) => layer.id === "googleTileMap"
      );
      const selectLayer = allLayerList.find(
        (layer) => layer.id === import_constants.LayerId.PolygonLayer
      );
      const hoverLayer = allLayerList.find(
        (layer) => layer.id === "hoverLayer"
      );
      setHoverPolyonLayer(hoverLayer);
      setPolygonLayer(selectLayer);
      setTileLayer(targetLayer);
    }
  }, [allLayerList]);
  (0, import_react.useEffect)(() => {
    if (polygonLayer) {
      if (samOpen) {
        polygonLayer.on("unclick", onMapClick);
      } else {
        setSource({ data: { type: "FeatureCollection", features: [] } });
        setMarker(void 0);
        polygonLayer.off("unclick", onMapClick);
      }
    }
    return () => {
      polygonLayer == null ? void 0 : polygonLayer.off("unclick", onMapClick);
    };
  }, [samOpen, scene, onMapClick]);
  (0, import_react.useEffect)(() => {
    if (samOpen && samModel) {
      generateEmbedding();
    }
  }, [samOpen, samModel]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "bottomright" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Tooltip,
    {
      title: t("map_control_group.sam.zhiNengXuanZe"),
      placement: "left"
    },
    /* @__PURE__ */ import_react.default.createElement(import_antd.Spin, { spinning: loading }, /* @__PURE__ */ import_react.default.createElement(
      "button",
      {
        type: "button",
        className: style.L7EditorControl,
        onClick: () => {
          if (samModel) {
            setSamOpen(!samOpen);
          }
          if (samOpen) {
            import_antd.message.success(
              t("map_control_group.sam.zhiNengShiBieGuanBi")
            );
          }
        }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_iconfont.IconFont,
        {
          type: "icon-zhinengshibie",
          style: { fontSize: 20, color: samOpen ? "#1677ff" : "" },
          className: style.l7EditorIcon
        }
      )
    ))
  )), marker && /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.Marker,
    {
      lngLat: { lng: marker[0], lat: marker[1] },
      anchor: "center",
      offsets: [54, -16]
    },
    /* @__PURE__ */ import_react.default.createElement("div", { className: styles.marker }, t("map_control_group.sam.ziDongShiBie"))
  ), /* @__PURE__ */ import_react.default.createElement(import_larkmap.LineLayer, { ...options, source }));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SamControl
});
