// @antv/l7-editor/lib/hooks/use-draw-helper.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/hooks/use-draw-helper.ts
var use_draw_helper_exports = {};
__export(use_draw_helper_exports, {
  useDrawHelper: () => useDrawHelper
});
module.exports = __toCommonJS(use_draw_helper_exports);
var import_react_i18next = require("react-i18next");
function useDrawHelper() {
  const { t } = (0, import_react_i18next.useTranslation)();
  return {
    draw: `${t("map_contorl_group.draw.draw")}`,
    drawFinish: `${t("map_contorl_group.draw.drawFinish")}`,
    drawContinue: `${t("map_contorl_group.draw.drawContinue")}`,
    pointHover: `${t("map_contorl_group.draw.pontHover")}`,
    lineHover: `${t("map_contorl_group.draw.lineHover")}`,
    polygonHover: `${t("map_contorl_group.draw.lineHover")}`,
    midPointHover: `${t(`map_contorl_group.draw.midPointHover`)}`
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  useDrawHelper
});
