// @antv/l7-editor/lib/hooks/use-filter-features.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/hooks/use-filter-features.ts
var use_filter_features_exports = {};
__export(use_filter_features_exports, {
  filterFeatureAtom: () => filterFeatureAtom,
  isEmptyFilter: () => isEmptyFilter,
  useFilterFeatures: () => useFilterFeatures
});
module.exports = __toCommonJS(use_filter_features_exports);
var import_lodash_es = require("lodash-es");
var import_react = require("react");
var import_recoil = require("recoil");
var import_recoil2 = require("../recoil");
function isEmptyFilter(filter) {
  const { operator, value, type, field } = filter;
  return (0, import_lodash_es.isUndefined)(operator) || (0, import_lodash_es.isUndefined)(value) || (0, import_lodash_es.isUndefined)(type) || (0, import_lodash_es.isUndefined)(field) || value === "" || Number.isNaN(value) || value === null;
}
function numberFilter(filter, properties) {
  const newField = properties[filter.field];
  switch (filter.operator) {
    case "<":
      return newField < filter.value;
    case "<=":
      return newField <= filter.value;
    case "=":
      return newField === filter.value;
    case ">":
      return newField > filter.value;
    case ">=":
      return newField >= filter.value;
    default:
      if (Array.isArray(filter.value)) {
        const [value1, value2] = filter.value;
        return newField >= value1 && newField <= value2;
      }
  }
}
function stringFilter(filter, properties) {
  const newField = String(properties[filter.field]);
  const value = filter.value;
  switch (filter.operator) {
    case "NOT_IN":
      return !(value == null ? void 0 : value.includes(newField));
    case "IN":
      return value == null ? void 0 : value.includes(newField);
    case "LIKE":
      return (newField == null ? void 0 : newField.indexOf(value)) > -1;
    default:
      return (newField == null ? void 0 : newField.indexOf(value)) <= -1;
  }
}
var filterFeatureAtom = (0, import_recoil.atom)({
  key: "filterFeature",
  default: []
});
function useFilterFeatures() {
  const { features } = (0, import_recoil2.useFeature)();
  const { filters } = (0, import_recoil2.useFilter)();
  const setFilterFeature = (0, import_recoil.useSetRecoilState)(filterFeatureAtom);
  const filterFeatures = (0, import_recoil.useRecoilValue)(filterFeatureAtom);
  (0, import_react.useEffect)(() => {
    if ((0, import_lodash_es.isEmpty)(filters)) {
      setFilterFeature([...features]);
      return;
    }
    let newFeature = [...features];
    const setNotEmptyFilter = filters.filter((a) => !isEmptyFilter(a));
    const andFilters = setNotEmptyFilter.filter((item) => item.logic === "and");
    const orFilters = setNotEmptyFilter.filter((item) => item.logic === "or");
    if (!(0, import_lodash_es.isEmpty)(orFilters)) {
      newFeature = newFeature.filter(({ properties }) => {
        return orFilters.some((filter) => {
          const func = filter.type === "number" ? numberFilter : stringFilter;
          return func(filter, properties);
        });
      });
    }
    if (!(0, import_lodash_es.isEmpty)(andFilters)) {
      newFeature = newFeature.filter(({ properties }) => {
        return andFilters.every((filter) => {
          const func = filter.type === "number" ? numberFilter : stringFilter;
          return func(filter, properties);
        });
      });
    }
    setFilterFeature(newFeature);
  }, [features, filters, setFilterFeature]);
  return {
    features: filterFeatures
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  filterFeatureAtom,
  isEmptyFilter,
  useFilterFeatures
});
