// @antv/l7-editor/lib/components/map-control-group/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/index.tsx
var map_control_group_exports = {};
__export(map_control_group_exports, {
  MapControlGroup: () => MapControlGroup
});
module.exports = __toCommonJS(map_control_group_exports);
var import_larkmap = require("@antv/larkmap");
var import_react = __toESM(require("react"));
var import_recoil = require("../../recoil");
var import_administrative_select_control = require("./administrative-select-control");
var import_auto_control = require("./auto-control");
var import_clear_control = require("./clear-control");
var import_draw_control = __toESM(require("./draw-control"));
var import_export_image_control = require("./export-image-control");
var import_filter_form_list_control = __toESM(require("./filter-form-list-control"));
var import_l7_option_control = require("./l7-option-control");
var import_layer_color_control = __toESM(require("./layer-color-control"));
var import_location_search_control = __toESM(require("./location-search-control"));
var import_map_administrative_control = require("./map-administrative-control");
var import_map_theme_control = __toESM(require("./map-theme-control"));
var import_official_layer_control = require("./official-layer-control");
var import_styles = __toESM(require("./styles"));
var import_text_layer_control = require("./text-layer-control");
var DefaultMapControl = {
  drawControl: true,
  clearControl: true,
  zoomControl: true,
  scaleControl: true,
  locationSearchControl: true,
  mouseLocationControl: true,
  filterControl: true,
  officialLayerControl: true,
  mapThemeControl: true,
  geoLocateControl: true,
  layerColorControl: true,
  autoControl: true,
  fullscreenControl: true,
  administrativeSelectControl: true,
  mapAdministrativeControl: true,
  logoControl: true,
  textLayerControl: true,
  exportImageControl: true,
  L7MapOptionControl: true
};
var MapControlGroup = ({
  mapControl
}) => {
  const { baseMap } = (0, import_recoil.useGlobal)();
  const styles = (0, import_styles.default)();
  const [isControlGroupState, setIsControlGroup] = (0, import_react.useState)(DefaultMapControl);
  (0, import_react.useEffect)(() => {
    setIsControlGroup({ ...DefaultMapControl, ...mapControl });
  }, [mapControl]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, isControlGroupState.logoControl && /* @__PURE__ */ import_react.default.createElement(import_larkmap.LogoControl, { position: "leftbottom" }), isControlGroupState.scaleControl && /* @__PURE__ */ import_react.default.createElement(import_larkmap.ScaleControl, { position: "leftbottom", className: styles.scalesControl }), isControlGroupState.mouseLocationControl && /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.MouseLocationControl,
    {
      position: "leftbottom",
      className: styles.mouseLocatio
    }
  ), isControlGroupState.drawControl && /* @__PURE__ */ import_react.default.createElement(import_draw_control.default, null), isControlGroupState.clearControl && /* @__PURE__ */ import_react.default.createElement(import_clear_control.ClearControl, null), isControlGroupState.zoomControl && /* @__PURE__ */ import_react.default.createElement(import_larkmap.ZoomControl, { className: styles.zoom, showZoom: true, position: "rightbottom" }), isControlGroupState.mapAdministrativeControl && /* @__PURE__ */ import_react.default.createElement(import_map_administrative_control.MapAdministrativeControl, null), isControlGroupState.administrativeSelectControl && /* @__PURE__ */ import_react.default.createElement(import_administrative_select_control.AdministrativeSelect, null), isControlGroupState.locationSearchControl && /* @__PURE__ */ import_react.default.createElement(import_location_search_control.default, null), isControlGroupState.filterControl && /* @__PURE__ */ import_react.default.createElement(import_filter_form_list_control.default, null), baseMap === "Gaode" && isControlGroupState.officialLayerControl && /* @__PURE__ */ import_react.default.createElement(import_official_layer_control.OfficialLayerControl, null), isControlGroupState.mapThemeControl && /* @__PURE__ */ import_react.default.createElement(import_map_theme_control.default, null), isControlGroupState.geoLocateControl && /* @__PURE__ */ import_react.default.createElement(import_larkmap.GeoLocateControl, { position: "topleft", className: styles.fullScreen }), isControlGroupState.layerColorControl && /* @__PURE__ */ import_react.default.createElement(import_layer_color_control.default, null), isControlGroupState.autoControl && /* @__PURE__ */ import_react.default.createElement(import_auto_control.AutoControl, null), isControlGroupState.fullscreenControl && /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.FullscreenControl,
    {
      position: "bottomright",
      className: styles.fullScreen
    }
  ), isControlGroupState.textLayerControl && /* @__PURE__ */ import_react.default.createElement(import_text_layer_control.TextLayerControl, null), isControlGroupState.exportImageControl && /* @__PURE__ */ import_react.default.createElement(import_export_image_control.ExportImage, null), isControlGroupState.L7MapOptionControl && /* @__PURE__ */ import_react.default.createElement(import_l7_option_control.L7MapOptionControl, null));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MapControlGroup
});
