// @antv/l7-editor/lib/components/map-control-group/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyles = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const {
    colorBgContainer,
    colorBorder,
    colorTextDescription,
    colorText,
    boxShadow,
    colorIcon
  } = token;
  return {
    L7EditorControl: import_css.css`
      display: inline-flex;
      align-items: center;
      justify-content: center;
      border-radius: 2px;
      width: 28px;
      height: 28px;
      padding: 0;
      text-align: center;
      background: ${colorBgContainer};
      box-shadow: ${boxShadow} !important;
      border-width: 0;
      color: ${colorIcon};
      outline: 0;
      cursor: pointer;
      &:hover {
        background: ${colorBgContainer};
        svg {
          color: ${colorText};
        }
      }
    `,
    l7EditorIcon: import_css.css`
      font-size: 16px;
      border: none;
      cursor: pointer;
      color: ${colorIcon};
    `,
    zoom: import_css.css`
      .l7-button-control {
        color: ${colorTextDescription};
        font-size: 14px;
        background: ${colorBgContainer} !important;
        border-color: ${colorBorder};

        svg {
          fill: ${colorTextDescription} !important;
        }

        &:hover {
          svg {
            fill: ${colorText} !important;
          }
        }
      }
      .l7-button-control:first-child {
        border-bottom: 1px solid ${colorBorder};
      }
    `,
    scalesControl: import_css.css`
      .l7-control-scale-line {
        color: ${colorTextDescription};
        background: ${colorBgContainer} !important;
        border-color: ${colorBorder};
      }
    `,
    mouseLocatio: import_css.css`
      color: ${colorTextDescription} !important;
      background: ${colorBgContainer} !important;
      border-color: ${colorBorder};
    `,
    fullScreen: import_css.css`
      color: ${colorTextDescription};
      background: ${colorBgContainer} !important;
      border-color: ${colorBorder};

      svg {
        fill: ${colorTextDescription} !important;
      }

      &:hover {
        svg {
          fill: ${colorText} !important;
        }
      }
    `
  };
};
var styles_default = useStyles;
