// @antv/l7-editor/es/components/app-header/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { DingtalkOutlined, FlagOutlined, GithubOutlined, SaveOutlined } from '@ant-design/icons';
import { useKeyPress, useLocalStorageState } from 'ahooks';
import { Button, Divider, Dropdown, Popover, Space, Tooltip, Tour } from 'antd';
import classNames from 'classnames';
import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { LocalStorageKey } from "../../constants";
import { useFeature, useGlobal } from "../../recoil";
import { IconFont } from "../iconfont";
import { BaseMap } from "./base-map";
import DownloadBtn from "./btn/download-btn";
import HandBackBtn from "./btn/handback-btn";
import I18nBtn from "./btn/i18n-btn";
import { ImportBtn } from "./btn/import-btn";
import { SettingBtn } from "./btn/setting-btn";
import useStyle from "./styles";
var isTooBar = {
  logo: true,
  import: true,
  download: true,
  guide: true,
  help: true,
  setting: true,
  theme: true,
  dingTalk: true,
  i18n: true,
  github: true,
  baseMap: true
};
export var AppHeader = function AppHeader(_ref) {
  var toolbar = _ref.toolbar;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var _useState = useState({
      key: '',
      open: false
    }),
    _useState2 = _slicedToArray(_useState, 2),
    open = _useState2[0],
    setOpen = _useState2[1];
  var _useGlobal = useGlobal(),
    autoFitBounds = _useGlobal.autoFitBounds,
    theme = _useGlobal.theme,
    setTheme = _useGlobal.setTheme;
  var _useFeature = useFeature(),
    saveEditorText = _useFeature.saveEditorText,
    savable = _useFeature.savable,
    bboxAutoFit = _useFeature.bboxAutoFit;
  var _useState3 = useState(isTooBar),
    _useState4 = _slicedToArray(_useState3, 2),
    isTooBarState = _useState4[0],
    setIsTooBar = _useState4[1];
  var styles = useStyle();
  var _useLocalStorageState = useLocalStorageState(LocalStorageKey.firstOpening, {
      defaultValue: true
    }),
    _useLocalStorageState2 = _slicedToArray(_useLocalStorageState, 2),
    firstOpen = _useLocalStorageState2[0],
    setFirstOpen = _useLocalStorageState2[1];
  useEffect(function () {
    if (firstOpen) {
      setOpen({
        key: 'basics',
        open: true
      });
      setFirstOpen(false);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  var DropdownMenuItems = [{
    key: 'basics',
    label: t('app_header.constants.pingTaiJiChuShi')
  }, {
    key: 'function',
    label: t('app_header.constants.quanGongNengShiYong')
  }];
  var steps = [{
    title: t('import_btn.index.shangChuan'),
    description: t('app_header.constants.shangChuanGEO'),
    cover: /*#__PURE__*/React.createElement("img", {
      alt: t('app_header.constants.shangChuanPNG'),
      src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*RJfORKi3ntsAAAAAAAAAAAAADjWqAQ/original"
    }),
    target: function target() {
      return document.getElementById('l7-editor-upload');
    }
  }, {
    title: t('app_header.constants.bianJiQi'),
    description: t('app_header.constants.keYiTongGuoBian'),
    target: function target() {
      return document.getElementById('l7-editor-panel');
    },
    placement: 'left'
  }, {
    title: t('app_header.constants.huiZhi'),
    cover: /*#__PURE__*/React.createElement("img", {
      alt: t('app_header.constants.huiZhiPNG'),
      src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*NNcDQrzkdt8AAAAAAAAAAAAADjWqAQ/original"
    }),
    description: t('app_header.constants.keYiJiHuoDi'),
    target: function target() {
      return document.getElementById('l7-editor-draw');
    },
    placement: 'right'
  }, {
    title: t('app_header.constants.baoCun'),
    description: t('app_header.constants.shiYongCTR'),
    target: function target() {
      return document.getElementById('l7-editor-save');
    }
  }, {
    title: t('app_header.constants.xuanRan'),
    description: t('app_header.constants.diTuShangChaKan'),
    target: function target() {
      return document.getElementById('l7-editor-map');
    },
    placement: 'right'
  }];
  var functionSteps = [{
    title: t('import_btn.index.shangChuan'),
    description: t('app_header.constants.shangChuanGEO'),
    cover: /*#__PURE__*/React.createElement("img", {
      alt: t('app_header.constants.shangChuanPNG'),
      src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*RJfORKi3ntsAAAAAAAAAAAAADjWqAQ/original"
    }),
    target: function target() {
      return document.getElementById('l7-editor-upload');
    }
  }, {
    title: t('app_header.constants.baoCun'),
    description: t('app_header.constants.shiYongCTR'),
    target: function target() {
      return document.getElementById('l7-editor-save');
    }
  }, {
    title: t('btn.download_btn.xiaZai'),
    description: t('app_header.constants.dianJiXiaZaiKe'),
    target: function target() {
      return document.getElementById('l7-editor-download');
    }
  }, {
    title: t('btn.setting_btn.sheZhi'),
    description: t('app_header.constants.yongHuKeSheZhi'),
    target: function target() {
      return document.getElementById('l7-editor-set');
    }
  }, {
    title: t('app_header.constants.zhuTi'),
    description: t('app_header.constants.dianJiKeQieHuan'),
    target: function target() {
      return document.getElementById('l7-editor-theme');
    }
  }, {
    title: t('app_header.constants.xingZhengQuXuanZe'),
    description: t('app_header.constants.keYiKuaiSuXuan'),
    target: function target() {
      return document.getElementById('l7-editor-administrativeSelect');
    }
  }, {
    title: t('app_header.constants.chengShiChaXun'),
    description: t('app_header.constants.souSuoChengShiHou'),
    target: function target() {
      return document.getElementById('l7-editor-citySelect');
    }
  }, {
    title: t('app_header.constants.huiZhi'),
    cover: /*#__PURE__*/React.createElement("img", {
      alt: t('app_header.constants.huiZhiPNG'),
      src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*NNcDQrzkdt8AAAAAAAAAAAAADjWqAQ/original"
    }),
    description: t('app_header.constants.keYiJiHuoDi'),
    target: function target() {
      return document.getElementById('l7-editor-draw');
    },
    placement: 'right'
  }, {
    title: t('app_header.constants.qingChu'),
    description: t('app_header.constants.dianJiQingChuAn'),
    target: function target() {
      return document.getElementById('l7-editor-clear');
    }
  }, {
    title: t('app_header.constants.shaiXuan'),
    description: t('app_header.constants.keGenJuShuJu'),
    target: function target() {
      return document.getElementById('l7-editor-filter');
    }
  }, {
    title: t('app_header.constants.ziDongSuoFang'),
    description: t('app_header.constants.dianJiZiDongSuo'),
    target: function target() {
      return document.getElementById('l7-editor-auto');
    }
  }, {
    title: t('app_header.constants.yanSeXuanZeQi'),
    description: t('app_header.constants.xuanZeYanSeLai'),
    target: function target() {
      return document.getElementById('l7-editor-color');
    }
  }, {
    title: t('app_header.constants.zhuTiSe'),
    description: t('app_header.constants.xuanZeBuTongZhu'),
    //@ts-ignore
    target: function target() {
      return document.getElementsByClassName('l7-editor-mapTheme')[0];
    }
  }, {
    title: t('app_header.constants.guanFangTuCeng'),
    description: t('app_header.constants.keXuanZeBuTong'),
    target: function target() {
      return document.getElementById('l7-editor-aMap');
    }
  }, {
    title: t('text-layer-control_wenBenTuCengPeiZhi'),
    description: t('text-layer-control_description'),
    target: function target() {
      return document.getElementById('text-layer-control');
    }
  }, {
    title: t('app_header.constants.gEOJS'),
    description: t('app_header.constants.keYiTongGuoBian'),
    target: function target() {
      return document.getElementById('l7-editor-panel');
    },
    placement: 'left'
  }, {
    title: t('app_header.constants.wKTBianJi'),
    description: t('app_header.constants.keYiTongGuoW'),
    target: function target() {
      return document.getElementById('l7-editor-wkt');
    },
    placement: 'left'
  }, {
    title: t('app_header.constants.biaoGe'),
    description: t('app_header.constants.keYiTongGuoBiao'),
    target: function target() {
      return document.getElementById('l7-editor-table');
    },
    placement: 'left'
  }];
  var onSave = function onSave() {
    if (!savable) {
      return;
    }
    var features = saveEditorText();
    if (autoFitBounds) {
      bboxAutoFit(features);
    }
  };
  useKeyPress(['ctrl.s', 'meta.s'], function (e) {
    e.preventDefault();
    onSave();
  });
  var onDownload = function onDownload(key) {
    setOpen({
      key: key,
      open: true
    });
  };
  useEffect(function () {
    setIsTooBar(_objectSpread(_objectSpread({}, isTooBar), toolbar));
  }, [toolbar]);
  return /*#__PURE__*/React.createElement("div", {
    className: classNames([styles.mapHeader, 'l7-editor-header'])
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.mapHeaderLeft
  }, isTooBarState.logo && /*#__PURE__*/React.createElement("div", {
    className: classNames([styles.mapHeaderLogo, 'l7-editor-header__logo'])
  }, /*#__PURE__*/React.createElement("img", {
    src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*RSdESJd70P8AAAAAAAAAAAAADjWqAQ/original"
  }), /*#__PURE__*/React.createElement("span", {
    className: styles.mapHeaderTitle
  }, "L7 Editor")), isTooBarState.logo && isTooBarState.baseMap && /*#__PURE__*/React.createElement(Divider, {
    type: "vertical"
  }), isTooBarState.baseMap && /*#__PURE__*/React.createElement(BaseMap, null)), /*#__PURE__*/React.createElement("div", {
    className: styles.mapHeaderRight
  }, /*#__PURE__*/React.createElement(Space, null, isTooBarState.import && /*#__PURE__*/React.createElement(ImportBtn, null), /*#__PURE__*/React.createElement(Tooltip, {
    title: t('app_header.index.baoCunShuJu'),
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(Button, {
    id: "l7-editor-save",
    icon: /*#__PURE__*/React.createElement(SaveOutlined, null),
    disabled: !savable,
    onClick: onSave
  }, t('app_header.constants.baoCun'))), isTooBarState.download && /*#__PURE__*/React.createElement(DownloadBtn, null), isTooBarState.guide && /*#__PURE__*/React.createElement(Dropdown, {
    menu: {
      items: DropdownMenuItems,
      onClick: function onClick(_ref2) {
        var key = _ref2.key;
        onDownload(key);
      }
    },
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(Button, {
    icon: /*#__PURE__*/React.createElement(FlagOutlined, null)
  }, t('app_header.index.yinDao'))), isTooBarState.help && /*#__PURE__*/React.createElement(HandBackBtn, null), isTooBarState.setting && /*#__PURE__*/React.createElement(SettingBtn, null)), /*#__PURE__*/React.createElement(Divider, {
    type: "vertical"
  }), /*#__PURE__*/React.createElement(Space, null, isTooBarState.dingTalk && /*#__PURE__*/React.createElement(Popover, {
    placement: "bottom",
    title: t('app_header.index.dingTalk'),
    content: /*#__PURE__*/React.createElement("img", {
      style: {
        width: 300
      },
      src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*CTHXQbdMcDcAAAAAAAAAAAAADjWqAQ/original"
    }),
    trigger: "hover"
  }, /*#__PURE__*/React.createElement(Button, {
    icon: /*#__PURE__*/React.createElement(DingtalkOutlined, null)
  })), isTooBarState.theme && /*#__PURE__*/React.createElement(Tooltip, {
    title: t('app_header.index.zhutiqiehuan'),
    trigger: "hover",
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(Button, {
    id: "l7-editor-theme",
    icon: /*#__PURE__*/React.createElement(IconFont, {
      className: styles.themeIcon,
      type: theme === 'dark' ? 'icon-a-qingtianwanshang' : 'icon-taiyang'
    })
    // className={styles.theme}
    ,
    onClick: function onClick() {
      setTheme(theme === 'dark' ? 'light' : 'dark');
    }
  })), isTooBarState.i18n && /*#__PURE__*/React.createElement(I18nBtn, null), isTooBarState.github && /*#__PURE__*/React.createElement(Tooltip, {
    title: 'Github',
    trigger: "hover",
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(Button, {
    icon: /*#__PURE__*/React.createElement(GithubOutlined, null),
    onClick: function onClick() {
      window.open('https://github.com/antvis/L7Editor');
    }
  })))), /*#__PURE__*/React.createElement(Tour, {
    open: open.open,
    onClose: function onClose() {
      return setOpen({
        key: '',
        open: false
      });
    },
    steps: open.key === 'basics' ? steps : functionSteps
  }));
};
