// @antv/l7-editor/lib/components/app-header/btn/import-btn/file-upload.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/btn/import-btn/file-upload.tsx
var file_upload_exports = {};
__export(file_upload_exports, {
  default: () => file_upload_default
});
module.exports = __toCommonJS(file_upload_exports);
var import_icons = require("@ant-design/icons");
var import_turf = require("@turf/turf");
var import_antd = require("antd");
var import_lodash = require("lodash");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_wellknown = require("wellknown");
var import_variable_type = require("../../../../constants/variable-type");
var import_upload = require("../../../../utils/upload");
var import_styles = __toESM(require("../../styles"));
var { Dragger } = import_antd.Upload;
var FileUpload = (0, import_react.forwardRef)(function FileUpload2({}, ref) {
  const [uploadData, setUploadData] = (0, import_react.useState)([]);
  const [fileList, setFileList] = (0, import_react.useState)([]);
  const [selectList, setSelectList] = (0, import_react.useState)([]);
  const [form] = import_antd.Form.useForm();
  const { t } = (0, import_react_i18next.useTranslation)();
  const styles = (0, import_styles.default)();
  const customRequest = (uploadRequestOption) => {
    const { file, onSuccess, onError } = uploadRequestOption;
    (0, import_upload.parserFileToSource)(file, t).then((dataSource) => {
      var _a, _b;
      if (!(dataSource == null ? void 0 : dataSource.columns)) {
        if ((_b = (_a = dataSource.data) == null ? void 0 : _a.features) == null ? void 0 : _b.length) {
          const newData = uploadData;
          newData.push({
            id: dataSource.id,
            features: dataSource.data.features
          });
          setUploadData(newData);
        }
      } else {
        const wktValue = dataSource.data[0];
        let val = void 0;
        for (const key of Object.keys(wktValue)) {
          const values = wktValue[key];
          if ((0, import_upload.isWkt)(values)) {
            form.setFieldValue(dataSource.id, key);
            val = key;
          } else if ((0, import_upload.isGeometry)(values)) {
            form.setFieldValue(dataSource.id, key);
            val = key;
          }
        }
        if (val) {
          const data = dataSource == null ? void 0 : dataSource.data.map((value) => {
            return value[val];
          });
          const propertiesList = dataSource == null ? void 0 : dataSource.data.map((v) => {
            const properties = (0, import_lodash.cloneDeep)(v);
            delete properties[val];
            return { ...properties };
          });
          let newGeoJson = [];
          if ((0, import_upload.isWkt)(data[0])) {
            newGeoJson = data.map((v, index) => {
              const geometry = (0, import_wellknown.parse)(v);
              return (0, import_turf.feature)(geometry, { ...propertiesList[index] });
            });
          } else if ((0, import_upload.isGeometry)(data[0])) {
            newGeoJson = data.map((v, index) => {
              return (0, import_turf.feature)(JSON.parse(v), { ...propertiesList[index] });
            });
          }
          const newDates = uploadData;
          newDates.push({ features: newGeoJson, id: dataSource == null ? void 0 : dataSource.id });
          setUploadData(newDates);
        }
        setSelectList((pre) => [...pre, dataSource]);
      }
      onSuccess();
    }).catch(() => {
      onError();
      import_antd.message.error(t("import_btn.file_upload.shuJuGeShiBu"));
    });
  };
  (0, import_react.useImperativeHandle)(
    ref,
    () => ({
      getData: () => new Promise((resolve, reject) => {
        if (!uploadData.length) {
          reject(t("import_btn.file_upload.qingTianJiaWenJian"));
        }
        const isErrorList = fileList.filter(
          (item) => item.status === "error"
        );
        if (!!isErrorList.length) {
          reject(t("import_btn.file_upload.qingShanChuBaoCuo"));
        }
        const data = uploadData.map((item) => item.features).flat();
        if (import_variable_type.FeatureCollectionVT.check((0, import_turf.featureCollection)(data))) {
          resolve((0, import_turf.featureCollection)(data));
        } else {
          import_antd.message.info(t("import_btn.file_upload.qingJianChaShuJu"));
        }
      })
    }),
    [fileList, t, uploadData]
  );
  const selectItem = (0, import_react.useMemo)(() => {
    if (selectList.length) {
      const data = selectList.map((item, index) => {
        const options = item.columns.map((option) => {
          return {
            value: option,
            label: option
          };
        });
        return /* @__PURE__ */ import_react.default.createElement(
          import_antd.Form.Item,
          {
            name: item.id,
            label: `${t("import_btn.file_upload.wenJian")}${index + 1}`,
            key: item.id
          },
          /* @__PURE__ */ import_react.default.createElement(
            import_antd.Select,
            {
              options,
              style: { width: 300 },
              onChange: (e) => {
                const newData1 = selectList.find((v) => v.id === item.id);
                const newData2 = newData1 == null ? void 0 : newData1.data.map((value) => {
                  return value[e];
                });
                const propertiesList = newData1 == null ? void 0 : newData1.data.map((v) => {
                  const properties = (0, import_lodash.cloneDeep)(v);
                  delete properties[e];
                  return { ...properties };
                });
                let newGeoJson = [];
                if ((0, import_upload.isWkt)(newData2[0])) {
                  newGeoJson = newData2.map(
                    (v, geojsonIndex) => {
                      const geometry = (0, import_wellknown.parse)(v);
                      return (0, import_turf.feature)(geometry, {
                        ...propertiesList[geojsonIndex]
                      });
                    }
                  );
                } else if ((0, import_upload.isGeometry)(newData2[0])) {
                  newGeoJson = newData2.map(
                    (v, geojsonIndex) => {
                      return (0, import_turf.feature)(JSON.parse(v), {
                        ...propertiesList[geojsonIndex]
                      });
                    }
                  );
                } else {
                  import_antd.message.error(
                    t("import_btn.file_upload.ciZiDuanFeiDiLiZiDuan")
                  );
                }
                const newDates = uploadData.filter(
                  (updateItem) => updateItem.id !== (newData1 == null ? void 0 : newData1.id)
                );
                newDates.push({ features: newGeoJson, id: newData1 == null ? void 0 : newData1.id });
                setUploadData(newDates);
              }
            }
          )
        );
      });
      return data;
    }
  }, [selectList, t, uploadData]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_antd.Form, { form }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      name: "file",
      rules: [{ required: true }],
      style: { marginTop: 16, marginBottom: 4 }
    },
    /* @__PURE__ */ import_react.default.createElement(
      Dragger,
      {
        accept: ".json,.geojson,.kml,.wkt,.csv,.xlsx,.xls",
        customRequest,
        multiple: true,
        onRemove: (file) => {
          const newData = uploadData.filter((item) => item.id !== file.uid);
          setUploadData(newData);
          const newSelect = selectList.filter(
            (item) => item.id !== file.uid
          );
          setSelectList(newSelect);
          return true;
        },
        onChange: (file) => {
          setFileList(file.fileList);
        }
      },
      /* @__PURE__ */ import_react.default.createElement("div", { className: styles.upload }, /* @__PURE__ */ import_react.default.createElement("span", { style: { fontSize: 40 } }, /* @__PURE__ */ import_react.default.createElement(import_icons.FileTextOutlined, null)), /* @__PURE__ */ import_react.default.createElement("div", null, t("import_btn.file_upload.dianJiHuoJiangWenJian")))
    )
  ), /* @__PURE__ */ import_react.default.createElement("div", { style: { color: "#777" } }, t("import_btn.file_upload.jinZhiChiJS")), selectList.length ? /* @__PURE__ */ import_react.default.createElement("div", { style: { display: "flex" } }, /* @__PURE__ */ import_react.default.createElement("span", { className: styles.uploadTitle }, t("import_btn.file_upload.diLiZiDuanXuanZe")), /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: t("import_btn.file_upload.dangQianJinZhiChi") }, /* @__PURE__ */ import_react.default.createElement(import_icons.QuestionCircleOutlined, null))) : null, selectItem));
});
var file_upload_default = FileUpload;
