// @antv/l7-editor/lib/components/app-header/btn/import-btn/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/btn/import-btn/index.tsx
var import_btn_exports = {};
__export(import_btn_exports, {
  ImportBtn: () => ImportBtn
});
module.exports = __toCommonJS(import_btn_exports);
var import_icons = require("@ant-design/icons");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_geojson_editor = require("../../../../components/geojson-editor");
var import_variable_type = require("../../../../constants/variable-type");
var import_recoil = require("../../../../recoil");
var import_file_upload = __toESM(require("./file-upload"));
var import_url_upload = __toESM(require("./url-upload"));
var ImportBtn = () => {
  const [isModalOpen, setIsModalOpen] = (0, import_react.useState)(false);
  const { resetFeatures, features, bboxAutoFit } = (0, import_recoil.useFeature)();
  const [confirmLoading, setConfirmLoading] = (0, import_react.useState)(false);
  const [activeTab, setActiveTab] = (0, import_react.useState)("file");
  const [selectRadio, setSelectRadio] = (0, import_react.useState)("cover");
  const formRef = (0, import_react.useRef)(null);
  const { t } = (0, import_react_i18next.useTranslation)();
  const items = [
    {
      key: "file",
      label: /* @__PURE__ */ import_react.default.createElement("div", null, t("import_btn.file_upload.wenJianShangChuan")),
      children: /* @__PURE__ */ import_react.default.createElement(import_file_upload.default, { ref: formRef })
    },
    {
      key: "url",
      label: /* @__PURE__ */ import_react.default.createElement("div", null, t("import_btn.index.uRLShangChuan")),
      children: /* @__PURE__ */ import_react.default.createElement(import_url_upload.default, { ref: formRef })
    },
    // {
    //   key: 'lngLatUpload',
    //   label: <div>经纬度上传</div>,
    //   children: <LngLatImportBtn ref={formRef} />,
    // },
    {
      key: "script",
      label: /* @__PURE__ */ import_react.default.createElement("div", null, t("import_btn.index.jAVAS")),
      children: /* @__PURE__ */ import_react.default.createElement("div", { style: { width: "100%", height: 300 } }, /* @__PURE__ */ import_react.default.createElement(import_geojson_editor.GeoJsonEditor, { language: "javascript", ref: formRef }))
    }
  ];
  const handleCancel = () => {
    setIsModalOpen(false);
    setSelectRadio("cover");
  };
  const checkWithRestData = async () => {
    var _a;
    setConfirmLoading(true);
    try {
      const fc = await ((_a = formRef.current) == null ? void 0 : _a.getData());
      if (import_variable_type.FeatureCollectionVT.check(fc)) {
        const newFeatures = selectRadio === "cover" ? fc.features : [...features, ...fc.features];
        resetFeatures(newFeatures);
        bboxAutoFit(newFeatures);
        handleCancel();
      }
    } catch (error) {
      import_antd.message.error(`${error}`);
    }
    setConfirmLoading(false);
  };
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: t("import_btn.index.shangChuanShuJu"), placement: "bottom" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Button,
    {
      icon: /* @__PURE__ */ import_react.default.createElement(import_icons.CloudUploadOutlined, null),
      onClick: () => setIsModalOpen(true),
      id: "l7-editor-upload"
    },
    t("import_btn.index.shangChuan")
  )), isModalOpen && /* @__PURE__ */ import_react.default.createElement(
    import_antd.Modal,
    {
      title: t("import_btn.index.shangChuan"),
      open: isModalOpen,
      onOk: checkWithRestData,
      onCancel: handleCancel,
      destroyOnClose: true,
      confirmLoading,
      width: 1e3
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Tabs,
      {
        activeKey: activeTab,
        className: "map-content__right",
        items,
        destroyInactiveTabPane: true,
        onChange: (e) => {
          setActiveTab(e);
        }
      }
    ),
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Form.Item,
      {
        label: t("import_btn.index.shuJuCaoZuo"),
        rules: [{ required: true }],
        style: { marginTop: 8 }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Radio.Group,
        {
          value: selectRadio,
          onChange: (e) => {
            setSelectRadio(e.target.value);
          }
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { value: "cover" }, t("import_btn.index.fuGai")),
        /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { value: "merge" }, t("import_btn.index.zhuiJia"))
      )
    )
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ImportBtn
});
