// @antv/l7-editor/lib/components/app-header/btn/import-btn/lnglat-import-btn.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/btn/import-btn/lnglat-import-btn.tsx
var lnglat_import_btn_exports = {};
__export(lnglat_import_btn_exports, {
  default: () => lnglat_import_btn_default
});
module.exports = __toCommonJS(lnglat_import_btn_exports);
var import_turf = require("@turf/turf");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../../../recoil");
var LngLatImportBtn = (0, import_react.forwardRef)(({}, ref) => {
  const {
    lngLatText,
    setLngLatText,
    importLngLatText,
    lngLatImportType,
    setLngLatImportType
  } = (0, import_recoil.useLnglat)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const LngLatImportTypeOptions = [
    {
      label: t("import_btn.lnglat_import_btn.dian"),
      value: "Point"
    },
    {
      label: t("import_btn.lnglat_import_btn.xian"),
      value: "LingString"
    },
    {
      label: t("import_btn.lnglat_import_btn.mian"),
      value: "Polygon"
    }
  ];
  (0, import_react.useImperativeHandle)(
    ref,
    () => ({
      getData: () => new Promise((resolve, reject) => {
        if (!lngLatText) {
          reject(t("import_btn.lnglat_import_btn.qingShuRuJingWei"));
        }
        const data = importLngLatText(lngLatText);
        resolve((0, import_turf.featureCollection)(data));
        reject(t("import_btn.lnglat_import_btn.lNGLA"));
      })
    }),
    [lngLatText, importLngLatText, t]
  );
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_antd.Form, null, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      style: { marginTop: 16 },
      label: t("import_btn.lnglat_import_btn.shuJuLeiXing")
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Radio.Group,
      {
        value: lngLatImportType,
        buttonStyle: "solid",
        onChange: (e) => {
          setLngLatImportType(e.target.value);
        }
      },
      LngLatImportTypeOptions.map((item) => /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { key: item.value, value: item.value }, item.label))
    )
  ), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      style: { marginTop: 16 },
      label: t("import_btn.lnglat_import_btn.shuJuNeiRong")
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Input.TextArea,
      {
        placeholder: t("import_btn.lnglat_import_btn.qingShuRuLianXu"),
        rows: 10,
        onChange: (e) => {
          setLngLatText(e.target.value);
        }
      }
    )
  )));
});
var lnglat_import_btn_default = LngLatImportBtn;
