// @antv/l7-editor/lib/components/app-header/btn/import-btn/url-upload.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/btn/import-btn/url-upload.tsx
var url_upload_exports = {};
__export(url_upload_exports, {
  default: () => url_upload_default
});
module.exports = __toCommonJS(url_upload_exports);
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_utils = require("../../../../utils");
var UrlUpload = (0, import_react.forwardRef)(({}, ref) => {
  const [inputValue, setInputValue] = (0, import_react.useState)("");
  const { t } = (0, import_react_i18next.useTranslation)();
  const [radioValue, setRadioValue] = (0, import_react.useState)("GeoJSON");
  (0, import_react.useImperativeHandle)(
    ref,
    () => ({
      getData: () => new Promise((resolve, reject) => {
        if (inputValue) {
          resolve((0, import_utils.getUrlFeatureCollection)(inputValue, radioValue, t));
          reject(t("import_btn.url_upload.shuJuGeShiCuo"));
        } else {
          reject(t("import_btn.url_upload.qingShuRuWenBen"));
        }
      })
    }),
    [inputValue, radioValue, t]
  );
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_antd.Form, { layout: "vertical" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      name: "urlType",
      label: t("import_btn.url_upload.shuJuLeiXing"),
      rules: [{ required: true }],
      style: { marginTop: 16, marginBottom: 4 }
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Radio.Group,
      {
        defaultValue: "GeoJSON",
        onChange: (e) => {
          setRadioValue(e.target.value);
        }
      },
      /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { value: "GeoJSON" }, "GeoJSON"),
      /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { value: "WKT" }, "WKT"),
      /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { value: "KML" }, "KML")
    )
  ), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      name: "url",
      label: t("import_btn.url_upload.uRLDiZhi"),
      rules: [{ required: true }],
      style: { marginTop: 16, marginBottom: 4 }
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Input,
      {
        placeholder: "https://...",
        onChange: (e) => {
          setInputValue(e.target.value);
        }
      }
    )
  )));
});
var url_upload_default = UrlUpload;
