// @antv/l7-editor/es/components/map-control-group/official-layer-control/index.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { CaretRightOutlined, DeleteOutlined, FormOutlined, MinusCircleOutlined, PlusOutlined, UploadOutlined } from '@ant-design/icons';
import { CustomControl, RasterLayer } from '@antv/larkmap';
import { useLocalStorageState } from 'ahooks';
import { Button, Form, Input, Modal, Popconfirm, Space, Upload } from 'antd';
import classNames from 'classnames';
import { cloneDeep } from 'lodash-es';
import React, { useMemo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { GOOGLE_TILE_MAP_URL, OfficeLayerEnum } from "../../../constants";
import { useGlobal } from "../../../recoil";
import useStyle from "./styles";
var layout = {
  labelCol: {
    span: 4
  },
  wrapperCol: {
    span: 16
  }
};
var enLayout = {
  labelCol: {
    span: 6
  },
  wrapperCol: {
    span: 15
  }
};
export function OfficialLayerControl() {
  var _Form$useForm = Form.useForm(),
    _Form$useForm2 = _slicedToArray(_Form$useForm, 1),
    form = _Form$useForm2[0];
  var styles = useStyle();
  var _useGlobal = useGlobal(),
    layerType = _useGlobal.layerType,
    setLayerType = _useGlobal.setLayerType,
    customTiles = _useGlobal.customTiles,
    setCustomTiles = _useGlobal.setCustomTiles,
    locale = _useGlobal.locale;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var _useState = useState(layerType.length ? layerType[0] : OfficeLayerEnum.VectorMap),
    _useState2 = _slicedToArray(_useState, 2),
    radioValue = _useState2[0],
    setRadioValue = _useState2[1];
  var _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    isModalOpen = _useState4[0],
    setIsModalOpen = _useState4[1];
  var _useState5 = useState(false),
    _useState6 = _slicedToArray(_useState5, 2),
    isEdit = _useState6[0],
    setIsEdit = _useState6[1];
  var _useState7 = useState(-1),
    _useState8 = _slicedToArray(_useState7, 2),
    editIndex = _useState8[0],
    setEditIndex = _useState8[1];
  var _useState9 = useState(null),
    _useState10 = _slicedToArray(_useState9, 2),
    base64 = _useState10[0],
    setBase64 = _useState10[1];
  var _useState11 = useState([]),
    _useState12 = _slicedToArray(_useState11, 2),
    fileList = _useState12[0],
    setFileList = _useState12[1];
  var _useLocalStorageState = useLocalStorageState('hideOfficeLayer', {
      defaultValue: true
    }),
    _useLocalStorageState2 = _slicedToArray(_useLocalStorageState, 2),
    hideOfficeLayer = _useLocalStorageState2[0],
    setHideOfficeLayer = _useLocalStorageState2[1];
  var handleOk = function handleOk() {
    form.submit();
  };
  var BASE_LAYER_GROUP = [{
    id: OfficeLayerEnum.VectorMap,
    title: t('official_layer_control.index.shiLiangDiTu'),
    image: 'https://mdn.alipayobjects.com/huamei_qa8qxu/afts/img/A*qdFDSbvIalgAAAAAAAAAAAAADmJ7AQ/original',
    layers: []
  }
  // {
  //   id: OfficeLayerEnum.GoogleSatellite,
  //   title: t('official_layer_control.index.guGeWeiXingTu'),
  //   image:
  //     'https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*zi2jSqqZ2-8AAAAAAAAAAAAADjWqAQ/original',
  //   layers: [GOOGLE_TILE_MAP_URL, GOOGLE_TILE_MAP_ROUTER_URL],
  // },
  ];
  var officeLayerGroup = useMemo(function () {
    if (customTiles.length) {
      return [].concat(BASE_LAYER_GROUP, _toConsumableArray(customTiles));
    }
    return BASE_LAYER_GROUP;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [customTiles, t]);
  var handleBeforeUpload = function handleBeforeUpload(file) {
    var reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = function () {
      var base64Image = reader.result;
      setBase64(base64Image);
    };
    return false; // 阻止antd组件自动上传
  };
  var onItemClick = function onItemClick(item) {
    setRadioValue(item.id);
    setLayerType(item.id === OfficeLayerEnum.VectorMap ? [] : [item.id]);
  };
  var onFinish = function onFinish(e) {
    if (isEdit) {
      var cloneCustomTiles = cloneDeep(customTiles);
      var newImgUrl = Array.isArray(e.img) ? e.img[0].url : "".concat(base64);
      cloneCustomTiles[editIndex - 2] = {
        id: e.name,
        image: newImgUrl,
        title: e.name,
        layers: e.urls
      };
      setCustomTiles(cloneCustomTiles);
      setIsEdit(false);
      setEditIndex(-1);
    } else {
      setCustomTiles(function (prevState) {
        return [].concat(_toConsumableArray(prevState), [{
          id: e.name,
          image: "".concat(base64),
          title: e.name,
          layers: e.urls
        }]);
      });
    }
    setIsEdit(false);
    setFileList([]);
    setIsModalOpen(false);
  };
  var handleCancel = function handleCancel() {
    setIsModalOpen(false);
    setIsEdit(false);
    setFileList([]);
    setEditIndex(-1);
    form.resetFields();
  };
  var _onConfirm = function onConfirm(e, item) {
    e === null || e === void 0 || e.stopPropagation();
    var newCustomTiles = customTiles.filter(function (val) {
      return val.id !== item.id;
    });
    if (item.id === radioValue) {
      setRadioValue(OfficeLayerEnum.VectorMap);
      setLayerType([]);
    }
    setCustomTiles(newCustomTiles);
  };
  var validateSpace = function validateSpace(_, value) {
    var lowerCaseValue = value.toLowerCase();
    var listArr = [];
    if (isEdit) {
      var cloneOfficeLayerGroup = cloneDeep(officeLayerGroup);
      cloneOfficeLayerGroup.splice(editIndex, 1);
      listArr = _toConsumableArray(cloneOfficeLayerGroup);
    } else {
      listArr = officeLayerGroup;
    }
    var hasDuplicate = listArr.every(function (item) {
      return item.title.toLowerCase() !== lowerCaseValue;
    });
    if (!hasDuplicate) {
      return Promise.reject(t('official_layer_control.index.mingChengChongFu'));
    }
    if (value && value.trim() === '') {
      return Promise.reject(t('official_layer_control.index.kongGe'));
    }
    return Promise.resolve();
  };
  var uploadValidateSpace = function uploadValidateSpace(_, value) {
    var reject = Promise.reject(t('official_layer_control.index.shangchuan'));
    if (value.fileList) {
      if (!value.fileList.length) {
        return reject;
      } else {
        return Promise.resolve();
      }
    } else {
      if (!value.length) {
        return reject;
      } else {
        return Promise.resolve();
      }
    }
  };
  var handleChange = function handleChange(info) {
    var newFileList = _toConsumableArray(info.fileList);
    newFileList = newFileList.slice(-2);
    newFileList = newFileList.map(function (file) {
      if (file.response) {
        file.url = file.response.url;
      }
      return file;
    });
    setFileList(newFileList);
  };
  var rasterLayer = useMemo(function () {
    if (layerType.length) {
      var findItem = officeLayerGroup.find(function (item) {
        return item.id === layerType[0];
      });
      return findItem === null || findItem === void 0 ? void 0 : findItem.layers.map(function (item) {
        return (
          /*#__PURE__*/
          // eslint-disable-next-line react/jsx-key
          React.createElement(RasterLayer, {
            zIndex: 1,
            id: findItem.id === OfficeLayerEnum.GoogleSatellite && item === GOOGLE_TILE_MAP_URL ? 'googleTileMap' : undefined,
            source: {
              data: item,
              parser: {
                type: 'rasterTile',
                tileSize: 256,
                zoomOffset: 0
              }
            }
          })
        );
      });
    } else {
      return null;
    }
  }, [layerType, officeLayerGroup]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CustomControl, {
    position: "bottomleft"
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.mapTab,
    id: "l7-editor-aMap"
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.hideOfficeLayerBtn,
    onClick: function onClick() {
      setHideOfficeLayer(!hideOfficeLayer);
    }
  }, /*#__PURE__*/React.createElement(CaretRightOutlined, {
    style: {
      transform: hideOfficeLayer ? 'rotate(-180deg)' : undefined
    }
  })), hideOfficeLayer && /*#__PURE__*/React.createElement("div", {
    className: styles.amapInfo
  }, officeLayerGroup.map(function (item, index) {
    return /*#__PURE__*/React.createElement("div", {
      key: item.id,
      className: classNames([styles.amapInfoItem, item.id === radioValue ? styles.itemBorderActive : styles.itemBorder, index === officeLayerGroup.length - 1 ? 'item-hover' : '']),
      onClick: function onClick() {
        onItemClick(item);
      }
    }, index > 1 && /*#__PURE__*/React.createElement(Popconfirm, {
      title: t('official_layer_control.index.shanChuDiTu'),
      onConfirm: function onConfirm(e) {
        return _onConfirm(e, item);
      },
      onCancel: function onCancel(e) {
        e === null || e === void 0 || e.stopPropagation();
      }
    }, /*#__PURE__*/React.createElement("div", {
      className: 'item-clear',
      onClick: function onClick(e) {
        e.stopPropagation();
      }
    }, /*#__PURE__*/React.createElement(DeleteOutlined, null))), index > 1 && /*#__PURE__*/React.createElement("div", {
      className: 'item-edit',
      onClick: function onClick(e) {
        setEditIndex(index);
        e.stopPropagation();
        setIsEdit(true);
        setIsModalOpen(true);
        setFileList([{
          uid: '-1',
          name: "".concat(item.title),
          status: 'done',
          url: item.image
        }]);
        form.setFieldsValue({
          name: item.title,
          urls: item.layers,
          img: [{
            uid: '-1',
            name: "".concat(item.title),
            status: 'done',
            url: item.image
          }]
        });
      }
    }, /*#__PURE__*/React.createElement(FormOutlined, null)), /*#__PURE__*/React.createElement("img", {
      src: item.image,
      alt: "",
      className: styles.amapInfoItemImage
    }), /*#__PURE__*/React.createElement("div", {
      className: styles.amapInfoItemTitle,
      style: {
        marginTop: 0
      }
    }, item.title));
  }), /*#__PURE__*/React.createElement("div", {
    className: "add-map"
  }, /*#__PURE__*/React.createElement("div", {
    onClick: function onClick() {
      setIsModalOpen(true);
      form.resetFields();
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.addMapIcon
  }, /*#__PURE__*/React.createElement(PlusOutlined, null)))))), /*#__PURE__*/React.createElement(Modal, {
    title: t('official_layer_control.index.tianJiaDitu'),
    open: isModalOpen,
    destroyOnClose: true,
    onOk: handleOk,
    onCancel: handleCancel,
    width: 600
  }, /*#__PURE__*/React.createElement(Form, {
    form: form,
    initialValues: {
      urls: ['']
    },
    onFinish: onFinish
  }, /*#__PURE__*/React.createElement(Form.Item, _extends({}, locale === 'zh-CN' ? layout : enLayout, {
    name: "name",
    label: t('official_layer_control.index.name'),
    rules: [{
      required: true
    }, {
      validator: validateSpace
    }]
  }), /*#__PURE__*/React.createElement(Input, {
    placeholder: t('official_layer_control.index.addName'),
    style: {
      width: 390
    }
  })), /*#__PURE__*/React.createElement(Form.Item, _extends({}, locale === 'zh-CN' ? layout : enLayout, {
    name: "img",
    label: t('official_layer_control.index.shiLiTuPian'),
    rules: [{
      required: true,
      validator: uploadValidateSpace
    }]
  }), /*#__PURE__*/React.createElement(Upload, {
    beforeUpload: handleBeforeUpload,
    accept: ".png,.jpg",
    maxCount: 1,
    onChange: handleChange,
    fileList: fileList,
    onRemove: function onRemove() {
      setFileList([]);
    }
  }, /*#__PURE__*/React.createElement(Button, {
    icon: /*#__PURE__*/React.createElement(UploadOutlined, null)
  }, t('import_btn.index.shangChuan')))), /*#__PURE__*/React.createElement(Form.List, {
    name: "urls"
  }, function (fields, _ref) {
    var add = _ref.add,
      remove = _ref.remove;
    return /*#__PURE__*/React.createElement(React.Fragment, null, fields.map(function (field, index) {
      return /*#__PURE__*/React.createElement(Space, {
        key: field.key,
        style: {
          display: 'flex',
          marginBottom: 8
        },
        align: "baseline"
      }, /*#__PURE__*/React.createElement(Form.Item, _extends({}, field, {
        label: index === 0 ? t('official_layer_control.index.tuCengDiZhi') : '',
        rules: [{
          required: true,
          message: t('official_layer_control.index.qiShuRutuCengDiZhi')
        }, {
          validator: validateSpace
        }],
        style: {
          marginLeft: locale === 'zh-CN' ? index === 0 ? 10 : 90 : index === 0 ? 18 : 134
        }
      }), /*#__PURE__*/React.createElement(Input, {
        placeholder: GOOGLE_TILE_MAP_URL,
        style: {
          width: 390
        }
      })), fields.length > 1 ? /*#__PURE__*/React.createElement(MinusCircleOutlined, {
        onClick: function onClick() {
          return remove(field.name);
        }
      }) : null);
    }), /*#__PURE__*/React.createElement(Form.Item, {
      style: {
        textAlign: 'center'
      }
    }, /*#__PURE__*/React.createElement(Button, {
      type: "dashed",
      onClick: function onClick() {
        return add();
      },
      icon: /*#__PURE__*/React.createElement(PlusOutlined, null),
      style: {
        width: 390,
        marginLeft: locale === 'zh-CN' ? 20 : 104
      }
    }, t('official_layer_control.index.tinJiaWaPian'))));
  })))), /*#__PURE__*/React.createElement("div", null, rasterLayer));
}
