// @antv/l7-editor/es/components/map-control-group/administrative-select-control/index.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { CustomControl, LineLayer } from '@antv/larkmap';
import { featureCollection, multiLineString } from '@turf/turf';
import { Cascader, Dropdown, Empty, Tooltip, message } from 'antd';
import classNames from 'classnames';
import { cloneDeep } from 'lodash-es';
import React, { useEffect, useMemo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { LayerZIndex } from "../../../constants";
import { useFeature, useGlobal } from "../../../recoil";
import { IconFont } from "../../iconfont";
import { useStyle } from "./styles";
var DistrictLayerOptions = {
  shape: 'line',
  color: '#ff0000',
  size: 2,
  zIndex: LayerZIndex,
  style: {
    opacity: 0.8
  }
};
export var AdministrativeSelect = function AdministrativeSelect() {
  var styles = useStyle();
  var _useGlobal = useGlobal(),
    cityHistory = _useGlobal.cityHistory,
    setCityHistory = _useGlobal.setCityHistory;
  var _useFeature = useFeature(),
    scene = _useFeature.scene;
  var _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    districtFeature = _useState2[0],
    setDistrictFeature = _useState2[1];
  var _useState3 = useState(),
    _useState4 = _slicedToArray(_useState3, 2),
    data = _useState4[0],
    setData = _useState4[1];
  var _useState5 = useState([]),
    _useState6 = _slicedToArray(_useState5, 2),
    value = _useState6[0],
    setValue = _useState6[1];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var getCascadeData = function getCascadeData(list) {
    list.sort(function (a, b) {
      return +a.adcode - +b.adcode;
    });
    if (list.length) {
      return list.map(function (item) {
        var name = item.name,
          districts = item.districts,
          adcode = item.adcode;
        return {
          adcode: adcode,
          value: adcode,
          label: name,
          children: getCascadeData(districts)
        };
      });
    } else {
      return [];
    }
  };
  useEffect(function () {
    fetch('https://restapi.amap.com/v3/config/district?key=98d10f05a2da96697313a2ce35ebf1a2&keywords=中华人民共和国&subdistrict=3&extensions=base').then(function (res) {
      return res.json();
    }).then(function (res) {
      setData(getCascadeData(res.districts[0].districts));
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  var onChange = function onChange(items, option) {
    setValue(items);
    if (option) {
      var code = [];
      var label = [];
      option.forEach(function (item) {
        code.push(item.adcode);
        label.push(item.label);
      });
      var item = {
        value: JSON.stringify(code),
        label: label.join('/')
      };
      var arr = [item].concat(_toConsumableArray(cityHistory));
      var formatArr = function formatArr() {
        var map = new Map();
        var _iterator = _createForOfIteratorHelper(arr),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var o = _step.value;
            if (!map.has(o.value)) {
              map.set(o.value, o);
            }
          }
          //@ts-ignore
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        return _toConsumableArray(map.values());
      };
      setCityHistory(formatArr());
      if (cityHistory.length >= 10) {
        var arrHistory = cloneDeep(cityHistory);
        arrHistory.pop();
        setCityHistory(arrHistory);
      }
    } else {
      setDistrictFeature(null);
    }
  };
  var filter = function filter(inputValue, path) {
    return path.some(function (option) {
      return option.label.toLowerCase().indexOf(inputValue.toLowerCase()) > -1;
    });
  };
  var historyItem = useMemo(function () {
    if (cityHistory.length) {
      var item = cityHistory.map(function (city) {
        return {
          key: city.value,
          label: /*#__PURE__*/React.createElement("div", {
            onClick: function onClick() {
              setValue(JSON.parse(city.value));
            }
          }, city.label)
        };
      });
      return item;
    } else {
      return [{
        key: 'undefined',
        label: /*#__PURE__*/React.createElement(Empty, null)
      }];
    }
  }, [cityHistory]);
  useEffect(function () {
    if (value) {
      var name = value[value.length - 1];
      fetch("https://restapi.amap.com/v3/config/district?keywords=".concat(name, "&subdistrict=0&key=98d10f05a2da96697313a2ce35ebf1a2&extensions=all")).then(function (res) {
        return res.json();
      }).then(function (res) {
        var _res$districts;
        if (res.status === '1' && (_res$districts = res.districts) !== null && _res$districts !== void 0 && _res$districts.length && scene) {
          var _split$map = res.districts[0].center.split(',').map(function (item) {
              return +item;
            }),
            _split$map2 = _slicedToArray(_split$map, 2),
            lng = _split$map2[0],
            lat = _split$map2[1];
          scene.setZoomAndCenter(9, [lng, lat]);
          var positions = [];
          res.districts.forEach(function (district) {
            district.polyline.split('|').forEach(function (chunk) {
              positions.push(chunk.split(';').map(function (item) {
                return item.split(',').map(function (num) {
                  return +num;
                });
              }));
            });
          });
          setDistrictFeature(multiLineString(positions));
        }
      }).catch(function () {
        message.error(t('administrative_select_control.index.weiLanShuJuQing'));
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [value]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CustomControl, {
    position: "lefttop"
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex'
    }
  }, /*#__PURE__*/React.createElement("div", {
    id: "l7-editor-administrativeSelect"
  }, /*#__PURE__*/React.createElement(Cascader, {
    options: data,
    value: value
    // @ts-ignore
    ,
    onChange: onChange,
    className: styles.cascader,
    allowClear: true,
    showSearch: {
      filter: filter
    },
    placeholder: t('administrative_select_control.index.keXuanZeShengShi'),
    changeOnSelect: true,
    style: {
      width: 250
    },
    popupClassName: styles.cascaderPopup,
    expandTrigger: "hover"
  })), /*#__PURE__*/React.createElement(Dropdown, {
    menu: {
      items: historyItem
    },
    placement: "bottomRight",
    trigger: ['click']
  }, /*#__PURE__*/React.createElement(Tooltip, {
    title: t('administrative_select_control.index.xingZhengQuHuaLi'),
    placement: "right"
  }, /*#__PURE__*/React.createElement("div", {
    className: classNames([styles.history, 'l7-draw-control__btn'])
  }, /*#__PURE__*/React.createElement(IconFont, {
    type: "icon-lishi",
    className: styles.historyIcon
  })))))), /*#__PURE__*/React.createElement(LineLayer, _extends({
    source: {
      data: featureCollection(districtFeature ? [districtFeature] : [])
    }
  }, DistrictLayerOptions)));
};
