// @antv/l7-editor/es/components/map-control-group/clear-control/index.js
import { CustomControl } from '@antv/larkmap';
import { Popconfirm, Tooltip } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useFeature } from "../../../recoil";
import { prettierText } from "../../../utils/prettier-text";
import { IconFont } from "../../iconfont";
import useStyle from "./styles";
export var ClearControl = function ClearControl() {
  var styles = useStyle();
  var _useFeature = useFeature(),
    saveEditorText = _useFeature.saveEditorText;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(CustomControl, {
    position: "topright"
  }, /*#__PURE__*/React.createElement(Popconfirm, {
    title: t('clear_control.index.queRenQingKongSuo'),
    onConfirm: function onConfirm() {
      saveEditorText(prettierText({
        content: {
          type: 'FeatureCollection',
          features: []
        }
      }));
    }
  }, /*#__PURE__*/React.createElement(Tooltip, {
    title: t('app_header.constants.qingChu'),
    placement: "bottom"
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.clear,
    id: "l7-editor-clear"
  }, /*#__PURE__*/React.createElement(IconFont, {
    type: "icon-qingkong",
    className: styles.clearSvg
  }))))));
};
