// @antv/l7-editor/lib/components/map-control-group/layer-color-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/layer-color-control/index.tsx
var layer_color_control_exports = {};
__export(layer_color_control_exports, {
  default: () => layer_color_control_default
});
module.exports = __toCommonJS(layer_color_control_exports);
var import_larkmap = require("@antv/larkmap");
var import_antd = require("antd");
var import_classnames = __toESM(require("classnames"));
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../../recoil");
var import_styles = __toESM(require("./styles"));
var LayerColorControl = () => {
  const { layerColor, setLayerColor } = (0, import_recoil.useGlobal)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const styles = (0, import_styles.default)();
  return /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "bottomright" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Tooltip,
    {
      title: t("layer_color_control.index.tuCengYanSeXuan"),
      placement: "left"
    },
    /* @__PURE__ */ import_react.default.createElement("div", { id: "l7-editor-color" }, /* @__PURE__ */ import_react.default.createElement(
      import_antd.ColorPicker,
      {
        value: layerColor,
        className: (0, import_classnames.default)([styles.l7ButtonControl]),
        onChange: (value, hex) => {
          setLayerColor(hex);
        },
        format: "rgb",
        presets: [
          {
            label: t("layer_color_control.index.changYongYanSe"),
            colors: [
              "#F5222D",
              "#FA8C16",
              "#FADB14",
              "#8BBB11",
              "#52C41A",
              "#13A8A8",
              "#1677FF",
              "#2F54EB",
              "#722ED1",
              "#EB2F96"
            ]
          }
        ]
      }
    ))
  ));
};
var layer_color_control_default = LayerColorControl;
