// @antv/l7-editor/lib/components/app-header/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/index.tsx
var app_header_exports = {};
__export(app_header_exports, {
  AppHeader: () => AppHeader
});
module.exports = __toCommonJS(app_header_exports);
var import_icons = require("@ant-design/icons");
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_classnames = __toESM(require("classnames"));
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_constants = require("../../constants");
var import_recoil = require("../../recoil");
var import_iconfont = require("../iconfont");
var import_base_map = require("./base-map");
var import_download_btn = __toESM(require("./btn/download-btn"));
var import_handback_btn = __toESM(require("./btn/handback-btn"));
var import_i18n_btn = __toESM(require("./btn/i18n-btn"));
var import_import_btn = require("./btn/import-btn");
var import_setting_btn = require("./btn/setting-btn");
var import_styles = __toESM(require("./styles"));
var isTooBar = {
  logo: true,
  import: true,
  download: true,
  guide: true,
  help: true,
  setting: true,
  theme: true,
  dingTalk: true,
  i18n: true,
  github: true,
  baseMap: true
};
var AppHeader = ({ toolbar }) => {
  const { t } = (0, import_react_i18next.useTranslation)();
  const [open, setOpen] = (0, import_react.useState)({ key: "", open: false });
  const { autoFitBounds, theme, setTheme } = (0, import_recoil.useGlobal)();
  const { saveEditorText, savable, bboxAutoFit } = (0, import_recoil.useFeature)();
  const [isTooBarState, setIsTooBar] = (0, import_react.useState)(isTooBar);
  const styles = (0, import_styles.default)();
  const [firstOpen, setFirstOpen] = (0, import_ahooks.useLocalStorageState)(
    import_constants.LocalStorageKey.firstOpening,
    {
      defaultValue: true
    }
  );
  (0, import_react.useEffect)(() => {
    if (firstOpen) {
      setOpen({
        key: "basics",
        open: true
      });
      setFirstOpen(false);
    }
  }, []);
  const DropdownMenuItems = [
    {
      key: "basics",
      label: t("app_header.constants.pingTaiJiChuShi")
    },
    {
      key: "function",
      label: t("app_header.constants.quanGongNengShiYong")
    }
  ];
  const steps = [
    {
      title: t("import_btn.index.shangChuan"),
      description: t("app_header.constants.shangChuanGEO"),
      cover: /* @__PURE__ */ import_react.default.createElement(
        "img",
        {
          alt: t("app_header.constants.shangChuanPNG"),
          src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*RJfORKi3ntsAAAAAAAAAAAAADjWqAQ/original"
        }
      ),
      target: () => document.getElementById("l7-editor-upload")
    },
    {
      title: t("app_header.constants.bianJiQi"),
      description: t("app_header.constants.keYiTongGuoBian"),
      target: () => document.getElementById("l7-editor-panel"),
      placement: "left"
    },
    {
      title: t("app_header.constants.huiZhi"),
      cover: /* @__PURE__ */ import_react.default.createElement(
        "img",
        {
          alt: t("app_header.constants.huiZhiPNG"),
          src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*NNcDQrzkdt8AAAAAAAAAAAAADjWqAQ/original"
        }
      ),
      description: t("app_header.constants.keYiJiHuoDi"),
      target: () => document.getElementById("l7-editor-draw"),
      placement: "right"
    },
    {
      title: t("app_header.constants.baoCun"),
      description: t("app_header.constants.shiYongCTR"),
      target: () => document.getElementById("l7-editor-save")
    },
    {
      title: t("app_header.constants.xuanRan"),
      description: t("app_header.constants.diTuShangChaKan"),
      target: () => document.getElementById("l7-editor-map"),
      placement: "right"
    }
  ];
  const functionSteps = [
    {
      title: t("import_btn.index.shangChuan"),
      description: t("app_header.constants.shangChuanGEO"),
      cover: /* @__PURE__ */ import_react.default.createElement(
        "img",
        {
          alt: t("app_header.constants.shangChuanPNG"),
          src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*RJfORKi3ntsAAAAAAAAAAAAADjWqAQ/original"
        }
      ),
      target: () => document.getElementById("l7-editor-upload")
    },
    {
      title: t("app_header.constants.baoCun"),
      description: t("app_header.constants.shiYongCTR"),
      target: () => document.getElementById("l7-editor-save")
    },
    {
      title: t("btn.download_btn.xiaZai"),
      description: t("app_header.constants.dianJiXiaZaiKe"),
      target: () => document.getElementById("l7-editor-download")
    },
    {
      title: t("btn.setting_btn.sheZhi"),
      description: t("app_header.constants.yongHuKeSheZhi"),
      target: () => document.getElementById("l7-editor-set")
    },
    {
      title: t("app_header.constants.zhuTi"),
      description: t("app_header.constants.dianJiKeQieHuan"),
      target: () => document.getElementById("l7-editor-theme")
    },
    {
      title: t("app_header.constants.xingZhengQuXuanZe"),
      description: t("app_header.constants.keYiKuaiSuXuan"),
      target: () => document.getElementById("l7-editor-administrativeSelect")
    },
    {
      title: t("app_header.constants.chengShiChaXun"),
      description: t("app_header.constants.souSuoChengShiHou"),
      target: () => document.getElementById("l7-editor-citySelect")
    },
    {
      title: t("app_header.constants.huiZhi"),
      cover: /* @__PURE__ */ import_react.default.createElement(
        "img",
        {
          alt: t("app_header.constants.huiZhiPNG"),
          src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*NNcDQrzkdt8AAAAAAAAAAAAADjWqAQ/original"
        }
      ),
      description: t("app_header.constants.keYiJiHuoDi"),
      target: () => document.getElementById("l7-editor-draw"),
      placement: "right"
    },
    {
      title: t("app_header.constants.qingChu"),
      description: t("app_header.constants.dianJiQingChuAn"),
      target: () => document.getElementById("l7-editor-clear")
    },
    {
      title: t("app_header.constants.shaiXuan"),
      description: t("app_header.constants.keGenJuShuJu"),
      target: () => document.getElementById("l7-editor-filter")
    },
    {
      title: t("app_header.constants.ziDongSuoFang"),
      description: t("app_header.constants.dianJiZiDongSuo"),
      target: () => document.getElementById("l7-editor-auto")
    },
    {
      title: t("app_header.constants.yanSeXuanZeQi"),
      description: t("app_header.constants.xuanZeYanSeLai"),
      target: () => document.getElementById("l7-editor-color")
    },
    {
      title: t("app_header.constants.zhuTiSe"),
      description: t("app_header.constants.xuanZeBuTongZhu"),
      //@ts-ignore
      target: () => document.getElementsByClassName("l7-editor-mapTheme")[0]
    },
    {
      title: t("app_header.constants.guanFangTuCeng"),
      description: t("app_header.constants.keXuanZeBuTong"),
      target: () => document.getElementById("l7-editor-aMap")
    },
    {
      title: t("text-layer-control_wenBenTuCengPeiZhi"),
      description: t("text-layer-control_description"),
      target: () => document.getElementById("text-layer-control")
    },
    {
      title: t("app_header.constants.gEOJS"),
      description: t("app_header.constants.keYiTongGuoBian"),
      target: () => document.getElementById("l7-editor-panel"),
      placement: "left"
    },
    {
      title: t("app_header.constants.wKTBianJi"),
      description: t("app_header.constants.keYiTongGuoW"),
      target: () => document.getElementById("l7-editor-wkt"),
      placement: "left"
    },
    {
      title: t("app_header.constants.biaoGe"),
      description: t("app_header.constants.keYiTongGuoBiao"),
      target: () => document.getElementById("l7-editor-table"),
      placement: "left"
    }
  ];
  const onSave = () => {
    if (!savable) {
      return;
    }
    const features = saveEditorText();
    if (autoFitBounds) {
      bboxAutoFit(features);
    }
  };
  (0, import_ahooks.useKeyPress)(["ctrl.s", "meta.s"], (e) => {
    e.preventDefault();
    onSave();
  });
  const onDownload = (key) => {
    setOpen({ key, open: true });
  };
  (0, import_react.useEffect)(() => {
    setIsTooBar({ ...isTooBar, ...toolbar });
  }, [toolbar]);
  return /* @__PURE__ */ import_react.default.createElement("div", { className: (0, import_classnames.default)([styles.mapHeader, "l7-editor-header"]) }, /* @__PURE__ */ import_react.default.createElement("div", { className: styles.mapHeaderLeft }, isTooBarState.logo && /* @__PURE__ */ import_react.default.createElement(
    "div",
    {
      className: (0, import_classnames.default)([
        styles.mapHeaderLogo,
        "l7-editor-header__logo"
      ])
    },
    /* @__PURE__ */ import_react.default.createElement("img", { src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*RSdESJd70P8AAAAAAAAAAAAADjWqAQ/original" }),
    /* @__PURE__ */ import_react.default.createElement("span", { className: styles.mapHeaderTitle }, "L7 Editor")
  ), isTooBarState.logo && isTooBarState.baseMap && /* @__PURE__ */ import_react.default.createElement(import_antd.Divider, { type: "vertical" }), isTooBarState.baseMap && /* @__PURE__ */ import_react.default.createElement(import_base_map.BaseMap, null)), /* @__PURE__ */ import_react.default.createElement("div", { className: styles.mapHeaderRight }, /* @__PURE__ */ import_react.default.createElement(import_antd.Space, null, isTooBarState.import && /* @__PURE__ */ import_react.default.createElement(import_import_btn.ImportBtn, null), /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: t("app_header.index.baoCunShuJu"), placement: "bottom" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Button,
    {
      id: "l7-editor-save",
      icon: /* @__PURE__ */ import_react.default.createElement(import_icons.SaveOutlined, null),
      disabled: !savable,
      onClick: onSave
    },
    t("app_header.constants.baoCun")
  )), isTooBarState.download && /* @__PURE__ */ import_react.default.createElement(import_download_btn.default, null), isTooBarState.guide && /* @__PURE__ */ import_react.default.createElement(
    import_antd.Dropdown,
    {
      menu: {
        items: DropdownMenuItems,
        onClick: ({ key }) => {
          onDownload(key);
        }
      },
      placement: "bottom"
    },
    /* @__PURE__ */ import_react.default.createElement(import_antd.Button, { icon: /* @__PURE__ */ import_react.default.createElement(import_icons.FlagOutlined, null) }, t("app_header.index.yinDao"))
  ), isTooBarState.help && /* @__PURE__ */ import_react.default.createElement(import_handback_btn.default, null), isTooBarState.setting && /* @__PURE__ */ import_react.default.createElement(import_setting_btn.SettingBtn, null)), /* @__PURE__ */ import_react.default.createElement(import_antd.Divider, { type: "vertical" }), /* @__PURE__ */ import_react.default.createElement(import_antd.Space, null, isTooBarState.dingTalk && /* @__PURE__ */ import_react.default.createElement(
    import_antd.Popover,
    {
      placement: "bottom",
      title: t("app_header.index.dingTalk"),
      content: /* @__PURE__ */ import_react.default.createElement(
        "img",
        {
          style: { width: 300 },
          src: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*CTHXQbdMcDcAAAAAAAAAAAAADjWqAQ/original"
        }
      ),
      trigger: "hover"
    },
    /* @__PURE__ */ import_react.default.createElement(import_antd.Button, { icon: /* @__PURE__ */ import_react.default.createElement(import_icons.DingtalkOutlined, null) })
  ), isTooBarState.theme && /* @__PURE__ */ import_react.default.createElement(
    import_antd.Tooltip,
    {
      title: t("app_header.index.zhutiqiehuan"),
      trigger: "hover",
      placement: "bottom"
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Button,
      {
        id: "l7-editor-theme",
        icon: /* @__PURE__ */ import_react.default.createElement(
          import_iconfont.IconFont,
          {
            className: styles.themeIcon,
            type: theme === "dark" ? "icon-a-qingtianwanshang" : "icon-taiyang"
          }
        ),
        onClick: () => {
          setTheme(theme === "dark" ? "light" : "dark");
        }
      }
    )
  ), isTooBarState.i18n && /* @__PURE__ */ import_react.default.createElement(import_i18n_btn.default, null), isTooBarState.github && /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: "Github", trigger: "hover", placement: "bottom" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Button,
    {
      icon: /* @__PURE__ */ import_react.default.createElement(import_icons.GithubOutlined, null),
      onClick: () => {
        window.open("https://github.com/antvis/L7Editor");
      }
    }
  )))), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Tour,
    {
      open: open.open,
      onClose: () => setOpen({ key: "", open: false }),
      steps: open.key === "basics" ? steps : functionSteps
    }
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AppHeader
});
