// @antv/l7-editor/lib/components/app-header/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const { colorBorder, colorBgContainer, colorText } = token;
  return {
    mapHeader: import_css.css`
      padding: 0 16px;
      height: 50px;
      border-bottom: 1px solid ${colorBorder};
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      align-items: center;
      background: ${colorBgContainer};
      z-index: 1;
      .ant-btn {
        border: 0;
      }
    `,
    mapHeaderLogo: import_css.css`
      display: flex;
      align-items: center;
      img {
        width: auto;
        height: 24px;
        margin-right: 4px;
      }
    `,
    mapHeaderTitle: import_css.css`
      font-size: 16px;
      font-weight: bold;
      margin-left: 4px;
      margin-right: 16px;
      color: ${colorText};
    `,
    mapHeaderLeft: import_css.css`
      display: flex;
      align-items: center;
    `,
    mapHeaderRight: import_css.css`
      display: flex;
      align-items: center;
    `,
    locale: import_css.css`
      font-size: 16px;
    `,
    localeIcon: import_css.css`
      font-size: 16px;
    `,
    upload: import_css.css`
      height: 200px;
      display: flex;
      overflow: hidden;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    `,
    uploadTitle: import_css.css`
      font-size: 14px;
      padding: 10px 6px 10px 0;
    `,
    theme: import_css.css`
      cursor: pointer;
      padding: 0 !important;
      width: 32px;
      height: 32px;
      display: flex;
      align-items: center;
      justify-content: center;
    `,
    themeIcon: import_css.css`
      font-size: 20px !important;
    `
  };
};
var styles_default = useStyle;
