// @antv/l7-editor/lib/components/resize-panel/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/resize-panel/index.tsx
var resize_panel_exports = {};
__export(resize_panel_exports, {
  ResizePanel: () => ResizePanel
});
module.exports = __toCommonJS(resize_panel_exports);
var import_icons = require("@ant-design/icons");
var import_ahooks = require("ahooks");
var import_classnames = __toESM(require("classnames"));
var import_re_resizable = require("re-resizable");
var import_react = __toESM(require("react"));
var import_constants = require("../../constants");
var import_recoil = require("../../recoil");
var import_styles = __toESM(require("./styles"));
var ResizePanel = ({
  left,
  right,
  onFeatureChange
}) => {
  const { hideEditor, setHideEditor, rightPanelWidth, setRightWidth } = (0, import_recoil.useGlobal)();
  const [resizePanel, setResizePanel] = (0, import_react.useState)(null);
  const [minRightWidth, maxRightWidth] = import_constants.RightPanelWidthRange;
  const { width: containerWidth = 0 } = (0, import_ahooks.useSize)(resizePanel) ?? {};
  const { features } = (0, import_recoil.useFeature)();
  const styles = (0, import_styles.default)();
  const onResize = (event) => {
    const { left: leftWidth = 0 } = (resizePanel == null ? void 0 : resizePanel.getBoundingClientRect()) ?? {};
    let newRightPanelWidth = 100 * (1 - (event.clientX - leftWidth) / containerWidth);
    if (newRightPanelWidth < minRightWidth) {
      newRightPanelWidth = minRightWidth;
    }
    if (newRightPanelWidth > maxRightWidth) {
      newRightPanelWidth = maxRightWidth;
    }
    setRightWidth(newRightPanelWidth);
  };
  const calcRightWidth = (0, import_react.useMemo)(() => {
    return hideEditor ? 0 : rightPanelWidth;
  }, [hideEditor, rightPanelWidth]);
  (0, import_react.useEffect)(() => {
    onFeatureChange(features);
  }, [features]);
  return /* @__PURE__ */ import_react.default.createElement(
    "div",
    {
      className: (0, import_classnames.default)([styles.resizePanel, "l7-editor-content"]),
      ref: (ref) => {
        setResizePanel(ref);
      }
    },
    /* @__PURE__ */ import_react.default.createElement(
      "div",
      {
        className: (0, import_classnames.default)([
          styles.resizePanelLeft,
          "l7-editor-content__left"
        ]),
        style: { width: `${100 - calcRightWidth}%` }
      },
      left
    ),
    /* @__PURE__ */ import_react.default.createElement(
      import_re_resizable.Resizable,
      {
        enable: {
          left: true
        },
        style: {
          marginLeft: `${100 - calcRightWidth}%`
        },
        size: {
          width: `${calcRightWidth}%`,
          height: "100%"
        },
        minWidth: `${minRightWidth / 100 * containerWidth}px`,
        maxWidth: `${maxRightWidth / 100 * containerWidth}px`,
        className: (0, import_classnames.default)([
          styles.resizePanelRight,
          "l7-editor-content__right"
        ]),
        handleClasses: {
          left: styles.resizePanelDragLine
        },
        handleStyles: {
          left: { width: 6, left: -3 }
        },
        defaultSize: {
          width: `${calcRightWidth}%`,
          height: "100%"
        },
        onResize
      },
      right,
      /* @__PURE__ */ import_react.default.createElement(
        "div",
        {
          className: styles.resizePanelToggleBtn,
          onClick: () => {
            setHideEditor(!hideEditor);
          }
        },
        /* @__PURE__ */ import_react.default.createElement(
          import_icons.CaretRightOutlined,
          {
            style: {
              transform: hideEditor ? "rotate(-180deg)" : void 0
            }
          }
        )
      )
    )
  );
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ResizePanel
});
