// @antv/l7-editor/lib/components/resize-panel/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/resize-panel/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const {
    colorBgContainer,
    boxShadow,
    colorBorderSecondary,
    colorInfoHover,
    colorIcon
  } = token;
  return {
    resizePanel: import_css.css`
      width: 100%;
      height: calc(100% - 50px);
      display: flex;
      position: relative;
    `,
    resizePanelLeft: import_css.css`
      position: absolute;
      left: 0;
      top: 0;
      bottom: 0;
      overflow: hidden;
    `,
    resizePanelRight: import_css.css`
      position: relative;
      border-left: 1px solid ${colorBorderSecondary};
      box-shadow: ${boxShadow};
    `,
    resizePanelDragLine: import_css.css`
      transition: all 0.5s;
      cursor: move !important;
      z-index: 100;

      &:hover,
      &:focus {
        background-color: ${colorInfoHover};
      }
    `,
    resizePanelToggleBtn: import_css.css`
      height: 60px;
      width: 20px;
      display: flex;
      align-items: center;
      justify-content: center;
      background: ${colorBgContainer};
      position: absolute;
      top: calc(50% - 30px);
      left: -20px;
      border-top-left-radius: 4px;
      border-bottom-left-radius: 4px;
      cursor: pointer;
      z-index: 2;
      .anticon {
        transition: all 0.2s;
      }
      color: ${colorIcon};
    `
  };
};
var styles_default = useStyle;
