// @antv/l7-editor/es/components/text-layer/index.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { TextLayer } from '@antv/larkmap';
import { center } from '@turf/turf';
import { cloneDeep } from 'lodash-es';
import React, { useMemo } from 'react';
import { FeatureKey } from "../../constants";
import { useFilterFeatures } from "../../hooks";
import { useFeature, useGlobal } from "../../recoil";
import { centerOfLine } from "../../utils";
export var EditorTextLayer = function EditorTextLayer() {
  var _useFeature = useFeature(),
    transformCoord = _useFeature.transformCoord;
  var _useFilterFeatures = useFilterFeatures(),
    newFeatures = _useFilterFeatures.features;
  var _useGlobal = useGlobal(),
    layerColor = _useGlobal.layerColor,
    textLayerFields = _useGlobal.textLayerFields;
  var textOptions = useMemo(function () {
    return {
      zIndex: 101,
      field: 'text',
      style: {
        fill: "".concat(layerColor),
        opacity: 1,
        fontSize: 16,
        stroke: '#fff',
        strokeWidth: 2,
        textAllowOverlap: true,
        padding: [10, 10]
      }
    };
  }, [layerColor]);
  var pointTextOptions = useMemo(function () {
    var newLayerOptions = cloneDeep(textOptions);
    newLayerOptions.style.textOffset = [0, -20];
    return newLayerOptions;
  }, [textOptions]);
  var sourceData = useMemo(function () {
    var data = transformCoord(newFeatures).map(function (item) {
      var _item$properties;
      // @ts-ignore
      var featureIndex = (_item$properties = item.properties) === null || _item$properties === void 0 ? void 0 : _item$properties[FeatureKey.Index];
      var _ref = function () {
          if (item.geometry.type === 'LineString') {
            return centerOfLine(item).geometry.coordinates;
          }
          if (item.geometry.type === 'Point') {
            return item.geometry.coordinates;
          } else {
            return center(item).geometry.coordinates;
          }
        }(),
        _ref2 = _slicedToArray(_ref, 2),
        x = _ref2[0],
        y = _ref2[1];
      var text = "".concat(featureIndex + 1);
      if (textLayerFields !== null && textLayerFields !== void 0 && textLayerFields.length) {
        text = textLayerFields.map(function (field) {
          var _item$properties$fiel, _item$properties2;
          return String((_item$properties$fiel = (_item$properties2 = item.properties) === null || _item$properties2 === void 0 ? void 0 : _item$properties2[field]) !== null && _item$properties$fiel !== void 0 ? _item$properties$fiel : '');
        }).join(' ');
      }
      return {
        x: x,
        y: y,
        text: text,
        type: item.geometry.type
      };
    });
    return data;
  }, [transformCoord, newFeatures, textLayerFields]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(TextLayer, _extends({}, textOptions, {
    source: {
      data: sourceData.filter(function (item) {
        return item.type !== 'Point';
      }),
      parser: {
        type: 'json',
        x: 'x',
        y: 'y'
      }
    }
  })), /*#__PURE__*/React.createElement(TextLayer, _extends({}, pointTextOptions, {
    source: {
      data: sourceData.filter(function (item) {
        return item.type === 'Point';
      }),
      parser: {
        type: 'json',
        x: 'x',
        y: 'y'
      }
    }
  })));
};
