// @antv/l7-editor/lib/components/map-control-group/map-theme-control/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/map-theme-control/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const {
    colorTextDescription,
    colorBgContainer,
    colorPrimary,
    colorBorder,
    colorText,
    colorBgElevated,
    colorFillSecondary
  } = token;
  return {
    mapTheme: import_css.css`
      color: ${colorTextDescription};
      background: ${colorBgContainer} !important;
      border-color: ${colorBorder};

      svg {
        fill: ${colorTextDescription} !important;
      }

      &:hover {
        svg {
          fill: ${colorText} !important;
        }
      }
    `,
    mapThemePopper: import_css.css`
      *::-webkit-scrollbar {
        width: 8px;
        background-color: ${colorBgElevated};
      }
      *::-webkit-scrollbar-thumb {
        background-color: ${colorFillSecondary};
        border-radius: 10px;
      }
      .l7-popper-content {
        color: ${colorText};
        background: ${colorBgContainer};

        .l7-select-control-item:hover {
          box-shadow: 0 5px 12px 4px rgba(0, 0, 0, 0.09),
            0 3px 6px 0 rgba(0, 0, 0, 0.12), 0 1px 2px -2px rgba(0, 0, 0, 0.16);
        }

        .l7-select-control-item {
          border: 1px solid ${colorBorder};
          border-radius: 4px;
        }

        .l7-select-control-item-active {
          border: 1px solid ${colorPrimary};
        }
      }

      &.l7-popper-right .l7-popper-arrow  {
        border-right-color: ${colorBgElevated};
      }

      &.l7-popper-left .l7-popper-arrow  {
        border-left-color: ${colorBgElevated};
      }

      &.l7-popper-bottom .l7-popper-arrow  {
        border-bottom-color: ${colorBgElevated};
      }

      &.l7-popper-top .l7-popper-arrow  {
        border-top-color: ${colorBgElevated};
      }
    `
  };
};
var styles_default = useStyle;
