// @antv/l7-editor/es/components/wkt-editor/index.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { useDebounceFn } from 'ahooks';
import { Input } from 'antd';
import React, { forwardRef, useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useFeature, useGlobal } from "../../recoil";
import { GeoJSON2Wkt, Wkt2GeoJSON } from "../../utils";
var TextArea = Input.TextArea;
export var WktEditor = /*#__PURE__*/forwardRef(function () {
  var _useState = useState(''),
    _useState2 = _slicedToArray(_useState, 2),
    input = _useState2[0],
    setInput = _useState2[1];
  var _useFeature = useFeature(),
    fc = _useFeature.fc,
    resetFeatures = _useFeature.resetFeatures,
    bboxAutoFit = _useFeature.bboxAutoFit;
  var _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    isFocus = _useState4[0],
    setIsFocus = _useState4[1];
  var _useGlobal = useGlobal(),
    autoFitBounds = _useGlobal.autoFitBounds;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  useEffect(function () {
    if (isFocus) {
      return;
    }
    //@ts-ignore
    var result = GeoJSON2Wkt(fc);
    if (result !== input) {
      setInput(result);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [fc, isFocus]);
  var _useDebounceFn = useDebounceFn(function (wkt) {
      //@ts-ignore
      var newWkt = wkt.replaceAll('(', ' (');
      var _Wkt2GeoJSON = Wkt2GeoJSON(newWkt),
        features = _Wkt2GeoJSON.features;
      resetFeatures(features);
      if (autoFitBounds) {
        bboxAutoFit(features);
      }
    }, {
      wait: 1000,
      maxWait: 1000
    }),
    onInputChange = _useDebounceFn.run;
  return /*#__PURE__*/React.createElement(TextArea, {
    value: input,
    style: {
      margin: 8,
      height: 'calc(100% - 16px)',
      width: 'calc(100% - 16px)'
    },
    placeholder: t('wkt_editor.index.shuRuWKT'),
    onFocus: function onFocus() {
      return setIsFocus(true);
    },
    onBlur: function onBlur() {
      return setIsFocus(false);
    },
    onChange: function onChange(e) {
      setInput(e.target.value);
      onInputChange(e.target.value);
    }
  });
});
