// @antv/l7-editor/es/hooks/use-filter-features.js
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { isEmpty, isUndefined } from 'lodash-es';
import { useEffect } from 'react';
import { atom, useRecoilValue, useSetRecoilState } from 'recoil';
import { useFeature, useFilter } from "../recoil";
export function isEmptyFilter(filter) {
  var operator = filter.operator,
    value = filter.value,
    type = filter.type,
    field = filter.field;
  return isUndefined(operator) || isUndefined(value) || isUndefined(type) || isUndefined(field) || value === '' || Number.isNaN(value) || value === null;
}
function numberFilter(filter, properties) {
  var newField = properties[filter.field];
  switch (filter.operator) {
    case '<':
      return newField < filter.value;
    case '<=':
      return newField <= filter.value;
    case '=':
      return newField === filter.value;
    case '>':
      return newField > filter.value;
    case '>=':
      return newField >= filter.value;
    default:
      if (Array.isArray(filter.value)) {
        var _filter$value = _slicedToArray(filter.value, 2),
          value1 = _filter$value[0],
          value2 = _filter$value[1];
        return newField >= value1 && newField <= value2;
      }
  }
}
function stringFilter(filter, properties) {
  var newField = String(properties[filter.field]);
  var value = filter.value;
  switch (filter.operator) {
    case 'NOT_IN':
      return !(value !== null && value !== void 0 && value.includes(newField));
    case 'IN':
      return value === null || value === void 0 ? void 0 : value.includes(newField);
    case 'LIKE':
      return (newField === null || newField === void 0 ? void 0 : newField.indexOf(value)) > -1;
    default:
      return (newField === null || newField === void 0 ? void 0 : newField.indexOf(value)) <= -1;
  }
}
export var filterFeatureAtom = atom({
  key: 'filterFeature',
  default: []
});
export function useFilterFeatures() {
  var _useFeature = useFeature(),
    features = _useFeature.features;
  var _useFilter = useFilter(),
    filters = _useFilter.filters;
  var setFilterFeature = useSetRecoilState(filterFeatureAtom);
  var filterFeatures = useRecoilValue(filterFeatureAtom);
  useEffect(function () {
    if (isEmpty(filters)) {
      setFilterFeature(_toConsumableArray(features));
      return;
    }
    var newFeature = _toConsumableArray(features);
    // 过滤空值
    var setNotEmptyFilter = filters.filter(function (a) {
      return !isEmptyFilter(a);
    });
    // 查找 and 条件 且
    var andFilters = setNotEmptyFilter.filter(function (item) {
      return item.logic === 'and';
    });
    // 查找 or 条件 或
    var orFilters = setNotEmptyFilter.filter(function (item) {
      return item.logic === 'or';
    });
    if (!isEmpty(orFilters)) {
      newFeature = newFeature.filter(function (_ref) {
        var properties = _ref.properties;
        return orFilters.some(function (filter) {
          var func = filter.type === 'number' ? numberFilter : stringFilter;
          return func(filter, properties);
        });
      });
    }
    if (!isEmpty(andFilters)) {
      newFeature = newFeature.filter(function (_ref2) {
        var properties = _ref2.properties;
        return andFilters.every(function (filter) {
          var func = filter.type === 'number' ? numberFilter : stringFilter;
          return func(filter, properties);
        });
      });
    }
    setFilterFeature(newFeature);
  }, [features, filters, setFilterFeature]);
  return {
    features: filterFeatures
  };
}
