// @antv/l7-editor/lib/components/layer-popup/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/layer-popup/index.tsx
var layer_popup_exports = {};
__export(layer_popup_exports, {
  LayerPopup: () => LayerPopup
});
module.exports = __toCommonJS(layer_popup_exports);
var import_icons = require("@ant-design/icons");
var import_l7_draw = require("@antv/l7-draw");
var import_larkmap = require("@antv/larkmap");
var import_turf = require("@turf/turf");
var import_antd = require("antd");
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_constants = require("../../constants");
var import_hooks = require("../../hooks");
var import_recoil = require("../../recoil");
var import_utils = require("../../utils");
var import_prettier_text = require("../../utils/prettier-text");
var import_styles = __toESM(require("./styles"));
var { Paragraph } = import_antd.Typography;
var LayerPopup = () => {
  const [form] = import_antd.Form.useForm();
  const scene = (0, import_larkmap.useScene)();
  const {
    saveEditorText,
    isDraw,
    setIsDraw,
    resetFeatures,
    features,
    setFeatures,
    revertCoord,
    transformCoord
  } = (0, import_recoil.useFeature)();
  const { layerColor, popupTrigger, showDrawDistance, showDrawArea } = (0, import_recoil.useGlobal)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const helperText = (0, import_hooks.useDrawHelper)();
  const styles = (0, import_styles.default)();
  const [addOpen, setAddOpen] = (0, import_react.useState)(false);
  const [popupProps, setPopupProps] = (0, import_react.useState)({
    lngLat: {
      lng: 0,
      lat: 0
    },
    visible: false,
    feature: null
  });
  const [tableClick, setTableClick] = (0, import_react.useState)({
    isInput: false,
    index: null
  });
  const [addValue, setAddValue] = (0, import_react.useState)({
    label: void 0,
    value: void 0
  });
  const targetFeature = (0, import_react.useMemo)(() => {
    return features.find(
      (feature) => {
        var _a;
        return (
          // @ts-ignore
          ((_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.Index]) === popupProps.featureIndex
        );
      }
    );
  }, [features, popupProps]);
  const featureFields = Object.entries((targetFeature == null ? void 0 : targetFeature.properties) ?? {});
  const allLayerList = (0, import_larkmap.useLayerList)();
  const layerList = (0, import_react.useMemo)(() => {
    const layerIds = [
      import_constants.LayerId.PointLayer,
      import_constants.LayerId.LineLayer,
      import_constants.LayerId.PolygonLayer
    ];
    return allLayerList.filter((layer) => {
      return layerIds.includes(layer.id);
    });
  }, [allLayerList]);
  const disabledEdit = (feature) => {
    if (feature) {
      const reg = RegExp(/Multi/);
      return reg.test(feature.geometry.type);
    }
    return false;
  };
  const onLayerClick = (0, import_react.useCallback)(
    (e) => {
      if (!isDraw) {
        const { lngLat, feature } = e;
        const featureIndex = feature.properties[import_constants.FeatureKey.Index];
        const isIndex = popupProps.featureIndex === feature.properties[import_constants.FeatureKey.Index];
        if (popupProps.visible && isIndex) {
          setPopupProps((oldPopupProps) => {
            return {
              ...oldPopupProps,
              visible: false,
              featureIndex: void 0,
              feature: null
            };
          });
          return;
        }
        setPopupProps({
          lngLat,
          visible: true,
          featureIndex,
          feature: e.feature
        });
      }
    },
    [setPopupProps, popupProps, isDraw]
  );
  const onLayerMouseenter = (0, import_react.useCallback)(
    (e) => {
      if (!isDraw) {
        const { lngLat, feature } = e;
        const featureIndex = feature.properties[import_constants.FeatureKey.Index];
        setPopupProps({
          lngLat,
          visible: true,
          featureIndex
        });
      }
    },
    [setPopupProps, isDraw]
  );
  const onLayerMouseout = (0, import_react.useCallback)(() => {
    if (!isDraw) {
      setPopupProps((oldPopupProps) => {
        return {
          ...oldPopupProps,
          visible: false,
          featureIndex: void 0
        };
      });
    }
  }, [setPopupProps, isDraw]);
  const onEdit = (feature) => {
    setIsDraw(true);
    let newFeature = (0, import_lodash_es.cloneDeep)(feature);
    const newFeatures = features.filter((item) => {
      var _a;
      return (
        //@ts-ignore
        item.properties[import_constants.FeatureKey.Index] !== //@ts-ignore
        ((_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.Index])
      );
    });
    const index = features.findIndex((item) => {
      var _a;
      return (
        //@ts-ignore
        item.properties[import_constants.FeatureKey.Index] === //@ts-ignore
        ((_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.Index])
      );
    });
    const onEditFinish = (selectFeature, draw2) => {
      if (!selectFeature) {
        const data = draw2.getData();
        if (data.length) {
          newFeature = {
            ...data[0],
            properties: feature == null ? void 0 : feature.properties
          };
          const newTransformFeatures = revertCoord([newFeature])[0];
          features[index] = newTransformFeatures;
        } else {
          features.splice(index, 1);
        }
        draw2.destroy();
        setIsDraw(false);
        resetFeatures(features);
      }
    };
    const options = {
      initialData: [newFeature],
      maxCount: 1,
      style: (0, import_utils.getDrawStyle)(layerColor),
      helper: helperText,
      distanceOptions: showDrawDistance ? {} : void 0,
      areaOptions: showDrawArea ? {} : void 0
    };
    let draw;
    const [originFeature] = transformCoord([newFeature]);
    const type = originFeature == null ? void 0 : originFeature.geometry.type;
    if (type === "Point") {
      draw = new import_l7_draw.DrawPoint(scene, options);
    } else if (type === "LineString") {
      draw = new import_l7_draw.DrawLine(scene, options);
    } else if (type === "Polygon" && (0, import_utils.isRect)(originFeature)) {
      draw = new import_l7_draw.DrawRect(scene, options);
    } else if (type === "Polygon" && (0, import_utils.isCircle)(originFeature)) {
      draw = new import_l7_draw.DrawCircle(scene, options);
    } else {
      draw = new import_l7_draw.DrawPolygon(scene, options);
    }
    draw.enable();
    draw.setActiveFeature(draw.getData()[0]);
    setFeatures(newFeatures);
    draw.on(
      import_l7_draw.DrawEvent.Select,
      (selectFeature) => onEditFinish(selectFeature, draw)
    );
    setPopupProps({
      visible: false,
      featureIndex: void 0
    });
  };
  const onLayerDblClick = (e) => {
    const { feature } = e;
    if (!disabledEdit(feature) && !isDraw) {
      onEdit(feature);
    }
  };
  (0, import_react.useEffect)(() => {
    layerList.forEach((layer) => layer.on("dblclick", onLayerDblClick));
    return () => {
      layerList.forEach((layer) => layer.off("dblclick", onLayerDblClick));
    };
  }, [onLayerDblClick, layerList, popupTrigger, scene]);
  (0, import_react.useEffect)(() => {
    const layerEvent = {
      click: [{ event: "click", click: onLayerClick }],
      hover: [
        { event: "mouseenter", click: onLayerMouseenter },
        { event: "mouseout", click: onLayerMouseout }
      ]
    };
    layerEvent[popupTrigger].forEach((e) => {
      layerList.forEach((layer) => layer.on(e.event, e.click));
    });
    return () => {
      layerEvent[popupTrigger].forEach((e) => {
        layerList.forEach((layer) => layer.off(e.event, e.click));
      });
    };
  }, [onLayerClick, onLayerMouseenter, layerList, popupTrigger, scene]);
  const save = (key, value) => {
    const formValue = form.getFieldValue("input");
    if (value !== formValue) {
      const properties = {
        ...popupProps.feature.properties,
        [key]: formValue
      };
      const feature = { ...popupProps.feature, properties };
      setPopupProps((event) => ({ ...event, feature }));
      const index = features.findIndex((item) => {
        var _a;
        return (
          //@ts-ignore
          item.properties[import_constants.FeatureKey.Index] === //@ts-ignore
          ((_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.Index])
        );
      });
      features[index] = feature;
      saveEditorText((0, import_prettier_text.prettierText)({ content: (0, import_turf.featureCollection)(features) }));
    }
    setTableClick({ isInput: false, index: null });
  };
  const addBlur = () => {
    if (addValue.label && addValue.value) {
      const properties = {
        ...popupProps.feature.properties,
        [addValue.label]: addValue.value
      };
      const feature = { ...popupProps.feature, properties };
      setPopupProps((event) => ({ ...event, feature }));
      const index = features.findIndex((item) => {
        var _a;
        return (
          //@ts-ignore
          item.properties[import_constants.FeatureKey.Index] === //@ts-ignore
          ((_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.Index])
        );
      });
      features[index] = feature;
      saveEditorText((0, import_prettier_text.prettierText)({ content: (0, import_turf.featureCollection)(features) }));
      setAddValue({ label: void 0, value: void 0 });
      setAddOpen(!addOpen);
    }
  };
  const popupTable = /* @__PURE__ */ import_react.default.createElement("div", null, featureFields.length || addOpen ? /* @__PURE__ */ import_react.default.createElement(
    "div",
    {
      className: styles.layerPopupInfo,
      onWheel: (e) => {
        e.stopPropagation();
      }
    },
    /* @__PURE__ */ import_react.default.createElement(import_antd.Descriptions, { size: "small", bordered: true, column: 1 }, featureFields.map(([key, value], index) => {
      if (!(value instanceof Object)) {
        return /* @__PURE__ */ import_react.default.createElement(import_antd.Descriptions.Item, { label: key, key }, /* @__PURE__ */ import_react.default.createElement(
          Paragraph,
          {
            copyable: { text: `${value}` },
            style: {
              display: "flex",
              alignItems: "center",
              justifyContent: "space-between"
            }
          },
          tableClick.isInput && tableClick.index === index ? /* @__PURE__ */ import_react.default.createElement(import_antd.Form, { form, className: styles.form }, /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: "input" }, typeof value === "number" ? /* @__PURE__ */ import_react.default.createElement(
            import_antd.InputNumber,
            {
              autoFocus: true,
              onPressEnter: () => save(key, value),
              onBlur: () => save(key, value)
            }
          ) : /* @__PURE__ */ import_react.default.createElement(
            import_antd.Input,
            {
              autoFocus: true,
              onPressEnter: () => save(key, value),
              onBlur: () => save(key, value)
            }
          ))) : /* @__PURE__ */ import_react.default.createElement(
            "div",
            {
              style: { width: "100%" },
              onClick: () => {
                setTableClick({
                  isInput: !tableClick.isInput,
                  index
                });
                setAddOpen(false);
                form.setFieldsValue({ input: value });
              }
            },
            value ? String(value) : "-"
          )
        ));
      }
      return null;
    }), addOpen && /* @__PURE__ */ import_react.default.createElement(
      import_antd.Descriptions.Item,
      {
        contentStyle: { paddingRight: 8 },
        label: /* @__PURE__ */ import_react.default.createElement(
          import_antd.Input,
          {
            size: "small",
            onChange: (e) => {
              setAddValue((prevState) => ({
                ...prevState,
                label: e.target.value
              }));
            },
            onBlur: addBlur
          }
        )
      },
      /* @__PURE__ */ import_react.default.createElement("div", { className: styles.addField }, /* @__PURE__ */ import_react.default.createElement(
        import_antd.Input,
        {
          size: "small",
          onChange: (e) => {
            setAddValue((prevState) => ({
              ...prevState,
              value: e.target.value
            }));
          },
          onBlur: addBlur
        }
      ), /* @__PURE__ */ import_react.default.createElement(
        import_antd.Button,
        {
          type: "text",
          className: styles.addBut,
          icon: /* @__PURE__ */ import_react.default.createElement(import_icons.CloseOutlined, null),
          onClick: () => {
            setAddOpen(false);
          }
        }
      ))
    ))
  ) : /* @__PURE__ */ import_react.default.createElement(
    import_antd.Empty,
    {
      description: t("layer_popup.index.dangQianYuanSuWu"),
      style: { margin: "12px 0" }
    }
  ), /* @__PURE__ */ import_react.default.createElement("div", { style: { marginTop: 10 } }, !addOpen && popupTrigger === "click" && /* @__PURE__ */ import_react.default.createElement(
    import_antd.Button,
    {
      type: "dashed",
      block: true,
      style: { marginBottom: 6 },
      onClick: () => {
        setAddOpen(!addOpen);
      }
    },
    /* @__PURE__ */ import_react.default.createElement(import_icons.PlusSquareOutlined, null),
    t("layer_popup.index.tianJiaZiDuan")
  )));
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, popupProps.visible && typeof popupProps.featureIndex === "number" && targetFeature && /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.Popup,
    {
      className: styles.layerPopupContent,
      lngLat: popupProps.lngLat,
      closeButton: false,
      offsets: [0, 10],
      followCursor: popupTrigger === "hover",
      closeOnClick: true,
      onClose: () => {
        setAddOpen(false);
      }
    },
    /* @__PURE__ */ import_react.default.createElement(
      "div",
      {
        className: styles.layerPopup,
        onClick: (e) => {
          e.stopPropagation();
        }
      },
      popupTable,
      /* @__PURE__ */ import_react.default.createElement("div", { className: styles.layerPopupBtnGroup }, popupTrigger === "click" && /* @__PURE__ */ import_react.default.createElement(
        import_antd.Tooltip,
        {
          title: disabledEdit(popupProps.feature) ? t("layer_popup.index.mULTI") : ""
        },
        /* @__PURE__ */ import_react.default.createElement(
          import_antd.Button,
          {
            size: "small",
            type: "link",
            onClick: () => onEdit(popupProps.feature),
            disabled: disabledEdit(popupProps.feature)
          },
          t("layer_popup.index.gengGaiHuiZhi")
        )
      ), popupTrigger === "click" && /* @__PURE__ */ import_react.default.createElement(
        import_antd.Button,
        {
          size: "small",
          type: "link",
          danger: true,
          onClick: () => {
            resetFeatures(
              features.filter((_, index) => {
                return index !== popupProps.featureIndex;
              })
            );
            setPopupProps({
              visible: false,
              featureIndex: void 0
            });
          }
        },
        t("app_table.index.shanChu")
      ))
    )
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LayerPopup
});
