// @antv/l7-editor/lib/components/layer-popup/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/layer-popup/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var { useToken } = import_antd.theme;
var useStyle = () => {
  const { token } = useToken();
  const { colorBgContainer, colorBgElevated, colorFillSecondary } = token;
  return {
    layerPopupContent: import_css.css`
      width: 300px !important;
      z-index: 1000;
      .l7-popup-tip {
        border-top-color: ${colorBgContainer} !important;
      }
      .l7-popup-content {
        background: ${colorBgContainer} !important;
      }
    `,
    layerPopup: import_css.css`
      background: ${colorBgContainer} !important;
    `,
    layerPopupInfo: import_css.css`
      *::-webkit-scrollbar {
        width: 8px;
        background-color: ${colorBgElevated};
      }
      *::-webkit-scrollbar-thumb {
        background-color: ${colorFillSecondary};
        border-radius: 10px;
      }
      .ant-descriptions {
        max-height: 200px;
        overflow: auto;
      }
      .ant-typography {
        margin-bottom: 0;
        .ant-typography-copy {
          opacity: 0;
        }
        &:hover {
          .ant-typography-copy {
            opacity: 1;
          }
        }
      }
    `,
    form: import_css.css`
      .ant-form-item {
        margin-bottom: 0px !important;
      }
    `,
    layerPopupBtnGroup: import_css.css`
      display: flex;
      justify-content: space-between;
      align-items: center;
    `,
    addField: import_css.css`
      display: flex;
      align-items: center;
    `,
    addBut: import_css.css`
      padding: 4px;
      margin-left: 4px;
    `,
    addItem: import_css.css``
  };
};
var styles_default = useStyle;
