// @antv/l7-editor/es/components/map-control-group/layer-color-control/index.js
import { CustomControl } from '@antv/larkmap';
import { ColorPicker, Tooltip } from 'antd';
import classNames from 'classnames';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useGlobal } from "../../../recoil";
import useStyle from "./styles";
var LayerColorControl = function LayerColorControl() {
  var _useGlobal = useGlobal(),
    layerColor = _useGlobal.layerColor,
    setLayerColor = _useGlobal.setLayerColor;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var styles = useStyle();
  return /*#__PURE__*/React.createElement(CustomControl, {
    position: "bottomright"
  }, /*#__PURE__*/React.createElement(Tooltip, {
    title: t('layer_color_control.index.tuCengYanSeXuan'),
    placement: "left"
  }, /*#__PURE__*/React.createElement("div", {
    id: "l7-editor-color"
  }, /*#__PURE__*/React.createElement(ColorPicker, {
    value: layerColor,
    className: classNames([styles.l7ButtonControl]),
    onChange: function onChange(value, hex) {
      setLayerColor(hex);
    },
    format: "rgb",
    presets: [{
      label: t('layer_color_control.index.changYongYanSe'),
      colors: ['#F5222D', '#FA8C16', '#FADB14', '#8BBB11', '#52C41A', '#13A8A8', '#1677FF', '#2F54EB', '#722ED1', '#EB2F96']
    }]
  }))));
};
export default LayerColorControl;
