// @antv/l7-editor/es/components/map-control-group/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { FullscreenControl, GeoLocateControl, LogoControl, MouseLocationControl, ScaleControl, ZoomControl } from '@antv/larkmap';
import React, { useEffect, useState } from 'react';
import { useGlobal } from "../../recoil";
import { AdministrativeSelect } from "./administrative-select-control";
import { AutoControl } from "./auto-control";
import { ClearControl } from "./clear-control";
import DrawControl from "./draw-control";
import { ExportImage } from "./export-image-control";
import FilterControl from "./filter-form-list-control";
import { L7MapOptionControl } from "./l7-option-control";
import LayerColorControl from "./layer-color-control";
import LocationSearchControl from "./location-search-control";
import { MapAdministrativeControl } from "./map-administrative-control";
import MapThemeControl from "./map-theme-control";
import { OfficialLayerControl } from "./official-layer-control";
import useStyles from "./styles";
import { TextLayerControl } from "./text-layer-control";
var DefaultMapControl = {
  drawControl: true,
  clearControl: true,
  zoomControl: true,
  scaleControl: true,
  locationSearchControl: true,
  mouseLocationControl: true,
  filterControl: true,
  officialLayerControl: true,
  mapThemeControl: true,
  geoLocateControl: true,
  layerColorControl: true,
  autoControl: true,
  fullscreenControl: true,
  administrativeSelectControl: true,
  mapAdministrativeControl: true,
  logoControl: true,
  textLayerControl: true,
  exportImageControl: true,
  L7MapOptionControl: true
};
export var MapControlGroup = function MapControlGroup(_ref) {
  var mapControl = _ref.mapControl;
  var _useGlobal = useGlobal(),
    baseMap = _useGlobal.baseMap;
  var styles = useStyles();
  var _useState = useState(DefaultMapControl),
    _useState2 = _slicedToArray(_useState, 2),
    isControlGroupState = _useState2[0],
    setIsControlGroup = _useState2[1];
  useEffect(function () {
    setIsControlGroup(_objectSpread(_objectSpread({}, DefaultMapControl), mapControl));
  }, [mapControl]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, isControlGroupState.logoControl && /*#__PURE__*/React.createElement(LogoControl, {
    position: "leftbottom"
  }), isControlGroupState.scaleControl && /*#__PURE__*/React.createElement(ScaleControl, {
    position: "leftbottom",
    className: styles.scalesControl
  }), isControlGroupState.mouseLocationControl && /*#__PURE__*/React.createElement(MouseLocationControl, {
    position: "leftbottom",
    className: styles.mouseLocatio
  }), isControlGroupState.drawControl && /*#__PURE__*/React.createElement(DrawControl, null), isControlGroupState.clearControl && /*#__PURE__*/React.createElement(ClearControl, null), isControlGroupState.zoomControl && /*#__PURE__*/React.createElement(ZoomControl, {
    className: styles.zoom,
    showZoom: true,
    position: "rightbottom"
  }), isControlGroupState.mapAdministrativeControl && /*#__PURE__*/React.createElement(MapAdministrativeControl, null), isControlGroupState.administrativeSelectControl && /*#__PURE__*/React.createElement(AdministrativeSelect, null), isControlGroupState.locationSearchControl && /*#__PURE__*/React.createElement(LocationSearchControl, null), isControlGroupState.filterControl && /*#__PURE__*/React.createElement(FilterControl, null), baseMap === 'Gaode' && isControlGroupState.officialLayerControl && /*#__PURE__*/React.createElement(OfficialLayerControl, null), isControlGroupState.mapThemeControl && /*#__PURE__*/React.createElement(MapThemeControl, null), isControlGroupState.geoLocateControl && /*#__PURE__*/React.createElement(GeoLocateControl, {
    position: "topleft",
    className: styles.fullScreen
  }), isControlGroupState.layerColorControl && /*#__PURE__*/React.createElement(LayerColorControl, null), isControlGroupState.autoControl && /*#__PURE__*/React.createElement(AutoControl, null), isControlGroupState.fullscreenControl && /*#__PURE__*/React.createElement(FullscreenControl, {
    position: "bottomright",
    className: styles.fullScreen
  }), isControlGroupState.textLayerControl && /*#__PURE__*/React.createElement(TextLayerControl, null), isControlGroupState.exportImageControl && /*#__PURE__*/React.createElement(ExportImage, null), isControlGroupState.L7MapOptionControl && /*#__PURE__*/React.createElement(L7MapOptionControl, null));
};
