// @antv/l7-editor/lib/components/app-header/btn/i18n-btn.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/btn/i18n-btn.tsx
var i18n_btn_exports = {};
__export(i18n_btn_exports, {
  default: () => i18n_btn_default
});
module.exports = __toCommonJS(i18n_btn_exports);
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_locales = require("../../../locales");
var import_recoil = require("../../../recoil");
var import_iconfont = require("../../iconfont");
var import_styles = __toESM(require("../styles"));
var items = import_locales.LangList.map((item) => ({
  key: item.lang,
  label: item.name
}));
var I18nBtn = () => {
  const { i18n } = (0, import_react_i18next.useTranslation)();
  const { locale, setLocale } = (0, import_recoil.useGlobal)();
  const styles = (0, import_styles.default)();
  const onLocaleChange = ({ key }) => {
    i18n.changeLanguage(key);
    setLocale(key);
  };
  return /* @__PURE__ */ import_react.default.createElement(
    import_antd.Dropdown,
    {
      menu: {
        items,
        onClick: onLocaleChange,
        selectedKeys: [locale],
        selectable: true
      },
      placement: "bottom"
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Button,
      {
        className: styles.locale,
        icon: /* @__PURE__ */ import_react.default.createElement(import_iconfont.IconFont, { className: styles.localeIcon, type: "icon-in" })
      }
    )
  );
};
var i18n_btn_default = I18nBtn;
