// @antv/l7-editor/lib/components/app-header/btn/setting-btn.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/btn/setting-btn.tsx
var setting_btn_exports = {};
__export(setting_btn_exports, {
  SettingBtn: () => SettingBtn
});
module.exports = __toCommonJS(setting_btn_exports);
var import_icons = require("@ant-design/icons");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../../recoil");
var SettingBtn = () => {
  const [isModalOpen, setIsModalOpen] = (0, import_react.useState)(false);
  const {
    popupTrigger,
    setPopupTrigger,
    autoFitBounds,
    setAutoFitBounds,
    showDrawDistance,
    setShowDrawDistance,
    showDrawArea,
    setShowDrawArea
  } = (0, import_recoil.useGlobal)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const [form] = import_antd.Form.useForm();
  const showModal = () => {
    setIsModalOpen(true);
  };
  const handleOk = () => {
    form.submit();
  };
  const handleCancel = () => {
    setIsModalOpen(false);
    form.resetFields();
  };
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: t("btn.setting_btn.diTuSheZhi"), placement: "bottom" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Button,
    {
      id: "l7-editor-set",
      icon: /* @__PURE__ */ import_react.default.createElement(import_icons.SettingOutlined, null),
      onClick: showModal
    },
    t("btn.setting_btn.sheZhi")
  )), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Modal,
    {
      title: t("btn.setting_btn.sheZhi"),
      open: isModalOpen,
      onOk: handleOk,
      onCancel: handleCancel,
      destroyOnClose: true,
      okText: t("btn.setting_btn.queRen"),
      cancelText: t("btn.setting_btn.quXiao")
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Form,
      {
        form,
        initialValues: {
          popupTrigger,
          autoFitBounds,
          showDrawDistance,
          showDrawArea
        },
        style: { textAlign: "right" },
        onFinish: (e) => {
          setIsModalOpen(false);
          setPopupTrigger(e.popupTrigger);
          setAutoFitBounds(e.autoFitBounds);
          setShowDrawDistance(e.showDrawDistance);
          setShowDrawArea(e.showDrawArea);
        }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Form.Item,
        {
          name: "popupTrigger",
          label: t("btn.setting_btn.tuCengQiPaoZhan")
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Group, null, /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { value: "click" }, t("btn.setting_btn.dianJi")), /* @__PURE__ */ import_react.default.createElement(import_antd.Radio.Button, { value: "hover" }, t("btn.setting_btn.huaRu")))
      ),
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Form.Item,
        {
          name: "autoFitBounds",
          valuePropName: "checked",
          label: t("btn.setting_btn.ziDongSuoFangZhi")
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Switch, null)
      ),
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Form.Item,
        {
          name: "showDrawDistance",
          valuePropName: "checked",
          label: t("btn.setting_btn.juLi")
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Switch, null)
      ),
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Form.Item,
        {
          name: "showDrawArea",
          valuePropName: "checked",
          label: t("btn.setting_btn.mianJi")
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Switch, null)
      )
    )
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SettingBtn
});
