// @antv/l7-editor/lib/components/app-table/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-table/index.tsx
var app_table_exports = {};
__export(app_table_exports, {
  AppTable: () => AppTable
});
module.exports = __toCommonJS(app_table_exports);
var import_icons = require("@ant-design/icons");
var import_turf = require("@turf/turf");
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_classnames = __toESM(require("classnames"));
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_constants = require("../../constants");
var import_recoil = require("../../recoil");
var import_prettier_text = require("../../utils/prettier-text");
var import_styles = __toESM(require("./styles"));
var { Text } = import_antd.Typography;
var formatTableValue = (value) => {
  if ((0, import_lodash_es.isNull)(value) || (0, import_lodash_es.isUndefined)(value) || value === "") {
    return "-";
  }
  return value instanceof Object ? JSON.stringify(value) : /* @__PURE__ */ import_react.default.createElement(Text, { ellipsis: { tooltip: String(value) } }, String(value));
};
var FormContext = import_react.default.createContext(null);
var EditableRow = (props) => {
  const [form] = import_antd.Form.useForm();
  return /* @__PURE__ */ import_react.default.createElement(import_antd.Form, { form, component: false }, /* @__PURE__ */ import_react.default.createElement(FormContext.Provider, { value: form }, /* @__PURE__ */ import_react.default.createElement("tr", { ...props })));
};
var EditableCell = ({
  editable,
  children,
  dataIndex,
  record,
  inputType,
  handleSave,
  features,
  scene,
  isDraw,
  ...restProps
}) => {
  const [editing, setEditing] = (0, import_react.useState)(false);
  const inputRef = (0, import_react.useRef)(null);
  const styles = (0, import_styles.default)();
  const form = (0, import_react.useContext)(FormContext);
  (0, import_react.useEffect)(() => {
    var _a;
    if (editing) {
      (_a = inputRef.current) == null ? void 0 : _a.focus();
    }
  }, [editing]);
  const toggleEdit = () => {
    if (scene && !isDraw) {
      setEditing(!editing);
      form == null ? void 0 : form.setFieldsValue(
        inputType !== "object" ? {
          [dataIndex]: record[dataIndex]
        } : { [dataIndex]: JSON.stringify(record[dataIndex]) }
      );
    }
  };
  const save = async () => {
    try {
      const fieldValue = inputType !== "object" ? {
        [dataIndex]: record[dataIndex]
      } : { [dataIndex]: JSON.stringify(record[dataIndex]) };
      const values = await (form == null ? void 0 : form.validateFields());
      if (JSON.stringify(values) !== JSON.stringify(fieldValue)) {
        const newValues = inputType === "object" ? { [dataIndex]: JSON.parse(values[dataIndex]) } : values;
        toggleEdit();
        handleSave({
          ...record,
          ...newValues,
          newValues
        });
      } else {
        toggleEdit();
      }
    } catch (errInfo) {
      console.log("Save failed:", errInfo);
    }
  };
  let childNode = children;
  if (editable) {
    childNode = editing ? /* @__PURE__ */ import_react.default.createElement(
      import_antd.Form.Item,
      {
        style: {
          margin: 0
        },
        name: dataIndex
      },
      inputType === "number" ? /* @__PURE__ */ import_react.default.createElement(import_antd.InputNumber, { ref: inputRef, onPressEnter: save, onBlur: save }) : /* @__PURE__ */ import_react.default.createElement(import_antd.Input, { ref: inputRef, onPressEnter: save, onBlur: save })
    ) : /* @__PURE__ */ import_react.default.createElement(
      "div",
      {
        className: !isDraw ? styles.editableCellValueWrap : "",
        style: {
          paddingRight: 24
        },
        onClick: toggleEdit
      },
      children
    );
  }
  return /* @__PURE__ */ import_react.default.createElement("td", { ...restProps }, childNode);
};
var components = {
  body: {
    row: EditableRow,
    cell: EditableCell
  }
};
var AppTable = () => {
  const container = (0, import_react.useRef)(null);
  const styles = (0, import_styles.default)();
  const { height = 0 } = (0, import_ahooks.useSize)(container) ?? {};
  const { saveEditorText, isDraw, scene, features, resetFeatures } = (0, import_recoil.useFeature)();
  const [newDataSource, setNewDataSource] = (0, import_react.useState)([]);
  const { transformCoord } = (0, import_recoil.useFeature)();
  const [addInputValue, setAddInputValue] = (0, import_react.useState)(
    void 0
  );
  const { t } = (0, import_react_i18next.useTranslation)();
  (0, import_react.useEffect)(() => {
    const dataSource = features.map((item, index) => {
      const { properties } = item;
      return { __index: index + 1, ...properties };
    });
    setNewDataSource(dataSource);
  }, [features]);
  const defaultColumns = (0, import_react.useMemo)(() => {
    const newColumns2 = [];
    const featureKeyList = Array.from(
      new Set(
        features.map((item) => {
          const { properties } = item;
          return Object.keys(properties);
        }).flat()
      )
    );
    if (featureKeyList) {
      newColumns2.push({
        title: t("app_table.index.xuHao"),
        dataIndex: "__index",
        key: `__index`,
        width: 80,
        align: "center",
        fixed: "left",
        sorter: (a, b) => a.__index - b.__index
      });
    }
    const delConfirm = (key) => {
      const newFeatures = features.map((feature) => {
        const { properties } = feature;
        const newProperties = (0, import_lodash_es.cloneDeep)(properties);
        delete newProperties[key];
        return {
          ...feature,
          properties: newProperties
        };
      });
      resetFeatures(newFeatures);
    };
    featureKeyList.forEach((key, index) => {
      const options = (0, import_lodash_es.uniqBy)(
        newDataSource.map((item) => item[key]).map((value) => {
          return {
            text: formatTableValue(value),
            value
          };
        }),
        "text"
      );
      newColumns2.push({
        title: /* @__PURE__ */ import_react.default.createElement("div", { className: styles.deleteColumns }, /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: key }, /* @__PURE__ */ import_react.default.createElement(
          Text,
          {
            style: { wordWrap: "break-word", whiteSpace: "break-spaces" },
            ellipsis: { tooltip: key }
          },
          key
        )), /* @__PURE__ */ import_react.default.createElement(
          import_antd.Popconfirm,
          {
            title: t("app_table.index.shanChuLie"),
            onConfirm: () => {
              delConfirm(key);
            }
          },
          /* @__PURE__ */ import_react.default.createElement(
            import_antd.Button,
            {
              type: "text",
              className: (0, import_classnames.default)([styles.delButton, "delete"])
            },
            /* @__PURE__ */ import_react.default.createElement(import_icons.DeleteOutlined, null)
          )
        )),
        width: 200,
        align: "center",
        dataIndex: key,
        key: `${key}${index}`,
        editable: true,
        render: formatTableValue,
        filters: options.length ? options : void 0,
        onFilter: (value, record) => {
          return (record[key] ?? "") === value;
        },
        filterSearch: true,
        sorter: !options.length ? (a, b) => {
          return (typeof a[key] === "string" || !a[key] ? 0 : a[key]) - (typeof b[key] === "string" || !b[key] ? 0 : b[key]);
        } : void 0
      });
    });
    const confirm = () => {
      if (!addInputValue) {
        return;
      }
      const newFeatures = features.map((item) => {
        const { properties } = item;
        const newProperties = { ...properties, [addInputValue]: "" };
        return {
          ...item,
          properties: newProperties
        };
      });
      resetFeatures(newFeatures);
      setAddInputValue(void 0);
    };
    newColumns2.push({
      title: /* @__PURE__ */ import_react.default.createElement("div", { className: styles.addColumns }, /* @__PURE__ */ import_react.default.createElement(
        import_antd.Popconfirm,
        {
          placement: "bottom",
          icon: null,
          title: t("app_table.index.tianJiaLieZiDuan"),
          description: /* @__PURE__ */ import_react.default.createElement("div", { style: { display: "flex", alignItems: "center" } }, /* @__PURE__ */ import_react.default.createElement("div", { style: { width: 80 } }, t("app_table.index.ziDuanMing")), /* @__PURE__ */ import_react.default.createElement(
            import_antd.Input,
            {
              placeholder: t("app_table.index.qingShuRuLieZiDuan"),
              value: addInputValue,
              onChange: (e) => {
                setAddInputValue(e.target.value);
              }
            }
          )),
          onConfirm: confirm,
          onCancel: () => {
            setAddInputValue(void 0);
          }
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Button, { type: "text", className: styles.addButton }, /* @__PURE__ */ import_react.default.createElement(import_icons.PlusSquareOutlined, null))
      ), t("app_table.index.caoZuo")),
      key: "action",
      width: 120,
      align: "center",
      fixed: "right",
      render: (_, record) => /* @__PURE__ */ import_react.default.createElement(import_antd.Space, { size: "middle" }, /* @__PURE__ */ import_react.default.createElement(
        "a",
        {
          onClick: () => {
            if (scene) {
              const targetFeature = features.find((item) => {
                return item.properties[import_constants.FeatureKey.Index] === record[import_constants.FeatureKey.Index];
              });
              if (targetFeature) {
                const newBboxFit = transformCoord([targetFeature]);
                if (newBboxFit[0].geometry.type === "Point") {
                  const content = (0, import_turf.center)(newBboxFit[0]);
                  scene.setCenter(
                    content.geometry.coordinates
                  );
                } else {
                  const content = (0, import_turf.bbox)(newBboxFit[0]);
                  scene.fitBounds([
                    [content[0], content[1]],
                    [content[2], content[3]]
                  ]);
                }
              }
            }
          }
        },
        t("app_table.index.dingWei")
      ), /* @__PURE__ */ import_react.default.createElement(
        "a",
        {
          onClick: () => {
            resetFeatures(
              features.filter((feature, index) => {
                return index !== record[import_constants.FeatureKey.Index];
              })
            );
            import_antd.message.success(t("app_table.index.shuJuShanChuCheng"));
          },
          style: { color: "#ff0000" }
        },
        t("app_table.index.shanChu")
      ))
    });
    return newColumns2;
  }, [
    addInputValue,
    features,
    newDataSource,
    resetFeatures,
    scene,
    styles.addButton,
    styles.addColumns,
    styles.delButton,
    styles.deleteColumns,
    t,
    transformCoord
  ]);
  const handleSave = (row) => {
    const newData = [...newDataSource];
    const index = newData.findIndex((item2) => row.__index === (item2 == null ? void 0 : item2.__index));
    const item = newData[index];
    newData.splice(index, 1, {
      ...item,
      ...row
    });
    const { __index, newValues } = row;
    const indexProperties = {
      ...features[__index - 1].properties,
      ...newValues
    };
    const feature = { ...features[__index - 1], properties: indexProperties };
    features[index] = feature;
    saveEditorText((0, import_prettier_text.prettierText)({ content: (0, import_turf.featureCollection)(features) }));
    setNewDataSource(newData);
  };
  const newColumns = defaultColumns.map((col) => {
    if (!col.editable) {
      return col;
    }
    return {
      ...col,
      onCell: (record) => ({
        record,
        editable: col.editable,
        dataIndex: col.dataIndex,
        title: col.title,
        inputType: typeof record[col.dataIndex],
        newDataSource,
        features,
        scene,
        isDraw,
        handleSave
      })
    };
  });
  return /* @__PURE__ */ import_react.default.createElement("div", { style: { width: "100%", height: "100%" }, ref: container }, (newDataSource == null ? void 0 : newDataSource.length) ? /* @__PURE__ */ import_react.default.createElement(
    import_antd.Table,
    {
      className: styles.tableContent,
      components,
      columns: newColumns,
      rowClassName: () => "editable-row",
      dataSource: newDataSource,
      bordered: true,
      scroll: { y: height - 54, x: "max-content" },
      size: "small",
      pagination: false,
      rowKey: "__index"
    }
  ) : /* @__PURE__ */ import_react.default.createElement(
    import_antd.Empty,
    {
      description: t("app_table.index.dangQianShuJuWu"),
      style: { margin: "12px 0" }
    }
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AppTable
});
