// @antv/l7-editor/lib/components/app-table/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-table/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const { colorBorder } = token;
  const { colorBgElevated, colorFillSecondary } = token;
  return {
    tableContent: import_css.css`
      *::-webkit-scrollbar {
        width: 5px;
        background-color: ${colorBgElevated};
      }
      *::-webkit-scrollbar-thumb {
        background-color: ${colorFillSecondary};
        border-radius: 4px;
      }
      *::-webkit-scrollbar-corner {
        background-color: ${colorBgElevated};
      }
    `,
    editableCell: import_css.css`
      position: relative;
    `,
    editableCellValueWrap: import_css.css`
      padding: 3px 12px;
      cursor: pointer;
      &:hover {
        padding: 2px 11px;
        border: 1px solid ${colorBorder};
        border-radius: 2px;
        cursor: pointer;
      }
    `,
    addColumns: import_css.css`
      position: relative;
    `,
    addButton: import_css.css`
      position: absolute;
      left: 0;
      top: -5px;
      padding: 4px;
    `,
    deleteColumns: import_css.css`
      position: relative;
      :hover {
        .delete {
          display: inline;
        }
      }
    `,
    delButton: import_css.css`
      display: none;
      position: absolute;
      padding: 4px;
      margin-left:5px;
      top: -5px;
    `
  };
};
var styles_default = useStyle;
