// @antv/l7-editor/lib/components/map-control-group/text-layer-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/text-layer-control/index.tsx
var text_layer_control_exports = {};
__export(text_layer_control_exports, {
  TextLayerControl: () => TextLayerControl
});
module.exports = __toCommonJS(text_layer_control_exports);
var import_larkmap = require("@antv/larkmap");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../../recoil");
var import_iconfont = require("../../iconfont");
var import_styles = __toESM(require("../styles"));
var import_style = __toESM(require("./style"));
var TextLayerControl = () => {
  const styles = (0, import_styles.default)();
  const style = (0, import_style.default)();
  const {
    showTextLayer,
    setShowTextLayer,
    textLayerFields,
    setTextLayerFields
  } = (0, import_recoil.useGlobal)();
  const { features } = (0, import_recoil.useFeature)();
  const [fields, setFields] = (0, import_react.useState)([]);
  const { t } = (0, import_react_i18next.useTranslation)();
  const refreshFields = () => {
    const newFieldSet = /* @__PURE__ */ new Set();
    features.forEach((feature) => {
      const properties = feature.properties;
      if (properties) {
        Object.keys(properties).forEach((key) => {
          newFieldSet.add(key);
        });
      }
    });
    setFields(Array.from(newFieldSet));
  };
  return /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "bottomleft" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Popover,
    {
      title: t("text-layer-control_wenBenBiaoZhu"),
      overlayStyle: { width: 300 },
      content: /* @__PURE__ */ import_react.default.createElement(import_antd.Form, { size: "small", className: style.textLayerForm }, /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { label: t("text-layer-control_shiFouZhanShiTuCeng") }, /* @__PURE__ */ import_react.default.createElement(import_antd.Switch, { value: showTextLayer, onChange: setShowTextLayer })), /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { label: t("text-layer-control_zhanShiZiDuan") }, /* @__PURE__ */ import_react.default.createElement(
        import_antd.Select,
        {
          value: textLayerFields,
          onChange: setTextLayerFields,
          placeholder: t("text-layer-control_buXuan"),
          mode: "multiple",
          options: fields.map((item) => {
            return { label: item, value: item };
          })
        }
      ))),
      trigger: "click",
      onOpenChange: (visible) => {
        if (visible) {
          refreshFields();
        }
      }
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Tooltip,
      {
        placement: "left",
        overlay: t("text-layer-control_wenBenTuCengPeiZhi")
      },
      /* @__PURE__ */ import_react.default.createElement("button", { className: styles.L7EditorControl, id: "text-layer-control" }, /* @__PURE__ */ import_react.default.createElement(import_iconfont.IconFont, { type: "icon-wenbenkuang" }))
    )
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  TextLayerControl
});
