// @antv/l7-editor/es/components/geojson-editor/editortool.js
/* eslint-disable guard-for-in */
import * as turf from '@turf/turf';
// @ts-ignore
import _ from 'lodash-es';
import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
export function provideCompletionItems(model, position, type) {
  var wordUntilPosition = model.getWordUntilPosition(position);
  var range = {
    startLineNumber: position.lineNumber,
    endLineNumber: position.lineNumber,
    startColumn: wordUntilPosition.startColumn,
    endColumn: wordUntilPosition.endColumn
  };
  var suggestions = [];
  if (type === 'turf') {
    // eslint-disable-next-line guard-for-in
    for (var key in turf) {
      var completionItem = {
        label: key,
        kind: monacoEditor.languages.CompletionItemKind.Function,
        documentation: turf[key].toString(),
        insertText: key,
        range: range
      };
      suggestions.push(completionItem);
    }
  } else {
    suggestions = _.chain(_).keys().map(function (key) {
      return {
        label: key,
        kind: monacoEditor.languages.CompletionItemKind.Function,
        insertText: key,
        range: range
      };
    }).value();
  }
  return {
    suggestions: suggestions
  };
}
