// @antv/l7-editor/lib/components/lnglat-editor/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/lnglat-editor/index.tsx
var lnglat_editor_exports = {};
__export(lnglat_editor_exports, {
  LngLatEditor: () => LngLatEditor
});
module.exports = __toCommonJS(lnglat_editor_exports);
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../recoil");
var import_utils = require("../../utils");
var { TextArea } = import_antd.Input;
var LngLatEditor = () => {
  const [input, setInput] = (0, import_react.useState)("");
  const { fc, resetFeatures, bboxAutoFit } = (0, import_recoil.useFeature)();
  const { autoFitBounds } = (0, import_recoil.useGlobal)();
  const [isFocus, setIsFocus] = (0, import_react.useState)(false);
  const { t } = (0, import_react_i18next.useTranslation)();
  (0, import_react.useEffect)(() => {
    if (isFocus) {
      return;
    }
    const result = (0, import_utils.GeoJSON2LngLat)(fc);
    if (result !== input) {
      setInput(result);
    }
  }, [fc, isFocus]);
  const { run: onInputChange } = (0, import_ahooks.useDebounceFn)(
    (newInput) => {
      const features = (0, import_utils.LngLat2GeoJson)(newInput);
      if (features) {
        resetFeatures(features);
        if (autoFitBounds) {
          bboxAutoFit(features);
        }
      }
    },
    {
      wait: 1e3,
      maxWait: 1e3
    }
  );
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(
    TextArea,
    {
      value: input,
      style: {
        margin: 8,
        height: "calc(100% - 16px)",
        width: "calc(100% - 16px)"
      },
      placeholder: t("import_btn.lnglat_import_btn.qingShuRuLianXu"),
      onFocus: () => setIsFocus(true),
      onBlur: () => setIsFocus(false),
      onChange: (e) => {
        setInput(e.target.value);
        onInputChange(e.target.value);
      }
    }
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LngLatEditor
});
