// @antv/l7-editor/es/components/app-header/base-map/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { Select, Space } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useGlobal } from "../../../recoil";
var Option = Select.Option;
export var BaseMap = function BaseMap() {
  var _useGlobal = useGlobal(),
    baseMap = _useGlobal.baseMap,
    setBaseMap = _useGlobal.setBaseMap,
    setMapOptions = _useGlobal.setMapOptions,
    mapOptions = _useGlobal.mapOptions,
    theme = _useGlobal.theme,
    setCoordConvert = _useGlobal.setCoordConvert;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var options = [{
    img: 'https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*POJeRJzjdNAAAAAAAAAAAAAADjWqAQ/original',
    label: t('btn.setting_btn.gaoDe'),
    value: 'Gaode'
  }, {
    img: 'https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*-G-sT4L_sNoAAAAAAAAAAAAADjWqAQ/original',
    label: 'Mapbox',
    value: 'Mapbox'
  }];
  var handleChange = function handleChange(e) {
    setBaseMap(e);
    setMapOptions(_objectSpread(_objectSpread({}, mapOptions), {}, {
      style: theme === 'dark' ? 'dark' : 'normal'
    }));
    setCoordConvert(e === 'Gaode' ? 'GCJ02' : 'WGS84');
    if (e !== baseMap) {
      location.reload();
    }
  };
  return /*#__PURE__*/React.createElement(Select, {
    style: {
      width: '120px'
    },
    bordered: false,
    defaultValue: baseMap,
    onChange: handleChange
  }, options.map(function (item) {
    return /*#__PURE__*/React.createElement(Option, {
      key: item.value,
      value: item.value,
      label: item.label
    }, /*#__PURE__*/React.createElement(Space, null, /*#__PURE__*/React.createElement("span", {
      style: {
        display: 'flex'
      }
    }, /*#__PURE__*/React.createElement("img", {
      src: item.img,
      alt: "",
      style: {
        height: 15,
        width: 15
      }
    })), /*#__PURE__*/React.createElement("span", null, item.label)));
  }));
};
