// @antv/l7-editor/lib/components/map-control-group/clear-control/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/clear-control/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const { colorBgContainer, boxShadow, colorIcon, colorText } = token;
  return {
    clear: import_css.css`
      width: 30px;
      height: 30px;
      line-height: 41px;
      text-align: center;
      border-radius: 2px;
      background-color: ${colorBgContainer};
      box-shadow: ${boxShadow} !important;
      cursor: pointer;
      color: ${colorIcon};
      &:hover {
        color: ${colorText};
        background: ${colorBgContainer};
      }
    `,
    clearSvg: import_css.css`
      font-size: 24px;
    `
  };
};
var styles_default = useStyle;
