// @antv/l7-editor/es/components/app-header/btn/download-btn.js
import { CloudDownloadOutlined } from '@ant-design/icons';
import { coordAll } from '@turf/turf';
import { Button, Dropdown } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';
// @ts-ignore
import tokml from 'tokml';
import { downloadText } from "../../../utils";
import { prettierText } from "../../../utils/prettier-text";
import { GeoJSON2Wkt } from "../../../utils/wkt";
import { useFeature } from "./../../../recoil";
var DownloadBtn = function DownloadBtn() {
  var _useFeature = useFeature(),
    editorText = _useFeature.editorText,
    fc = _useFeature.fc;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var onDownload = function onDownload(key) {
    if (key === 'GeoJson') {
      downloadText(JSON.stringify(fc), 'json');
    } else if (key === 'FormatGeoJson') {
      var newText = prettierText({
        content: fc
      });
      downloadText(newText, 'json');
    } else if (key === 'LngLat') {
      downloadText(coordAll(fc).map(function (lngLat) {
        return lngLat.join(',');
      }).join(';'), 'txt');
    } else if (key === 'KML') {
      var kml = tokml(fc);
      downloadText(kml, 'kml');
    } else if (key === 'WKT') {
      downloadText(GeoJSON2Wkt(fc), 'wkt');
    } else {
      downloadText(editorText, 'json');
    }
  };
  var DownloadMenuItems = [{
    key: 'GeoJson',
    label: t('app_header.constants.xiaZaiGEO')
  }, {
    key: 'LngLat',
    label: t('app_header.constants.xiaZaiLNG')
  }, {
    key: 'KML',
    label: t('app_header.constants.xiaZaiKML')
  }, {
    key: 'WKT',
    label: t('app_header.constants.xiaZaiWKT')
  }];
  return /*#__PURE__*/React.createElement(Dropdown, {
    menu: {
      items: DownloadMenuItems,
      onClick: function onClick(_ref) {
        var key = _ref.key;
        onDownload(key);
      }
    }
  }, /*#__PURE__*/React.createElement(Button, {
    id: "l7-editor-download",
    icon: /*#__PURE__*/React.createElement(CloudDownloadOutlined, null)
  }, t('btn.download_btn.xiaZai')));
};
export default DownloadBtn;
