// @antv/l7-editor/es/components/app-header/btn/handback-btn.js
import { QuestionCircleOutlined } from '@ant-design/icons';
import { Button, Dropdown } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';
export default (function () {
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var onDownload = function onDownload(key) {
    if (key === 'guide') {
      window.open('/guide');
    } else if (key === 'api') {
      window.open('/docs');
    } else if (key === 'issue') {
      window.open('https://github.com/antvis/L7Editor/issues/new');
    }
  };
  var HandBackMenuItems = [{
    key: 'guide',
    label: t('app_header.constants.pingTaiShiYongWen')
  }, {
    key: 'api',
    label: t('app_header.constants.zuJianAPI')
  }, {
    key: 'issue',
    label: t('app_header.constants.issue')
  }];
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Dropdown, {
    menu: {
      items: HandBackMenuItems,
      onClick: function onClick(_ref) {
        var key = _ref.key;
        onDownload(key);
      }
    },
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(Button, {
    icon: /*#__PURE__*/React.createElement(QuestionCircleOutlined, null)
  }, t('btn.handback_btn.bangZhu'))));
});
