// @antv/l7-editor/es/components/app-header/btn/i18n-btn.js
import { Button, Dropdown } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { LangList } from "../../../locales";
import { useGlobal } from "../../../recoil";
import { IconFont } from "../../iconfont";
import useStyle from "../styles";
var items = LangList.map(function (item) {
  return {
    key: item.lang,
    label: item.name
  };
});
var I18nBtn = function I18nBtn() {
  var _useTranslation = useTranslation(),
    i18n = _useTranslation.i18n;
  var _useGlobal = useGlobal(),
    locale = _useGlobal.locale,
    setLocale = _useGlobal.setLocale;
  var styles = useStyle();
  var onLocaleChange = function onLocaleChange(_ref) {
    var key = _ref.key;
    i18n.changeLanguage(key);
    setLocale(key);
  };
  return /*#__PURE__*/React.createElement(Dropdown, {
    menu: {
      items: items,
      onClick: onLocaleChange,
      selectedKeys: [locale],
      selectable: true
    },
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(Button, {
    className: styles.locale,
    icon: /*#__PURE__*/React.createElement(IconFont, {
      className: styles.localeIcon,
      type: "icon-in"
    })
  }));
};
export default I18nBtn;
