// @antv/l7-editor/es/components/app-header/btn/setting-btn.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { SettingOutlined } from '@ant-design/icons';
import { Button, Form, Modal, Radio, Switch, Tooltip } from 'antd';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useGlobal } from "../../../recoil";
export var SettingBtn = function SettingBtn() {
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isModalOpen = _useState2[0],
    setIsModalOpen = _useState2[1];
  var _useGlobal = useGlobal(),
    popupTrigger = _useGlobal.popupTrigger,
    setPopupTrigger = _useGlobal.setPopupTrigger,
    autoFitBounds = _useGlobal.autoFitBounds,
    setAutoFitBounds = _useGlobal.setAutoFitBounds,
    showDrawDistance = _useGlobal.showDrawDistance,
    setShowDrawDistance = _useGlobal.setShowDrawDistance,
    showDrawArea = _useGlobal.showDrawArea,
    setShowDrawArea = _useGlobal.setShowDrawArea;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var _Form$useForm = Form.useForm(),
    _Form$useForm2 = _slicedToArray(_Form$useForm, 1),
    form = _Form$useForm2[0];
  var showModal = function showModal() {
    setIsModalOpen(true);
  };
  var handleOk = function handleOk() {
    form.submit();
  };
  var handleCancel = function handleCancel() {
    setIsModalOpen(false);
    form.resetFields();
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Tooltip, {
    title: t('btn.setting_btn.diTuSheZhi'),
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(Button, {
    id: "l7-editor-set",
    icon: /*#__PURE__*/React.createElement(SettingOutlined, null),
    onClick: showModal
  }, t('btn.setting_btn.sheZhi'))), /*#__PURE__*/React.createElement(Modal, {
    title: t('btn.setting_btn.sheZhi'),
    open: isModalOpen,
    onOk: handleOk /*  */,
    onCancel: handleCancel,
    destroyOnClose: true,
    okText: t('btn.setting_btn.queRen'),
    cancelText: t('btn.setting_btn.quXiao')
  }, /*#__PURE__*/React.createElement(Form, {
    form: form,
    initialValues: {
      popupTrigger: popupTrigger,
      autoFitBounds: autoFitBounds,
      showDrawDistance: showDrawDistance,
      showDrawArea: showDrawArea
    },
    style: {
      textAlign: 'right'
    },
    onFinish: function onFinish(e) {
      setIsModalOpen(false);
      setPopupTrigger(e.popupTrigger);
      setAutoFitBounds(e.autoFitBounds);
      setShowDrawDistance(e.showDrawDistance);
      setShowDrawArea(e.showDrawArea);
    }
  }, /*#__PURE__*/React.createElement(Form.Item, {
    name: "popupTrigger",
    label: t('btn.setting_btn.tuCengQiPaoZhan')
  }, /*#__PURE__*/React.createElement(Radio.Group, null, /*#__PURE__*/React.createElement(Radio.Button, {
    value: 'click'
  }, t('btn.setting_btn.dianJi')), /*#__PURE__*/React.createElement(Radio.Button, {
    value: 'hover'
  }, t('btn.setting_btn.huaRu')))), /*#__PURE__*/React.createElement(Form.Item, {
    name: "autoFitBounds",
    valuePropName: "checked",
    label: t('btn.setting_btn.ziDongSuoFangZhi')
  }, /*#__PURE__*/React.createElement(Switch, null)), /*#__PURE__*/React.createElement(Form.Item, {
    name: "showDrawDistance",
    valuePropName: "checked",
    label: t('btn.setting_btn.juLi')
  }, /*#__PURE__*/React.createElement(Switch, null)), /*#__PURE__*/React.createElement(Form.Item, {
    name: "showDrawArea",
    valuePropName: "checked",
    label: t('btn.setting_btn.mianJi')
  }, /*#__PURE__*/React.createElement(Switch, null)))));
};
