// @antv/l7-editor/es/constants/index.js
export var RightPanelWidthRange = [20, 80];
export var LocalStorageKey = /*#__PURE__*/function (LocalStorageKey) {
  LocalStorageKey["RightPanelWidth"] = "RightPanelWidth";
  LocalStorageKey["MapOptions"] = "MapOptions";
  LocalStorageKey["LayerColor"] = "LayerColor";
  LocalStorageKey["HideEditor"] = "HideEditor";
  LocalStorageKey["EditorText"] = "EditorText";
  LocalStorageKey["PopupTrigger"] = "PopupTrigger";
  LocalStorageKey["ActiveRightTabKey"] = "ActiveRightTabKey";
  LocalStorageKey["AutoFitBounds"] = "AutoFitBounds";
  LocalStorageKey["BaseMap"] = "BaseMap";
  LocalStorageKey["officialLayers"] = "officialLayers";
  LocalStorageKey["WktText"] = "WktText";
  LocalStorageKey["Convert"] = "Convert";
  LocalStorageKey["theme"] = "theme";
  LocalStorageKey["cityHistory"] = "cityHistory";
  LocalStorageKey["showTextLayer"] = "showTextLayer";
  LocalStorageKey["textLayerFields"] = "textLayerFields";
  LocalStorageKey["locale"] = "locale";
  LocalStorageKey["firstOpening"] = "firstOpening";
  LocalStorageKey["wasmPath"] = "wasmPath";
  LocalStorageKey["customTiles"] = "customTiles";
  LocalStorageKey["showDrawDistance"] = "showDrawDistance";
  LocalStorageKey["showDrawArea"] = "showDrawArea";
  return LocalStorageKey;
}({});
export var LayerId = /*#__PURE__*/function (LayerId) {
  LayerId["PointLayer"] = "PointLayer";
  LayerId["LineLayer"] = "LineLayer";
  LayerId["PolygonLayer"] = "PolygonLayer";
  return LayerId;
}({});
export * from "./variable-type";
export var dingGroupImg = 'https://mdn.alipayobjects.com/huamei_baaa7a/afts/img/A*mvwjR5RwS9QAAAAAAAAAAAAADqSCAQ/original';
export var PrimaryColor = '#1677ff';
export var FeatureKey = {
  Index: Symbol('index'),
  DrawType: Symbol('drawType'),
  IsEdit: Symbol('isEdit')
};
export var MapBoxConfig = {
  token: 'pk.eyJ1IjoibGl1dmlnb25nenVvc2hpIiwiYSI6ImNsaGs2czBrcTBvczUzbnFzOHU0dzk2ZWQifQ.hVvTgcbg_Ym-VQz36psLRg',
  WebGLParams: {
    preserveDrawingBuffer: true
  }
};
export var GaodeConfig = {
  token: '5ae4492ef912cbbc93034fea0e66ff2a',
  WebGLParams: {
    preserveDrawingBuffer: true
  }
};
export var LayerZIndex = 100;
export var GOOGLE_TILE_MAP_URL = 'https://gac-geo.googlecnapps.cn/maps/vt?lyrs=s&gl=CN&x={x}&y={y}&z={z}';
export var GOOGLE_TILE_MAP_ROUTER_URL = 'https://gac-geo.googlecnapps.cn/maps/vt?lyrs=h&gl=CN&x={x}&y={y}&z={z}';
export var OfficeLayerEnum = /*#__PURE__*/function (OfficeLayerEnum) {
  OfficeLayerEnum["GoogleSatellite"] = "googleSatellite";
  OfficeLayerEnum["VectorMap"] = "vectorMap";
  return OfficeLayerEnum;
}({});
