// @antv/l7-editor/es/pages/index.js
import { Result } from 'antd';
import React, { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { RecoilEnv, RecoilRoot } from 'recoil';
import { PrimaryColor } from "../constants";
import "../constants/iconfont.js";
import "../locales/index";
import { activeTabState, autoFitBoundsState, baseMapState, convertState, customTilesState, hideEditorState, layerColorState, localeState, mapOptionState, officialLayersState, popupTriggerState, rightWidthState, showDrawAreaState, showDrawDistanceState, showTextLayerState, textLayerFieldsState, themeState, wasmPathState } from "../recoil/atomState";
import { Editor } from "./components/editor";
RecoilEnv.RECOIL_DUPLICATE_ATOM_KEY_CHECKING_ENABLED = false;
export var L7Editor = function L7Editor(props) {
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var isPc = useMemo(function () {
    return !/Mobi|Android|iPhone/i.test(navigator.userAgent);
  }, []);
  var initializeState = useMemo(function () {
    return function (_ref) {
      var _props$baseMap, _props$primaryColor, _props$rightPanelWidt, _props$mapOption, _props$popupTrigger, _props$activeTab, _props$officialLayers, _props$hidePanel, _props$theme, _props$coordConvert, _props$locale, _props$wasmPath, _props$showTextLayer, _props$textLayerField, _props$showDrawDistan, _props$showDrawArea, _props$customTiles;
      var set = _ref.set;
      set(baseMapState, (_props$baseMap = props === null || props === void 0 ? void 0 : props.baseMap) !== null && _props$baseMap !== void 0 ? _props$baseMap : 'Gaode');
      set(layerColorState, (_props$primaryColor = props === null || props === void 0 ? void 0 : props.primaryColor) !== null && _props$primaryColor !== void 0 ? _props$primaryColor : PrimaryColor);
      set(rightWidthState, (_props$rightPanelWidt = props === null || props === void 0 ? void 0 : props.rightPanelWidth) !== null && _props$rightPanelWidt !== void 0 ? _props$rightPanelWidt : 40);
      set(mapOptionState, (_props$mapOption = props === null || props === void 0 ? void 0 : props.mapOption) !== null && _props$mapOption !== void 0 ? _props$mapOption : {
        style: 'normal',
        maxZoom: 24
      });
      set(autoFitBoundsState, !(props !== null && props !== void 0 && props.autoFitBounds));
      set(popupTriggerState, (_props$popupTrigger = props === null || props === void 0 ? void 0 : props.popupTrigger) !== null && _props$popupTrigger !== void 0 ? _props$popupTrigger : 'click');
      set(activeTabState, (_props$activeTab = props === null || props === void 0 ? void 0 : props.activeTab) !== null && _props$activeTab !== void 0 ? _props$activeTab : 'geojson');
      set(officialLayersState, (_props$officialLayers = props === null || props === void 0 ? void 0 : props.officialLayers) !== null && _props$officialLayers !== void 0 ? _props$officialLayers : []);
      set(hideEditorState, (_props$hidePanel = props === null || props === void 0 ? void 0 : props.hidePanel) !== null && _props$hidePanel !== void 0 ? _props$hidePanel : false);
      set(themeState, (_props$theme = props === null || props === void 0 ? void 0 : props.theme) !== null && _props$theme !== void 0 ? _props$theme : 'light');
      set(convertState, (_props$coordConvert = props === null || props === void 0 ? void 0 : props.coordConvert) !== null && _props$coordConvert !== void 0 ? _props$coordConvert : 'GCJ02');
      set(localeState, (_props$locale = props === null || props === void 0 ? void 0 : props.locale) !== null && _props$locale !== void 0 ? _props$locale : 'zh-CN');
      set(wasmPathState, (_props$wasmPath = props === null || props === void 0 ? void 0 : props.wasmPath) !== null && _props$wasmPath !== void 0 ? _props$wasmPath : '/');
      set(showTextLayerState, (_props$showTextLayer = props === null || props === void 0 ? void 0 : props.showTextLayer) !== null && _props$showTextLayer !== void 0 ? _props$showTextLayer : false);
      set(textLayerFieldsState, (_props$textLayerField = props === null || props === void 0 ? void 0 : props.textLayerFields) !== null && _props$textLayerField !== void 0 ? _props$textLayerField : undefined);
      set(showDrawDistanceState, (_props$showDrawDistan = props.showDrawDistance) !== null && _props$showDrawDistan !== void 0 ? _props$showDrawDistan : false);
      set(showDrawAreaState, (_props$showDrawArea = props.showDrawArea) !== null && _props$showDrawArea !== void 0 ? _props$showDrawArea : false);
      set(customTilesState, (_props$customTiles = props.customTiles) !== null && _props$customTiles !== void 0 ? _props$customTiles : []);
    };
  }, [props]);
  return isPc ? /*#__PURE__*/React.createElement(RecoilRoot, {
    initializeState: initializeState
  }, /*#__PURE__*/React.createElement(Editor, props)) : /*#__PURE__*/React.createElement(Result, {
    status: "404",
    title: t('pages.index.qingYongPCDuan')
  });
};
