// variable-type/lib/index.js
"use strict";
/**
 * Created by hustcc on 17/08/01.
 *
 * 👏 08-01 is my birthday on ID card.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var Type_1 = __importDefault(require("./Type"));
var and_1 = __importDefault(require("./types/and"));
var apply_1 = __importDefault(require("./types/apply"));
var arrayOf_1 = __importDefault(require("./types/arrayOf"));
var oneOf_1 = __importDefault(require("./types/oneOf"));
var instanceOf_1 = __importDefault(require("./types/instanceOf"));
var not_1 = __importDefault(require("./types/not"));
var or_1 = __importDefault(require("./types/or"));
var shape_1 = __importDefault(require("./types/shape"));
var typeOf_1 = __importDefault(require("./types/typeOf"));
var VT = {
    undefined: typeOf_1.default('undefined'),
    bool: typeOf_1.default('boolean'),
    func: typeOf_1.default('function'),
    number: typeOf_1.default('number'),
    string: typeOf_1.default('string'),
    null: typeOf_1.default('null'),
    object: typeOf_1.default('object'),
    array: typeOf_1.default('array'),
    and: and_1.default,
    or: or_1.default,
    not: not_1.default,
    any: new Type_1.default(function () { return true; }),
    instanceOf: instanceOf_1.default,
    typeOf: typeOf_1.default,
    in: oneOf_1.default,
    oneOf: oneOf_1.default,
    oneOfType: or_1.default,
    arrayOf: arrayOf_1.default,
    shape: shape_1.default,
    apply: apply_1.default
};
exports.default = VT;
