// @antv/l7-editor/es/components/map-control-group/text-layer-control/index.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { CustomControl } from '@antv/larkmap';
import { Form, Popover, Select, Switch, Tooltip } from 'antd';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useFeature, useGlobal } from "../../../recoil";
import { IconFont } from "../../iconfont";
import useStyles from "../styles";
import useStyle from "./style";
export var TextLayerControl = function TextLayerControl() {
  var styles = useStyles();
  var style = useStyle();
  var _useGlobal = useGlobal(),
    showTextLayer = _useGlobal.showTextLayer,
    setShowTextLayer = _useGlobal.setShowTextLayer,
    textLayerFields = _useGlobal.textLayerFields,
    setTextLayerFields = _useGlobal.setTextLayerFields;
  var _useFeature = useFeature(),
    features = _useFeature.features;
  var _useState = useState([]),
    _useState2 = _slicedToArray(_useState, 2),
    fields = _useState2[0],
    setFields = _useState2[1];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var refreshFields = function refreshFields() {
    var newFieldSet = new Set();
    features.forEach(function (feature) {
      var properties = feature.properties;
      if (properties) {
        Object.keys(properties).forEach(function (key) {
          newFieldSet.add(key);
        });
      }
    });
    setFields(Array.from(newFieldSet));
  };
  return /*#__PURE__*/React.createElement(CustomControl, {
    position: "bottomleft"
  }, /*#__PURE__*/React.createElement(Popover, {
    title: t('text-layer-control_wenBenBiaoZhu'),
    overlayStyle: {
      width: 300
    },
    content: /*#__PURE__*/React.createElement(Form, {
      size: "small",
      className: style.textLayerForm
    }, /*#__PURE__*/React.createElement(Form.Item, {
      label: t('text-layer-control_shiFouZhanShiTuCeng')
    }, /*#__PURE__*/React.createElement(Switch, {
      value: showTextLayer,
      onChange: setShowTextLayer
    })), /*#__PURE__*/React.createElement(Form.Item, {
      label: t('text-layer-control_zhanShiZiDuan')
    }, /*#__PURE__*/React.createElement(Select, {
      value: textLayerFields,
      onChange: setTextLayerFields,
      placeholder: t('text-layer-control_buXuan'),
      mode: "multiple",
      options: fields.map(function (item) {
        return {
          label: item,
          value: item
        };
      })
    }))),
    trigger: "click",
    onOpenChange: function onOpenChange(visible) {
      if (visible) {
        refreshFields();
      }
    }
  }, /*#__PURE__*/React.createElement(Tooltip, {
    placement: "left",
    overlay: t('text-layer-control_wenBenTuCengPeiZhi')
  }, /*#__PURE__*/React.createElement("button", {
    className: styles.L7EditorControl,
    id: "text-layer-control"
  }, /*#__PURE__*/React.createElement(IconFont, {
    type: "icon-wenbenkuang"
  })))));
};
