// @antv/l7-editor/lib/components/map-control-group/filter-form-list-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/filter-form-list-control/index.tsx
var filter_form_list_control_exports = {};
__export(filter_form_list_control_exports, {
  default: () => filter_form_list_control_default
});
module.exports = __toCommonJS(filter_form_list_control_exports);
var import_icons = require("@ant-design/icons");
var import_larkmap = require("@antv/larkmap");
var import_antd = require("antd");
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_use_filter_features = require("../../../hooks/use-filter-features");
var import_recoil = require("../../../recoil");
var import_styles = __toESM(require("../styles"));
var import_numberFilter = __toESM(require("./numberFilter"));
var import_stringFilter = __toESM(require("./stringFilter"));
var import_styles2 = __toESM(require("./styles"));
var { Option } = import_antd.Select;
var FilterControl = () => {
  const { dataSource } = (0, import_recoil.useFeature)();
  const { setFilters, filters } = (0, import_recoil.useFilter)();
  const [isVisible, setIsVisible] = (0, import_react.useState)(false);
  const { t } = (0, import_react_i18next.useTranslation)();
  const [form] = import_antd.Form.useForm();
  const styles = (0, import_styles2.default)();
  const style = (0, import_styles.default)();
  const addVisible = (0, import_react.useMemo)(() => {
    return !dataSource.length;
  }, [dataSource]);
  const onValuesChange = (_, all) => {
    if ((0, import_lodash_es.isEmpty)(all.filterFromList)) {
      setFilters([]);
      return;
    }
    const newValue = all.filterFromList.filter((item) => item).map((item) => {
      const { logic, operator } = item;
      if (item.field) {
        const { field, type } = JSON.parse(item.field || "");
        if (item.operator === "BETWEEN") {
          if (item.min && item.max) {
            return {
              logic,
              field,
              type,
              operator,
              value: [item.min, item.max]
            };
          }
        }
        if ((item.operator === "IN" || item.operator === "NO_IN") && item.value) {
          return {
            ...item,
            field,
            type,
            value: item.value.length ? item.value : void 0
          };
        }
        return { ...item, field, type };
      } else {
        return { ...item, field: void 0, type: void 0 };
      }
    });
    setFilters(newValue);
  };
  const isFilterActive = (0, import_react.useMemo)(() => {
    return filters.filter((item) => !(0, import_use_filter_features.isEmptyFilter)(item)).find((item) => item);
  }, [filters]);
  return /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.CustomControl,
    {
      position: "bottomright",
      className: styles.l7FilterPanel,
      style: { display: "flex" }
    },
    /* @__PURE__ */ import_react.default.createElement(
      "div",
      {
        className: styles.l7Filter,
        style: { display: isVisible ? "block" : "none" }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Form,
        {
          style: { width: "100%" },
          form,
          onValuesChange: (0, import_lodash_es.debounce)(onValuesChange, 500)
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Form.List, { name: "filterFromList" }, (fields, { add, remove }) => /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, fields.map(({ name }, index) => {
          var _a, _b;
          return /* @__PURE__ */ import_react.default.createElement(
            "div",
            {
              key: name,
              style: { display: "flex", marginBottom: 8 }
            },
            /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: [name, "logic"], initialValue: "and" }, /* @__PURE__ */ import_react.default.createElement(import_antd.Select, { style: { width: 70, marginRight: "8px" } }, /* @__PURE__ */ import_react.default.createElement(Option, { value: "and" }, t("filter_form_list_control.index.bingQie")), /* @__PURE__ */ import_react.default.createElement(Option, { value: "or" }, t("filter_form_list_control.index.huoZhe")))),
            /* @__PURE__ */ import_react.default.createElement(
              import_antd.Form.Item,
              {
                name: [name, "field"],
                initialValue: JSON.stringify({
                  field: (_a = dataSource[0]) == null ? void 0 : _a.field,
                  type: (_b = dataSource[0]) == null ? void 0 : _b.type
                })
              },
              /* @__PURE__ */ import_react.default.createElement(
                import_antd.Select,
                {
                  style: { width: 130, marginRight: "8px" },
                  placeholder: t(
                    "filter_form_list_control.index.qingXuanZeZiDuan"
                  ),
                  onChange: () => {
                    const newFilterFromList = (0, import_lodash_es.cloneDeep)(
                      form.getFieldValue("filterFromList")
                    );
                    newFilterFromList.forEach((v, i) => {
                      if (index === i) {
                        v.operator = void 0;
                        v.value = void 0;
                      }
                    });
                    form.setFieldValue(
                      "filterFromList",
                      newFilterFromList
                    );
                    setFilters(
                      newFilterFromList.map((item) => {
                        const { field, type } = JSON.parse(item.field);
                        return { ...item, field, type };
                      })
                    );
                  }
                },
                dataSource.map(({ field, type }) => {
                  return /* @__PURE__ */ import_react.default.createElement(
                    Option,
                    {
                      key: field,
                      value: JSON.stringify({ field, type })
                    },
                    /* @__PURE__ */ import_react.default.createElement(
                      "i",
                      {
                        style: {
                          fontSize: 20,
                          marginRight: 8,
                          color: "#999"
                        }
                      },
                      type === "number" ? /* @__PURE__ */ import_react.default.createElement(import_icons.FieldBinaryOutlined, null) : /* @__PURE__ */ import_react.default.createElement(import_icons.FieldStringOutlined, null)
                    ),
                    /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: field }, field)
                  );
                })
              )
            ),
            /* @__PURE__ */ import_react.default.createElement(
              import_antd.Form.Item,
              {
                shouldUpdate: (prevValues, curValues) => prevValues.field === curValues.field,
                style: { margin: 0 }
              },
              ({ getFieldsValue }) => {
                const { filterFromList } = getFieldsValue();
                if (filterFromList[index].field) {
                  const { type } = JSON.parse(
                    filterFromList[index].field
                  );
                  if (type === "number") {
                    return /* @__PURE__ */ import_react.default.createElement(
                      import_numberFilter.default,
                      {
                        name,
                        index,
                        form
                      }
                    );
                  }
                  return /* @__PURE__ */ import_react.default.createElement(
                    import_stringFilter.default,
                    {
                      name,
                      index,
                      form
                    }
                  );
                }
              }
            ),
            /* @__PURE__ */ import_react.default.createElement(
              import_antd.Button,
              {
                type: "text",
                onClick: () => {
                  remove(name);
                  setFilters(form.getFieldValue("filterFromList"));
                },
                icon: /* @__PURE__ */ import_react.default.createElement(import_icons.DeleteOutlined, null)
              }
            )
          );
        }), /* @__PURE__ */ import_react.default.createElement(
          import_antd.Button,
          {
            type: "dashed",
            disabled: addVisible,
            onClick: () => {
              add();
            },
            block: true,
            icon: /* @__PURE__ */ import_react.default.createElement(import_icons.PlusOutlined, null),
            style: { width: 500 }
          },
          t("filter_form_list_control.index.tianJiaShaiXuanTiao")
        )))
      )
    ),
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Tooltip,
      {
        title: t("filter_form_list_control.index.shuJuGuoLu"),
        placement: "left"
      },
      /* @__PURE__ */ import_react.default.createElement(
        "button",
        {
          type: "button",
          id: "l7-editor-filter",
          className: style.L7EditorControl,
          onClick: () => {
            setIsVisible(!isVisible);
          }
        },
        !isFilterActive ? /* @__PURE__ */ import_react.default.createElement(
          import_icons.FilterOutlined,
          {
            style: {
              lineHeight: "30px"
            }
          }
        ) : /* @__PURE__ */ import_react.default.createElement(
          import_icons.FilterFilled,
          {
            style: {
              color: "#597ef7"
            }
          }
        )
      )
    )
  );
};
var filter_form_list_control_default = FilterControl;
