// @antv/l7-editor/lib/components/map-control-group/filter-form-list-control/numberFilter.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/filter-form-list-control/numberFilter.tsx
var numberFilter_exports = {};
__export(numberFilter_exports, {
  default: () => numberFilter_default
});
module.exports = __toCommonJS(numberFilter_exports);
var import_antd = require("antd");
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../../recoil");
var NumberFilter = ({ name, index, form }) => {
  const { setFilters } = (0, import_recoil.useFilter)();
  const { dataSource } = (0, import_recoil.useFeature)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const select = [
    { label: ">", value: ">" },
    { label: ">=", value: ">=" },
    { label: "=", value: "=" },
    { label: "<=", value: "<=" },
    { label: "<", value: "<" },
    {
      label: t("filter_form_list_control.numberFilter.quJian"),
      value: "BETWEEN"
    }
  ];
  return /* @__PURE__ */ import_react.default.createElement("div", { style: { display: "flex" } }, /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: [name, "operator"] }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Select,
    {
      style: { width: "100px", marginRight: "8px" },
      placeholder: t(
        "filter_form_list_control.numberFilter.qingXuanZeGuoLu"
      ),
      options: select,
      onChange: () => {
        const newFilterFromList = (0, import_lodash_es.cloneDeep)(
          form.getFieldValue("filterFromList")
        );
        newFilterFromList.forEach((v, i) => {
          if (index === i) {
            v.value = void 0;
          }
        });
        form.setFieldValue("filterFromList", newFilterFromList);
        setFilters(
          newFilterFromList.map((item) => {
            const { field, type } = JSON.parse(item.field);
            return { ...item, field, type };
          })
        );
      }
    }
  )), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      style: { width: 150, marginBottom: 0 },
      shouldUpdate: (prevValues, curValues) => prevValues.operator === curValues.operator
    },
    ({ getFieldsValue }) => {
      var _a;
      const { filterFromList } = getFieldsValue();
      const fieldValue = (_a = JSON.parse(filterFromList[index].field)) == null ? void 0 : _a.field;
      const DataList = dataSource.find(
        (item) => (item == null ? void 0 : item.field) === fieldValue
      );
      if (filterFromList[index].operator === "BETWEEN") {
        return /* @__PURE__ */ import_react.default.createElement("div", null, /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: [name, "min"], style: { width: "70px" } }, /* @__PURE__ */ import_react.default.createElement(
          import_antd.InputNumber,
          {
            placeholder: t(
              "filter_form_list_control.numberFilter.qingShuRuShaiXuan"
            ),
            style: { width: "100%" },
            min: (DataList == null ? void 0 : DataList.min) ?? 0,
            max: (DataList == null ? void 0 : DataList.max) ?? 0
          }
        )), /* @__PURE__ */ import_react.default.createElement("span", null, " - "), /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: [name, "max"], style: { width: "70px" } }, /* @__PURE__ */ import_react.default.createElement(
          import_antd.InputNumber,
          {
            placeholder: t(
              "filter_form_list_control.numberFilter.qingShuRuShaiXuan"
            ),
            style: { width: "100%" },
            min: (DataList == null ? void 0 : DataList.min) ?? 0,
            max: (DataList == null ? void 0 : DataList.max) ?? 0
          }
        )));
      }
      return /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: [name, "value"] }, /* @__PURE__ */ import_react.default.createElement(
        import_antd.InputNumber,
        {
          placeholder: t(
            "filter_form_list_control.numberFilter.qingShuRuShaiXuan"
          ),
          style: { width: "100%" },
          min: (DataList == null ? void 0 : DataList.min) ?? 0,
          max: (DataList == null ? void 0 : DataList.max) ?? 0
        }
      ));
    }
  ));
};
var numberFilter_default = NumberFilter;
