// @antv/l7-editor/lib/components/map-control-group/filter-form-list-control/stringFilter.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/filter-form-list-control/stringFilter.tsx
var stringFilter_exports = {};
__export(stringFilter_exports, {
  default: () => stringFilter_default
});
module.exports = __toCommonJS(stringFilter_exports);
var import_antd = require("antd");
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_feature = __toESM(require("../../../recoil/feature"));
var import_filter = __toESM(require("../../../recoil/filter"));
var StringFilter = ({ name, index, form }) => {
  const { setFilters } = (0, import_filter.default)();
  const { dataSource } = (0, import_feature.default)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const select = [
    { label: t("filter_form_list_control.stringFilter.baoHan"), value: "IN" },
    {
      label: t("filter_form_list_control.stringFilter.buBaoHan"),
      value: "NOT_IN"
    },
    { label: t("filter_form_list_control.stringFilter.piPei"), value: "LIKE" },
    {
      label: t("filter_form_list_control.stringFilter.buPiPei"),
      value: "NOT_LIKE"
    }
  ];
  return /* @__PURE__ */ import_react.default.createElement("div", { style: { display: "flex" } }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      name: [name, "operator"],
      style: { width: "100px", marginRight: "8px" }
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Select,
      {
        placeholder: t(
          "filter_form_list_control.numberFilter.qingXuanZeGuoLu"
        ),
        options: select,
        onChange: () => {
          const newFilterFromList = (0, import_lodash_es.cloneDeep)(
            form.getFieldValue("filterFromList")
          );
          newFilterFromList.forEach((v, i) => {
            if (index === i) {
              v.value = void 0;
            }
          });
          form.setFieldValue("filterFromList", newFilterFromList);
          setFilters(
            newFilterFromList.map((item) => {
              const { field, type } = JSON.parse(item.field);
              return { ...item, field, type };
            })
          );
        }
      }
    )
  ), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Form.Item,
    {
      shouldUpdate: (prevValues, curValues) => prevValues.operator !== curValues.operator,
      style: { width: 150, margin: 0 }
    },
    ({ getFieldsValue }) => {
      var _a;
      const { filterFromList } = getFieldsValue();
      const fieldValue = (_a = JSON.parse(filterFromList[index].field)) == null ? void 0 : _a.field;
      const DataList = dataSource.find(
        (item) => (item == null ? void 0 : item.field) === fieldValue
      );
      const fieldType = filterFromList[index].operator;
      if (fieldType === "LIKE" || fieldType === "NOT_LIKE") {
        return /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: [name, "value"] }, /* @__PURE__ */ import_react.default.createElement(
          import_antd.Input,
          {
            placeholder: t(
              "filter_form_list_control.numberFilter.qingShuRuShaiXuan"
            ),
            style: { width: "100%" },
            allowClear: true
          }
        ));
      }
      return /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { name: [name, "value"] }, /* @__PURE__ */ import_react.default.createElement(
        import_antd.Select,
        {
          placeholder: t(
            "filter_form_list_control.stringFilter.qingXuanZeShaiXuan"
          ),
          style: { width: "100%" },
          mode: "multiple",
          maxTagCount: 1
        },
        ((0, import_lodash_es.uniq)(DataList == null ? void 0 : DataList.value) ?? []).filter((item) => item !== "undefined").map((item) => /* @__PURE__ */ import_react.default.createElement(import_antd.Select.Option, { value: item, key: item }, /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { title: item }, item)))
      ));
    }
  ));
};
var stringFilter_default = StringFilter;
