// @antv/l7-editor/lib/components/text-layer/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/text-layer/index.tsx
var text_layer_exports = {};
__export(text_layer_exports, {
  EditorTextLayer: () => EditorTextLayer
});
module.exports = __toCommonJS(text_layer_exports);
var import_larkmap = require("@antv/larkmap");
var import_turf = require("@turf/turf");
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_constants = require("../../constants");
var import_hooks = require("../../hooks");
var import_recoil = require("../../recoil");
var import_utils = require("../../utils");
var EditorTextLayer = () => {
  const { transformCoord } = (0, import_recoil.useFeature)();
  const { features: newFeatures } = (0, import_hooks.useFilterFeatures)();
  const { layerColor, textLayerFields } = (0, import_recoil.useGlobal)();
  const textOptions = (0, import_react.useMemo)(() => {
    return {
      zIndex: 101,
      field: "text",
      style: {
        fill: `${layerColor}`,
        opacity: 1,
        fontSize: 16,
        stroke: "#fff",
        strokeWidth: 2,
        textAllowOverlap: true,
        padding: [10, 10]
      }
    };
  }, [layerColor]);
  const pointTextOptions = (0, import_react.useMemo)(() => {
    const newLayerOptions = (0, import_lodash_es.cloneDeep)(textOptions);
    newLayerOptions.style.textOffset = [0, -20];
    return newLayerOptions;
  }, [textOptions]);
  const sourceData = (0, import_react.useMemo)(() => {
    const data = transformCoord(newFeatures).map((item) => {
      var _a;
      const featureIndex = (_a = item.properties) == null ? void 0 : _a[import_constants.FeatureKey.Index];
      const [x, y] = (() => {
        if (item.geometry.type === "LineString") {
          return (0, import_utils.centerOfLine)(item).geometry.coordinates;
        }
        if (item.geometry.type === "Point") {
          return item.geometry.coordinates;
        } else {
          return (0, import_turf.center)(item).geometry.coordinates;
        }
      })();
      let text = `${featureIndex + 1}`;
      if (textLayerFields == null ? void 0 : textLayerFields.length) {
        text = textLayerFields.map((field) => {
          var _a2;
          return String(((_a2 = item.properties) == null ? void 0 : _a2[field]) ?? "");
        }).join(" ");
      }
      return {
        x,
        y,
        text,
        type: item.geometry.type
      };
    });
    return data;
  }, [transformCoord, newFeatures, textLayerFields]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.TextLayer,
    {
      ...textOptions,
      source: {
        data: sourceData.filter((item) => item.type !== "Point"),
        parser: { type: "json", x: "x", y: "y" }
      }
    }
  ), /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.TextLayer,
    {
      ...pointTextOptions,
      source: {
        data: sourceData.filter((item) => item.type === "Point"),
        parser: { type: "json", x: "x", y: "y" }
      }
    }
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EditorTextLayer
});
