// @antv/l7-editor/lib/components/layer-list/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/layer-list/index.tsx
var layer_list_exports = {};
__export(layer_list_exports, {
  LayerList: () => LayerList
});
module.exports = __toCommonJS(layer_list_exports);
var import_larkmap = require("@antv/larkmap");
var import_ahooks = require("ahooks");
var import_color = __toESM(require("color"));
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_constants = require("../../constants");
var import_hooks = require("../../hooks");
var import_recoil = require("../../recoil");
var import_change_image_color = require("../../utils/change-image-color");
var import_text_layer = require("../text-layer");
var LayerList = () => {
  const scene = (0, import_larkmap.useScene)();
  const [isMounted, setIsMounted] = (0, import_react.useState)(false);
  const { layerColor, coordConvert, baseMap, showTextLayer } = (0, import_recoil.useGlobal)();
  const { transformCoord } = (0, import_recoil.useFeature)();
  const { features: newFeatures } = (0, import_hooks.useFilterFeatures)();
  const [features, setFeatures] = (0, import_react.useState)([]);
  (0, import_react.useEffect)(() => {
    if (newFeatures.length) {
      setFeatures(transformCoord(newFeatures));
    } else {
      setFeatures([]);
    }
  }, [newFeatures, coordConvert, baseMap]);
  const [
    pointSource,
    lineSource,
    polygonSource
  ] = (0, import_react.useMemo)(() => {
    const {
      Polygon: polygonList = [],
      LineString: lineStringList = [],
      Point: pointList = []
    } = (0, import_lodash_es.groupBy)(
      (0, import_lodash_es.cloneDeep)(features).filter((feature) => {
        var _a;
        return !((_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.IsEdit]);
      }),
      (feature) => feature.geometry.type.replace("Multi", "")
    );
    return [pointList, lineStringList, polygonList].map((typeFeatures) => {
      return {
        data: { type: "FeatureCollection", features: typeFeatures }
      };
    });
  }, [features]);
  (0, import_ahooks.useAsyncEffect)(async () => {
    const newLayerColor = (0, import_color.default)(layerColor).rgb().object();
    const imag2color = await (0, import_change_image_color.getPointImage)(newLayerColor, { x: 100, y: 100 });
    const imagColor = await (0, import_change_image_color.getPointImage)(newLayerColor, { x: 400, y: 400 });
    scene.addImage("drawImg", imagColor);
    scene.addImage("pointIcon", imag2color);
    setIsMounted(true);
  }, [layerColor]);
  const activeColor = (0, import_react.useMemo)(() => {
    const newLayerColor = (0, import_color.default)(layerColor).darken(0.3).hex();
    return newLayerColor;
  }, [layerColor]);
  return isMounted ? /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.PolygonLayer,
    {
      id: import_constants.LayerId.PolygonLayer,
      source: polygonSource,
      blend: "normal",
      shape: "fill",
      color: layerColor,
      style: { opacity: 0.15 },
      state: { active: { color: activeColor } },
      zIndex: import_constants.LayerZIndex
    }
  ), /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.PolygonLayer,
    {
      source: polygonSource,
      shape: "line",
      blend: "normal",
      color: layerColor,
      size: 2,
      zIndex: import_constants.LayerZIndex
    }
  ), /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.LineLayer,
    {
      id: import_constants.LayerId.LineLayer,
      source: lineSource,
      blend: "normal",
      color: layerColor,
      size: 2,
      state: { active: { color: activeColor } },
      zIndex: import_constants.LayerZIndex
    }
  ), /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.PointLayer,
    {
      id: import_constants.LayerId.PointLayer,
      source: pointSource,
      blend: "normal",
      size: 20,
      shape: "pointIcon",
      style: { offsets: [0, 20] },
      state: { active: { color: activeColor } },
      zIndex: import_constants.LayerZIndex
    }
  ), showTextLayer && /* @__PURE__ */ import_react.default.createElement(import_text_layer.EditorTextLayer, null)) : null;
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LayerList
});
