// @antv/l7-editor/es/components/layer-contextmenu-popup/code-block.js
import { message } from 'antd';
import Clipboard from 'clipboard';
import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
export default function CodeBlock(_ref) {
  var copyType = _ref.copyType,
    _text = _ref.text;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  useEffect(function () {
    var clipboard = new Clipboard("#".concat(copyType, "copy_btn"), {
      text: function text() {
        return _text;
      }
    });
    clipboard.on('success', function () {
      message.success(t('layer_contextmenu_popup.fuZhiChengGong'));
    });

    // 销毁 clipboard 实例
    return function () {
      clipboard.destroy();
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    id: copyType,
    style: {
      display: 'none'
    }
  }, _text), /*#__PURE__*/React.createElement("div", {
    id: "".concat(copyType, "copy_btn"),
    "data-clipboard-target": "#".concat(copyType)
  }, t('layer_contextmenu_popup.fuZhi'), " ", copyType));
}
