// @antv/l7-editor/es/components/layer-contextmenu-popup/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { Marker, useLayerList, useScene } from '@antv/larkmap';
import { featureCollection } from '@turf/turf';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { FeatureKey, LayerId } from "../../constants";
import { useFeature } from "../../recoil";
import { GeoJSON2Wkt } from "../../utils/wkt";
import CodeBlock from "./code-block";
import useStyle from "./styles";
export var LayerContextmenuPopup = function LayerContextmenuPopup() {
  var scene = useScene();
  var _useFeature = useFeature(),
    isDraw = _useFeature.isDraw,
    features = _useFeature.features;
  var styles = useStyle();
  var _useState = useState({
      lngLat: {
        lng: 0,
        lat: 0
      },
      visible: false,
      feature: null
    }),
    _useState2 = _slicedToArray(_useState, 2),
    markerProps = _useState2[0],
    setMarkerProps = _useState2[1];
  var targetFeature = useMemo(function () {
    return features.find(function (feature) {
      var _feature$properties;
      return (
        // @ts-ignore
        ((_feature$properties = feature.properties) === null || _feature$properties === void 0 ? void 0 : _feature$properties[FeatureKey.Index]) === markerProps.featureIndex
      );
    });
  }, [features, markerProps]);
  var allLayerList = useLayerList();
  var layerList = useMemo(function () {
    var layerIds = [LayerId.PointLayer, LayerId.LineLayer, LayerId.PolygonLayer];
    return allLayerList.filter(function (layer) {
      return layerIds.includes(layer.id);
    });
  }, [allLayerList]);
  var onLayerClick = useCallback(function (e) {
    if (!isDraw) {
      var lngLat = e.lngLat,
        feature = e.feature;
      var featureIndex = feature.properties[FeatureKey.Index];
      var isIndex = markerProps.featureIndex === feature.properties[FeatureKey.Index];
      if (markerProps.visible && isIndex) {
        setMarkerProps(function (oldPopupProps) {
          return _objectSpread(_objectSpread({}, oldPopupProps), {}, {
            visible: false,
            featureIndex: undefined,
            feature: null
          });
        });
        return;
      }
      setMarkerProps({
        lngLat: lngLat,
        visible: true,
        featureIndex: featureIndex,
        feature: e.feature
      });
    }
  }, [setMarkerProps, markerProps, isDraw]);

  // 单击事件关闭菜单
  var mapRightMenuClose = function mapRightMenuClose() {
    var timeOut = setTimeout(function () {
      if (timeOut) {
        clearTimeout(timeOut);
        setMarkerProps({
          lngLat: {
            lng: 0,
            lat: 0
          },
          visible: false,
          feature: null
        });
      }
    }, 0);
  };
  useEffect(function () {
    //@ts-ignore
    layerList.forEach(function (layer) {
      return layer.on('contextmenu', onLayerClick);
    });
    scene.on('click', mapRightMenuClose);
    return function () {
      //@ts-ignore
      layerList.forEach(function (layer) {
        return layer.off('contextmenu', onLayerClick);
      });
      scene.off('click', mapRightMenuClose);
    };
  }, [onLayerClick, layerList, scene]);
  var copyTypeList = useMemo(function () {
    if (!markerProps.feature) {
      return [];
    }
    var index = markerProps.feature.properties[FeatureKey.Index];
    var featureCollectionData = featureCollection([features[index]]);
    return [{
      copyType: 'GeoJSON',
      text: JSON.stringify(featureCollectionData)
    }, {
      copyType: 'WKT',
      text: GeoJSON2Wkt(featureCollectionData)
    }];
  }, [features, markerProps.feature]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, markerProps.visible && typeof markerProps.featureIndex === 'number' && targetFeature && markerProps.lngLat && /*#__PURE__*/React.createElement(Marker, {
    lngLat: markerProps.lngLat,
    anchor: "top-left"
    // @ts-ignore
    ,
    offsets: [0, 10]
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.layerPopupContent
  }, copyTypeList.map(function (item) {
    return /*#__PURE__*/React.createElement("div", {
      className: styles.layerPopupCopyRow,
      key: item.copyType
    }, /*#__PURE__*/React.createElement(CodeBlock, {
      copyType: item.copyType,
      text: item.text
    }));
  }))));
};
