// @antv/l7-editor/lib/utils/file-to-source.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/utils/file-to-source.ts
var file_to_source_exports = {};
__export(file_to_source_exports, {
  csv2json: () => csv2json,
  parserExcelFileToSource: () => parserExcelFileToSource,
  parserExcelToSource: () => parserExcelToSource,
  parserGeoJson: () => parserGeoJson,
  parserJsonToGeoJson: () => parserJsonToGeoJson,
  parserTextFileToSource: () => parserTextFileToSource,
  readFileAsArrayBuffer: () => readFileAsArrayBuffer,
  readFileAsText: () => readFileAsText
});
module.exports = __toCommonJS(file_to_source_exports);
var import_togeojson = __toESM(require("@mapbox/togeojson"));
var import_lodash_es = require("lodash-es");
var import_papaparse = __toESM(require("papaparse"));
var import_xlsx = require("xlsx");
var import_constants = require("../constants");
var import_wkt = require("./wkt");
var readFileAsText = (file) => {
  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.readAsText(file);
    reader.onload = (event) => {
      var _a;
      const result = (_a = event.target) == null ? void 0 : _a.result;
      resolve(result);
    };
    reader.onerror = (error) => {
      reject(error);
    };
  });
};
var parserGeoJson = (content, name, id) => {
  const originData = JSON.parse(
    content
  );
  return {
    id: id || (0, import_lodash_es.uniqueId)(id),
    metadata: { name },
    data: originData,
    type: "local"
  };
};
var parserJsonToGeoJson = (content, name, id) => {
  const data = JSON.parse(content);
  if (import_constants.FeatureCollectionVT.check(data)) {
    return parserGeoJson(content, name, id);
  }
  return {
    id: id || (0, import_lodash_es.uniqueId)(id),
    metadata: { name },
    data,
    type: "local"
  };
};
var parserTextFileToSource = async (file, name, id) => {
  const fileFullName = file.name;
  const fileExtension = fileFullName.substring(
    fileFullName.lastIndexOf(".") + 1
  );
  const content = await readFileAsText(file);
  if (fileExtension === "json") {
    return parserJsonToGeoJson(content, name, id);
  } else if (fileExtension === "geojson") {
    return parserGeoJson(content, name, id);
  } else if (fileExtension === "kml") {
    const xml = new DOMParser().parseFromString(content, "text/xml");
    const geojson = await import_togeojson.default.kml(xml, {
      style: true
    });
    return {
      id: id || (0, import_lodash_es.uniqueId)(id),
      metadata: { name },
      data: geojson,
      type: "local"
    };
  } else if (fileExtension === "wkt") {
    const geojson = (0, import_wkt.Wkt2GeoJSON)(content);
    return {
      id: id || (0, import_lodash_es.uniqueId)(id),
      metadata: { name },
      data: geojson,
      type: "local"
    };
  }
};
var csv2json = async (file, name, id) => {
  const content = await readFileAsText(file);
  const result = import_papaparse.default.parse(content, {
    header: true,
    skipEmptyLines: true,
    dynamicTyping: true
  });
  return {
    id: id || (0, import_lodash_es.uniqueId)(id),
    metadata: { name },
    data: result.data,
    type: "local",
    columns: result.meta.fields
  };
};
var readFileAsArrayBuffer = (file) => {
  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.readAsArrayBuffer(file);
    reader.onload = (event) => {
      var _a;
      const result = (_a = event.target) == null ? void 0 : _a.result;
      resolve(result);
    };
    reader.onerror = (error) => {
      reject(error);
    };
  });
};
var parserExcelToSource = (content, name, id) => {
  let data;
  try {
    const workbook = (0, import_xlsx.read)(content, { type: "array", cellDates: true });
    const sheet = workbook.Sheets[workbook.SheetNames[0]];
    Object.keys(sheet).forEach((key) => {
      const item = sheet[key];
      if (item.t === "d") {
        item.v = item.w;
      }
    });
    data = import_xlsx.utils.sheet_to_json(sheet);
  } catch (e) {
    throw e;
  }
  const columns = Object.keys(data[0]);
  return {
    id,
    metadata: { name },
    data,
    type: "local",
    columns
  };
};
var parserExcelFileToSource = async (file, name, id) => {
  let content;
  try {
    content = await readFileAsArrayBuffer(file);
  } catch (e) {
    throw e;
  }
  return parserExcelToSource(content, name, id);
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  csv2json,
  parserExcelFileToSource,
  parserExcelToSource,
  parserGeoJson,
  parserJsonToGeoJson,
  parserTextFileToSource,
  readFileAsArrayBuffer,
  readFileAsText
});
