// @antv/l7-editor/lib/utils/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/utils/index.ts
var utils_exports = {};
__export(utils_exports, {
  downloadImg: () => downloadImg,
  downloadText: () => downloadText,
  getDrawStyle: () => getDrawStyle,
  getOpacityColor: () => getOpacityColor,
  getParamsNew: () => getParamsNew,
  getUrlFeatureCollection: () => getUrlFeatureCollection,
  isCircle: () => isCircle,
  isPromise: () => isPromise,
  isRect: () => isRect
});
module.exports = __toCommonJS(utils_exports);
var import_l7_draw = require("@antv/l7-draw");
var import_togeojson = __toESM(require("@mapbox/togeojson"));
var import_turf = require("@turf/turf");
var import_antd = require("antd");
var import_color = __toESM(require("color"));
var import_dayjs = __toESM(require("dayjs"));
var import_lodash_es = require("lodash-es");
var import_constants = require("../constants");
var import_wkt = require("./wkt");
__reExport(utils_exports, require("./feature"), module.exports);
__reExport(utils_exports, require("./gcoord"), module.exports);
__reExport(utils_exports, require("./lnglat"), module.exports);
__reExport(utils_exports, require("./transform"), module.exports);
__reExport(utils_exports, require("./wkt"), module.exports);
var getOpacityColor = (color, alpha) => {
  const colorInstance = (0, import_color.default)(color).fade(alpha);
  return `rgba(${colorInstance.array().join(", ")})`;
};
var getDrawStyle = (color) => {
  const style = (0, import_l7_draw.getSingleColorStyle)(color);
  Object.keys(style).forEach((key) => {
    style[key].options = {
      // @ts-ignore
      ...style[key].options,
      zIndex: import_constants.LayerZIndex
    };
  });
  return style;
};
var downloadText = (text, ext) => {
  const aTag = document.createElement("a");
  aTag.setAttribute(
    "href",
    "data:text/plain;charset=utf-8," + encodeURIComponent(text)
  );
  aTag.setAttribute("download", `${(0, import_dayjs.default)().format("YYYY-MM-DD")}.${ext}`);
  aTag.style.display = "none";
  document.body.appendChild(aTag);
  aTag.click();
};
var downloadImg = (link) => {
  const aTag = document.createElement("a");
  aTag.href = link;
  aTag.download = `${(0, import_dayjs.default)().format("YYYY年MM月DD日HH时mm分ss秒")}.jpeg`;
  aTag.dispatchEvent(new MouseEvent("click"));
};
var getParamsNew = (key) => {
  const temData = new URLSearchParams(window.location.search);
  return temData.get(key);
};
var getUrlFeatureCollection = async (url, urlType = "GeoJSON", t) => {
  const json = await fetch(url);
  if (urlType === "GeoJSON") {
    try {
      const geoData = await json.json();
      return geoData;
    } catch (e) {
      throw new Error(t("utils.index.qingJianChaUR"));
    }
  } else if (urlType === "WKT") {
    const wktStr = await json.text();
    const geojson = (0, import_wkt.Wkt2GeoJSON)(wktStr);
    if (import_constants.FeatureCollectionVT.check(geojson)) {
      return geojson;
    } else {
      import_antd.message.error(t("utils.index.qingJianChaUR"));
    }
  } else if (urlType === "KML") {
    const KML = await json.text();
    const xml = new DOMParser().parseFromString(KML, "text/xml");
    if (xml.getElementsByTagName("parsererror").length > 0) {
      import_antd.message.error(t("utils.index.qingJianChaUR"));
    } else {
      const geojson = await import_togeojson.default.kml(xml, {
        style: true
      });
      return geojson;
    }
  }
};
var isPromise = (obj) => {
  return !(0, import_lodash_es.isUndefined)(obj) && obj instanceof Promise;
};
var isCircle = (feature) => {
  var _a;
  const drawType = (_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.DrawType];
  if (drawType === "circle") {
    return true;
  }
  const centerPosition = (0, import_turf.center)(feature).geometry.coordinates;
  const distanceList = (0, import_turf.coordAll)(feature).map((position) => {
    return Math.round(
      (0, import_turf.distance)(position, centerPosition, {
        units: "meters"
      })
    );
  });
  return Array.from(new Set(distanceList)).length === 1;
};
var isRect = (feature) => {
  var _a;
  const drawType = (_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.DrawType];
  if (drawType === "rect") {
    return true;
  }
  const positions = (0, import_turf.coordAll)(feature);
  if (positions.length !== 5) {
    return false;
  }
  const line1 = (0, import_turf.distance)(positions[0], positions[1], {
    units: "meters"
  }).toFixed(0);
  const line2 = (0, import_turf.distance)(positions[1], positions[2], {
    units: "meters"
  }).toFixed(0);
  const line3 = (0, import_turf.distance)(positions[2], positions[3], {
    units: "meters"
  }).toFixed(0);
  const line4 = (0, import_turf.distance)(positions[3], positions[4], {
    units: "meters"
  }).toFixed(0);
  if (line1 === line3 && line2 === line4) {
    const degree = Math.round(
      (0, import_turf.bearing)(positions[0], positions[3]) - (0, import_turf.bearing)(positions[0], positions[1])
    );
    return degree === 90 || degree === 270;
  }
  return false;
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  downloadImg,
  downloadText,
  getDrawStyle,
  getOpacityColor,
  getParamsNew,
  getUrlFeatureCollection,
  isCircle,
  isPromise,
  isRect,
  ...require("./feature"),
  ...require("./gcoord"),
  ...require("./lnglat"),
  ...require("./transform"),
  ...require("./wkt")
});
