// @antv/l7-editor/es/components/map-control-group/draw-control/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { ControlEvent, DrawEvent, DrawControl as L7DrawControl } from '@antv/l7-draw';
import { CustomControl, useScene } from '@antv/larkmap';
import { cloneDeep, fromPairs } from 'lodash-es';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { FeatureKey } from "../../../constants";
import { useDrawHelper } from "../../../hooks";
import { useFeature, useGlobal } from "../../../recoil";
import { getDrawStyle } from "../../../utils";
import useStyle from "./styles";
var DrawControl = function DrawControl() {
  var helperText = useDrawHelper();
  var scene = useScene();
  var styles = useStyle();
  var _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    drawControl = _useState2[0],
    setDrawControl = _useState2[1];
  var _useFeature = useFeature(),
    setIsDraw = _useFeature.setIsDraw,
    resetFeatures = _useFeature.resetFeatures,
    features = _useFeature.features,
    revertCoord = _useFeature.revertCoord;
  var _useGlobal = useGlobal(),
    layerColor = _useGlobal.layerColor,
    showDrawDistance = _useGlobal.showDrawDistance,
    showDrawArea = _useGlobal.showDrawArea;
  var editFeature = useMemo(function () {
    return cloneDeep(features.find(
    // @ts-ignore
    function (feature) {
      return feature.properties[FeatureKey.IsEdit];
    }));
  }, [features]);
  useEffect(function () {
    var option = {
      distanceOptions: showDrawDistance ? {} : undefined,
      areaOptions: showDrawArea ? {} : undefined
    };
    var newDrawControl;
    if (scene) {
      var _document$querySelect;
      newDrawControl = new L7DrawControl(scene, {
        position: 'topleft',
        drawConfig: {
          point: true,
          line: option,
          polygon: option,
          rect: option,
          circle: option
        },
        commonDrawOptions: {
          maxCount: 1,
          style: getDrawStyle(layerColor),
          //@ts-ignore
          helper: helperText
        }
      });
      setDrawControl(newDrawControl);
      scene.addControl(newDrawControl);
      newDrawControl.on(ControlEvent.DrawChange, function (newType) {
        setIsDraw(!!newType);
      });
      var drawDom = document.querySelector('.l7-draw-control');
      drawDom.style.marginTop = 0;
      (_document$querySelect = document.querySelector('#l7-draw-content')) === null || _document$querySelect === void 0 || _document$querySelect.appendChild(drawDom);
    }
    return function () {
      if (newDrawControl) {
        scene.removeControl(newDrawControl);
      }
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [scene, layerColor, showDrawDistance, showDrawArea]);
  var onDrawAdd = useCallback(function (drawType, newFeature) {
    newFeature.properties = _defineProperty({}, FeatureKey.DrawType, drawType);
    drawControl === null || drawControl === void 0 || drawControl.clearDrawData();
    drawControl === null || drawControl === void 0 || drawControl.setActiveType(null);
    var newFeatures = revertCoord([newFeature]);
    resetFeatures([].concat(_toConsumableArray(features), _toConsumableArray(newFeatures)));
  }, [drawControl, revertCoord, resetFeatures, features]);
  var onDrawEdit = useCallback(function (drawType, feature) {
    // @ts-ignore
    var index = feature.properties[FeatureKey.Index];
    var newFeatures = _toConsumableArray(features);
    // @ts-ignore
    newFeatures[index].properties[FeatureKey.IsEdit] = false;
    newFeatures[index].geometry = feature.geometry;
    drawControl === null || drawControl === void 0 || drawControl.clearDrawData();
    drawControl === null || drawControl === void 0 || drawControl.setActiveType(null);
    var revertFeatures = revertCoord(newFeatures);
    resetFeatures(_toConsumableArray(revertFeatures));
  }, [features, drawControl, revertCoord, resetFeatures]);
  useEffect(function () {
    var drawTypeList = ['point', 'line', 'polygon', 'rect', 'circle'];
    var onDrawAddMap = fromPairs(drawTypeList.map(function (drawType) {
      return [drawType, {
        onAdd: function onAdd(newFeature) {
          return onDrawAdd(drawType, newFeature);
        },
        onEdit: function onEdit(newFeature) {
          return onDrawEdit(drawType, newFeature);
        }
      }];
    }));
    drawTypeList.forEach(function (drawType) {
      var draw = drawControl === null || drawControl === void 0 ? void 0 : drawControl.getTypeDraw(drawType);
      draw === null || draw === void 0 || draw.on(DrawEvent.Add, onDrawAddMap[drawType].onAdd);
      draw === null || draw === void 0 || draw.on(DrawEvent.Edit, onDrawAddMap[drawType].onEdit);
    });
    return function () {
      drawTypeList.forEach(function (drawType) {
        var draw = drawControl === null || drawControl === void 0 ? void 0 : drawControl.getTypeDraw(drawType);
        draw === null || draw === void 0 || draw.off(DrawEvent.Add, onDrawAddMap[drawType].onAdd);
        draw === null || draw === void 0 || draw.off(DrawEvent.Edit, onDrawAddMap[drawType].onEdit);
      });
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [drawControl, onDrawAdd]);
  useEffect(function () {
    if (editFeature && drawControl) {
      var _editFeature$properti;
      // @ts-ignore
      var drawType = (_editFeature$properti = editFeature.properties) === null || _editFeature$properti === void 0 ? void 0 : _editFeature$properti[FeatureKey.DrawType];
      // @ts-ignore
      editFeature.properties.isActive = true;
      drawControl.setActiveType(drawType);
      var targetDraw = drawControl.getTypeDraw(drawType);
      targetDraw === null || targetDraw === void 0 || targetDraw.setData([editFeature]);
    }
  }, [editFeature, drawControl]);
  return /*#__PURE__*/React.createElement(CustomControl, {
    position: "topright"
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex'
    },
    id: "l7-editor-draw"
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.l7DrawControl
  }, /*#__PURE__*/React.createElement("div", {
    id: "l7-draw-content"
  }))));
};
export default DrawControl;
