// @antv/l7-editor/lib/components/map-control-group/l7-option-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/l7-option-control/index.tsx
var l7_option_control_exports = {};
__export(l7_option_control_exports, {
  L7MapOptionControl: () => L7MapOptionControl
});
module.exports = __toCommonJS(l7_option_control_exports);
var import_larkmap = require("@antv/larkmap");
var import_antd = require("antd");
var import_clipboard = __toESM(require("clipboard"));
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_iconfont = require("../../../components/iconfont");
var import_styles = __toESM(require("../styles"));
var { TextArea } = import_antd.Input;
var L7MapOptionControl = () => {
  const scene = (0, import_larkmap.useScene)();
  const styles = (0, import_styles.default)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const [isModalOpen, setIsModalOpen] = (0, import_react.useState)(false);
  const [textValue, setTextValue] = (0, import_react.useState)("");
  const showModal = () => {
    setIsModalOpen(true);
    const optionJson = {
      center: [scene.getCenter().lng, scene.getCenter().lat],
      zoom: scene.getZoom(),
      pitch: Number(scene.getPitch().toFixed(6)),
      rotation: scene.getRotation()
    };
    setTextValue(JSON.stringify(optionJson, null, 2));
  };
  const handleOk = () => {
    const clipboard = new import_clipboard.default(`#l7-option-copy_btn`, {
      text: () => textValue
    });
    clipboard.on("success", () => {
      setIsModalOpen(false);
      import_antd.message.success(t("layer_contextmenu_popup.fuZhiChengGong"));
      clipboard.destroy();
    });
  };
  const handleCancel = () => {
    setIsModalOpen(false);
  };
  return /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "bottomleft" }, /* @__PURE__ */ import_react.default.createElement(import_antd.Tooltip, { placement: "left", overlay: t("l7Options.huoqucanshu") }, /* @__PURE__ */ import_react.default.createElement(
    "button",
    {
      className: styles.L7EditorControl,
      id: "text-layer-control",
      onClick: showModal
    },
    /* @__PURE__ */ import_react.default.createElement(import_iconfont.IconFont, { type: "icon-xitongzhuangtai" })
  )), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Modal,
    {
      title: t("l7Options.dangqiancanshu"),
      open: isModalOpen,
      onCancel: handleCancel,
      width: 800,
      destroyOnClose: true,
      footer: /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_antd.Button, { onClick: handleCancel }, t("btn.setting_btn.guanBi")), /* @__PURE__ */ import_react.default.createElement(import_antd.Button, { type: "primary", id: "l7-option-copy_btn", onClick: handleOk }, t("layer_contextmenu_popup.fuZhi")))
    },
    /* @__PURE__ */ import_react.default.createElement(
      TextArea,
      {
        rows: 10,
        value: textValue,
        onChange: (e) => {
          setTextValue(e.target.value);
        }
      }
    )
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  L7MapOptionControl
});
