// @antv/l7-editor/lib/components/map-control-group/official-layer-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/official-layer-control/index.tsx
var official_layer_control_exports = {};
__export(official_layer_control_exports, {
  OfficialLayerControl: () => OfficialLayerControl
});
module.exports = __toCommonJS(official_layer_control_exports);
var import_icons = require("@ant-design/icons");
var import_larkmap = require("@antv/larkmap");
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_classnames = __toESM(require("classnames"));
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_constants = require("../../../constants");
var import_recoil = require("../../../recoil");
var import_styles = __toESM(require("./styles"));
var layout = {
  labelCol: { span: 4 },
  wrapperCol: { span: 16 }
};
var enLayout = {
  labelCol: { span: 6 },
  wrapperCol: { span: 15 }
};
function OfficialLayerControl() {
  const [form] = import_antd.Form.useForm();
  const styles = (0, import_styles.default)();
  const { layerType, setLayerType, customTiles, setCustomTiles, locale } = (0, import_recoil.useGlobal)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const [radioValue, setRadioValue] = (0, import_react.useState)(
    layerType.length ? layerType[0] : import_constants.OfficeLayerEnum.VectorMap
  );
  const [isModalOpen, setIsModalOpen] = (0, import_react.useState)(false);
  const [isEdit, setIsEdit] = (0, import_react.useState)(false);
  const [editIndex, setEditIndex] = (0, import_react.useState)(-1);
  const [base64, setBase64] = (0, import_react.useState)(null);
  const [fileList, setFileList] = (0, import_react.useState)([]);
  const [hideOfficeLayer, setHideOfficeLayer] = (0, import_ahooks.useLocalStorageState)(
    "hideOfficeLayer",
    {
      defaultValue: true
    }
  );
  const handleOk = () => {
    form.submit();
  };
  const BASE_LAYER_GROUP = [
    {
      id: import_constants.OfficeLayerEnum.VectorMap,
      title: t("official_layer_control.index.shiLiangDiTu"),
      image: "https://mdn.alipayobjects.com/huamei_qa8qxu/afts/img/A*qdFDSbvIalgAAAAAAAAAAAAADmJ7AQ/original",
      layers: []
    }
    // {
    //   id: OfficeLayerEnum.GoogleSatellite,
    //   title: t('official_layer_control.index.guGeWeiXingTu'),
    //   image:
    //     'https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*zi2jSqqZ2-8AAAAAAAAAAAAADjWqAQ/original',
    //   layers: [GOOGLE_TILE_MAP_URL, GOOGLE_TILE_MAP_ROUTER_URL],
    // },
  ];
  const officeLayerGroup = (0, import_react.useMemo)(() => {
    if (customTiles.length) {
      return [...BASE_LAYER_GROUP, ...customTiles];
    }
    return BASE_LAYER_GROUP;
  }, [customTiles, t]);
  const handleBeforeUpload = (file) => {
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = () => {
      const base64Image = reader.result;
      setBase64(base64Image);
    };
    return false;
  };
  const onItemClick = (item) => {
    setRadioValue(item.id);
    setLayerType(
      item.id === import_constants.OfficeLayerEnum.VectorMap ? [] : [item.id]
    );
  };
  const onFinish = (e) => {
    if (isEdit) {
      const cloneCustomTiles = (0, import_lodash_es.cloneDeep)(customTiles);
      const newImgUrl = Array.isArray(e.img) ? e.img[0].url : `${base64}`;
      cloneCustomTiles[editIndex - 2] = {
        id: e.name,
        image: newImgUrl,
        title: e.name,
        layers: e.urls
      };
      setCustomTiles(cloneCustomTiles);
      setIsEdit(false);
      setEditIndex(-1);
    } else {
      setCustomTiles((prevState) => [
        ...prevState,
        {
          id: e.name,
          image: `${base64}`,
          title: e.name,
          layers: e.urls
        }
      ]);
    }
    setIsEdit(false);
    setFileList([]);
    setIsModalOpen(false);
  };
  const handleCancel = () => {
    setIsModalOpen(false);
    setIsEdit(false);
    setFileList([]);
    setEditIndex(-1);
    form.resetFields();
  };
  const onConfirm = (e, item) => {
    e == null ? void 0 : e.stopPropagation();
    const newCustomTiles = customTiles.filter((val) => {
      return val.id !== item.id;
    });
    if (item.id === radioValue) {
      setRadioValue(import_constants.OfficeLayerEnum.VectorMap);
      setLayerType([]);
    }
    setCustomTiles(newCustomTiles);
  };
  const validateSpace = (_, value) => {
    const lowerCaseValue = value.toLowerCase();
    let listArr = [];
    if (isEdit) {
      const cloneOfficeLayerGroup = (0, import_lodash_es.cloneDeep)(officeLayerGroup);
      cloneOfficeLayerGroup.splice(editIndex, 1);
      listArr = [...cloneOfficeLayerGroup];
    } else {
      listArr = officeLayerGroup;
    }
    const hasDuplicate = listArr.every(
      (item) => item.title.toLowerCase() !== lowerCaseValue
    );
    if (!hasDuplicate) {
      return Promise.reject(t("official_layer_control.index.mingChengChongFu"));
    }
    if (value && value.trim() === "") {
      return Promise.reject(t("official_layer_control.index.kongGe"));
    }
    return Promise.resolve();
  };
  const uploadValidateSpace = (_, value) => {
    const reject = Promise.reject(t("official_layer_control.index.shangchuan"));
    if (value.fileList) {
      if (!value.fileList.length) {
        return reject;
      } else {
        return Promise.resolve();
      }
    } else {
      if (!value.length) {
        return reject;
      } else {
        return Promise.resolve();
      }
    }
  };
  const handleChange = (info) => {
    let newFileList = [...info.fileList];
    newFileList = newFileList.slice(-2);
    newFileList = newFileList.map((file) => {
      if (file.response) {
        file.url = file.response.url;
      }
      return file;
    });
    setFileList(newFileList);
  };
  const rasterLayer = (0, import_react.useMemo)(() => {
    if (layerType.length) {
      const findItem = officeLayerGroup.find(
        (item) => item.id === layerType[0]
      );
      return findItem == null ? void 0 : findItem.layers.map((item) => {
        return (
          // eslint-disable-next-line react/jsx-key
          /* @__PURE__ */ import_react.default.createElement(
            import_larkmap.RasterLayer,
            {
              zIndex: 1,
              id: findItem.id === import_constants.OfficeLayerEnum.GoogleSatellite && item === import_constants.GOOGLE_TILE_MAP_URL ? "googleTileMap" : void 0,
              source: {
                data: item,
                parser: { type: "rasterTile", tileSize: 256, zoomOffset: 0 }
              }
            }
          )
        );
      });
    } else {
      return null;
    }
  }, [layerType, officeLayerGroup]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "bottomleft" }, /* @__PURE__ */ import_react.default.createElement("div", { className: styles.mapTab, id: "l7-editor-aMap" }, /* @__PURE__ */ import_react.default.createElement(
    "div",
    {
      className: styles.hideOfficeLayerBtn,
      onClick: () => {
        setHideOfficeLayer(!hideOfficeLayer);
      }
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_icons.CaretRightOutlined,
      {
        style: {
          transform: hideOfficeLayer ? "rotate(-180deg)" : void 0
        }
      }
    )
  ), hideOfficeLayer && /* @__PURE__ */ import_react.default.createElement("div", { className: styles.amapInfo }, officeLayerGroup.map((item, index) => {
    return /* @__PURE__ */ import_react.default.createElement(
      "div",
      {
        key: item.id,
        className: (0, import_classnames.default)([
          styles.amapInfoItem,
          item.id === radioValue ? styles.itemBorderActive : styles.itemBorder,
          index === officeLayerGroup.length - 1 ? "item-hover" : ""
        ]),
        onClick: () => {
          onItemClick(item);
        }
      },
      index > 1 && /* @__PURE__ */ import_react.default.createElement(
        import_antd.Popconfirm,
        {
          title: t("official_layer_control.index.shanChuDiTu"),
          onConfirm: (e) => onConfirm(e, item),
          onCancel: (e) => {
            e == null ? void 0 : e.stopPropagation();
          }
        },
        /* @__PURE__ */ import_react.default.createElement(
          "div",
          {
            className: "item-clear",
            onClick: (e) => {
              e.stopPropagation();
            }
          },
          /* @__PURE__ */ import_react.default.createElement(import_icons.DeleteOutlined, null)
        )
      ),
      index > 1 && /* @__PURE__ */ import_react.default.createElement(
        "div",
        {
          className: "item-edit",
          onClick: (e) => {
            setEditIndex(index);
            e.stopPropagation();
            setIsEdit(true);
            setIsModalOpen(true);
            setFileList([
              {
                uid: "-1",
                name: `${item.title}`,
                status: "done",
                url: item.image
              }
            ]);
            form.setFieldsValue({
              name: item.title,
              urls: item.layers,
              img: [
                {
                  uid: "-1",
                  name: `${item.title}`,
                  status: "done",
                  url: item.image
                }
              ]
            });
          }
        },
        /* @__PURE__ */ import_react.default.createElement(import_icons.FormOutlined, null)
      ),
      /* @__PURE__ */ import_react.default.createElement(
        "img",
        {
          src: item.image,
          alt: "",
          className: styles.amapInfoItemImage
        }
      ),
      /* @__PURE__ */ import_react.default.createElement(
        "div",
        {
          className: styles.amapInfoItemTitle,
          style: { marginTop: 0 }
        },
        item.title
      )
    );
  }), /* @__PURE__ */ import_react.default.createElement("div", { className: "add-map" }, /* @__PURE__ */ import_react.default.createElement(
    "div",
    {
      onClick: () => {
        setIsModalOpen(true);
        form.resetFields();
      }
    },
    /* @__PURE__ */ import_react.default.createElement("div", { className: styles.addMapIcon }, /* @__PURE__ */ import_react.default.createElement(import_icons.PlusOutlined, null))
  )))), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Modal,
    {
      title: t("official_layer_control.index.tianJiaDitu"),
      open: isModalOpen,
      destroyOnClose: true,
      onOk: handleOk,
      onCancel: handleCancel,
      width: 600
    },
    /* @__PURE__ */ import_react.default.createElement(import_antd.Form, { form, initialValues: { urls: [""] }, onFinish }, /* @__PURE__ */ import_react.default.createElement(
      import_antd.Form.Item,
      {
        ...locale === "zh-CN" ? layout : enLayout,
        name: "name",
        label: t("official_layer_control.index.name"),
        rules: [{ required: true }, { validator: validateSpace }]
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Input,
        {
          placeholder: t("official_layer_control.index.addName"),
          style: {
            width: 390
          }
        }
      )
    ), /* @__PURE__ */ import_react.default.createElement(
      import_antd.Form.Item,
      {
        ...locale === "zh-CN" ? layout : enLayout,
        name: "img",
        label: t("official_layer_control.index.shiLiTuPian"),
        rules: [{ required: true, validator: uploadValidateSpace }]
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Upload,
        {
          beforeUpload: handleBeforeUpload,
          accept: ".png,.jpg",
          maxCount: 1,
          onChange: handleChange,
          fileList,
          onRemove: () => {
            setFileList([]);
          }
        },
        /* @__PURE__ */ import_react.default.createElement(import_antd.Button, { icon: /* @__PURE__ */ import_react.default.createElement(import_icons.UploadOutlined, null) }, t("import_btn.index.shangChuan"))
      )
    ), /* @__PURE__ */ import_react.default.createElement(import_antd.Form.List, { name: "urls" }, (fields, { add, remove }) => /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, fields.map((field, index) => /* @__PURE__ */ import_react.default.createElement(
      import_antd.Space,
      {
        key: field.key,
        style: { display: "flex", marginBottom: 8 },
        align: "baseline"
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_antd.Form.Item,
        {
          ...field,
          label: index === 0 ? t("official_layer_control.index.tuCengDiZhi") : "",
          rules: [
            {
              required: true,
              message: t(
                "official_layer_control.index.qiShuRutuCengDiZhi"
              )
            },
            { validator: validateSpace }
          ],
          style: {
            marginLeft: locale === "zh-CN" ? index === 0 ? 10 : 90 : index === 0 ? 18 : 134
          }
        },
        /* @__PURE__ */ import_react.default.createElement(
          import_antd.Input,
          {
            placeholder: import_constants.GOOGLE_TILE_MAP_URL,
            style: {
              width: 390
            }
          }
        )
      ),
      fields.length > 1 ? /* @__PURE__ */ import_react.default.createElement(
        import_icons.MinusCircleOutlined,
        {
          onClick: () => remove(field.name)
        }
      ) : null
    )), /* @__PURE__ */ import_react.default.createElement(import_antd.Form.Item, { style: { textAlign: "center" } }, /* @__PURE__ */ import_react.default.createElement(
      import_antd.Button,
      {
        type: "dashed",
        onClick: () => add(),
        icon: /* @__PURE__ */ import_react.default.createElement(import_icons.PlusOutlined, null),
        style: {
          width: 390,
          marginLeft: locale === "zh-CN" ? 20 : 104
        }
      },
      t("official_layer_control.index.tinJiaWaPian")
    )))))
  )), /* @__PURE__ */ import_react.default.createElement("div", null, rasterLayer));
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  OfficialLayerControl
});
