// @antv/l7-editor/lib/components/map-control-group/official-layer-control/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/official-layer-control/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const { colorBgElevated, colorBgContainer, colorText, colorIcon, boxShadow } = token;
  return {
    mapTab: import_css.css`
      background-color: ${colorBgElevated};
      border-radius: 4px;
      display: flex;
      align-items: center;
      box-shadow: ${boxShadow};
      &:hover .add-map {
        display: block;
      }
      .add-map {
        display: none;
        text-align: center;
        &:hover {
          display: block;
          color: #1677ff;
        }
      }
    `,
    hideOfficeLayerBtn: import_css.css`
      height: 127px;
      width: 28px;
      display: flex;
      align-items: center;
      justify-content: center;
      background: ${colorBgElevated};
      border-radius: 4px;
      cursor: pointer;
      font-size: 16px;
      z-index: 2;
      .anticon {
        transition: all 0.2s;
      }
      color: ${colorIcon};
    `,
    amapInfo: import_css.css`
      display: flex;
      align-items: center;
      padding: 10px;
      cursor: pointer;
    `,
    amapInfoItemImage: import_css.css`
      width: 142px;
      height: 80px;
    `,
    amapInfoItem: import_css.css`
      text-align: center;
      margin-left: 10px;
      object-fit: cover;
      &:first-child {
        margin-left: 0px;
      }
      box-sizing: border-box;
      border-radius: 4px;
      position: relative;

      &: hover .item-clear {
        display: block;
      }

      &: hover .item-edit {
        display: block;
      }

      .item-edit {
        position: absolute;
        right: 41px;
        bottom: 30px;
        display: none;
        width: 25px;
        height: 25px;
        line-height: 25px;
        text-align: center;
        background: #fff;
        border-radius: 4px;
        cursor: pointer;

        &:hover {
          color: #1677ff;
        }
      }

      .item-clear {
        position: absolute;
        right: 8px;
        bottom: 30px;
        display: none;
        width: 25px;
        height: 25px;
        line-height: 25px;
        text-align: center;
        background: #fff;
        border-radius: 4px;
        cursor: pointer;

        &:hover {
          color: #1677ff;
        }
      }
    `,
    amapInfoItemTitle: import_css.css`
      color: ${colorText};
    `,
    itemBorder: import_css.css`
      border: 2px solid ${colorBgContainer};
    `,
    itemBorderActive: import_css.css`
      border: solid 2px #1677ff;
    `,
    addMapIcon: import_css.css`
      border: dashed 1px rgb(217, 217, 217);
      height: 103px;
      width: 142px;
      object-fit: cover;
      border-radius: 4px;
      line-height: 103px;
      margin-left: 10px;
      cursor: pointer;
      &:hover {
        border: dashed 1px #1677ff;
      }
    `
  };
};
var styles_default = useStyle;
