// @antv/l7-editor/lib/components/map-control-group/administrative-select-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/administrative-select-control/index.tsx
var administrative_select_control_exports = {};
__export(administrative_select_control_exports, {
  AdministrativeSelect: () => AdministrativeSelect
});
module.exports = __toCommonJS(administrative_select_control_exports);
var import_larkmap = require("@antv/larkmap");
var import_turf = require("@turf/turf");
var import_antd = require("antd");
var import_classnames = __toESM(require("classnames"));
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_constants = require("../../../constants");
var import_recoil = require("../../../recoil");
var import_iconfont = require("../../iconfont");
var import_styles = require("./styles");
var DistrictLayerOptions = {
  shape: "line",
  color: "#ff0000",
  size: 2,
  zIndex: import_constants.LayerZIndex,
  style: {
    opacity: 0.8
  }
};
var AdministrativeSelect = () => {
  const styles = (0, import_styles.useStyle)();
  const { cityHistory, setCityHistory } = (0, import_recoil.useGlobal)();
  const { scene } = (0, import_recoil.useFeature)();
  const [districtFeature, setDistrictFeature] = (0, import_react.useState)(null);
  const [data, setData] = (0, import_react.useState)();
  const [value, setValue] = (0, import_react.useState)([]);
  const { t } = (0, import_react_i18next.useTranslation)();
  const getCascadeData = (list) => {
    list.sort((a, b) => {
      return +a.adcode - +b.adcode;
    });
    if (list.length) {
      return list.map((item) => {
        const { name, districts, adcode } = item;
        return {
          adcode,
          value: adcode,
          label: name,
          children: getCascadeData(districts)
        };
      });
    } else {
      return [];
    }
  };
  (0, import_react.useEffect)(() => {
    fetch(
      "https://restapi.amap.com/v3/config/district?key=98d10f05a2da96697313a2ce35ebf1a2&keywords=中华人民共和国&subdistrict=3&extensions=base"
    ).then((res) => res.json()).then((res) => {
      setData(getCascadeData(res.districts[0].districts));
    });
  }, []);
  const onChange = (items, option) => {
    setValue(items);
    if (option) {
      const code = [];
      const label = [];
      option.forEach((item2) => {
        code.push(item2.adcode);
        label.push(item2.label);
      });
      const item = { value: JSON.stringify(code), label: label.join("/") };
      const arr = [item, ...cityHistory];
      const formatArr = () => {
        const map = /* @__PURE__ */ new Map();
        for (const o of arr) {
          if (!map.has(o.value)) {
            map.set(o.value, o);
          }
        }
        return [...map.values()];
      };
      setCityHistory(formatArr());
      if (cityHistory.length >= 10) {
        const arrHistory = (0, import_lodash_es.cloneDeep)(cityHistory);
        arrHistory.pop();
        setCityHistory(arrHistory);
      }
    } else {
      setDistrictFeature(null);
    }
  };
  const filter = (inputValue, path) => path.some(
    (option) => option.label.toLowerCase().indexOf(inputValue.toLowerCase()) > -1
  );
  const historyItem = (0, import_react.useMemo)(() => {
    if (cityHistory.length) {
      const item = cityHistory.map((city) => {
        return {
          key: city.value,
          label: /* @__PURE__ */ import_react.default.createElement(
            "div",
            {
              onClick: () => {
                setValue(JSON.parse(city.value));
              }
            },
            city.label
          )
        };
      });
      return item;
    } else {
      return [{ key: "undefined", label: /* @__PURE__ */ import_react.default.createElement(import_antd.Empty, null) }];
    }
  }, [cityHistory]);
  (0, import_react.useEffect)(() => {
    if (value) {
      const name = value[value.length - 1];
      fetch(
        `https://restapi.amap.com/v3/config/district?keywords=${name}&subdistrict=0&key=98d10f05a2da96697313a2ce35ebf1a2&extensions=all`
      ).then((res) => res.json()).then((res) => {
        var _a;
        if (res.status === "1" && ((_a = res.districts) == null ? void 0 : _a.length) && scene) {
          const [lng, lat] = res.districts[0].center.split(",").map((item) => +item);
          scene.setZoomAndCenter(9, [lng, lat]);
          const positions = [];
          res.districts.forEach((district) => {
            district.polyline.split("|").forEach((chunk) => {
              positions.push(
                chunk.split(";").map((item) => item.split(",").map((num) => +num))
              );
            });
          });
          setDistrictFeature((0, import_turf.multiLineString)(positions));
        }
      }).catch(() => {
        import_antd.message.error(
          t("administrative_select_control.index.weiLanShuJuQing")
        );
      });
    }
  }, [value]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "lefttop" }, /* @__PURE__ */ import_react.default.createElement("div", { style: { display: "flex" } }, /* @__PURE__ */ import_react.default.createElement("div", { id: "l7-editor-administrativeSelect" }, /* @__PURE__ */ import_react.default.createElement(
    import_antd.Cascader,
    {
      options: data,
      value,
      onChange,
      className: styles.cascader,
      allowClear: true,
      showSearch: { filter },
      placeholder: t(
        "administrative_select_control.index.keXuanZeShengShi"
      ),
      changeOnSelect: true,
      style: { width: 250 },
      popupClassName: styles.cascaderPopup,
      expandTrigger: "hover"
    }
  )), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Dropdown,
    {
      menu: {
        items: historyItem
      },
      placement: "bottomRight",
      trigger: ["click"]
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Tooltip,
      {
        title: t("administrative_select_control.index.xingZhengQuHuaLi"),
        placement: "right"
      },
      /* @__PURE__ */ import_react.default.createElement(
        "div",
        {
          className: (0, import_classnames.default)([styles.history, "l7-draw-control__btn"])
        },
        /* @__PURE__ */ import_react.default.createElement(import_iconfont.IconFont, { type: "icon-lishi", className: styles.historyIcon })
      )
    )
  ))), /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.LineLayer,
    {
      source: {
        data: (0, import_turf.featureCollection)(districtFeature ? [districtFeature] : [])
      },
      ...DistrictLayerOptions
    }
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AdministrativeSelect
});
