// @antv/l7-editor/lib/components/map-control-group/administrative-select-control/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/administrative-select-control/styles.ts
var styles_exports = {};
__export(styles_exports, {
  useStyle: () => useStyle
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const {
    colorBgElevated,
    colorFillSecondary,
    colorBgContainer,
    colorBorder,
    colorText,
    colorIcon
  } = token;
  return {
    cascader: import_css.css`
      .ant-select-selector {
        border-radius: 0;
        border-radius: 6px 0 0 6px;
      }
    `,
    cascaderPopup: import_css.css`
      .ant-cascader-dropdown .ant-cascader-menu {
        height: 260px !important;
      }
      *::-webkit-scrollbar {
        width: 8px;
        background-color: ${colorBgElevated};
      }
      *::-webkit-scrollbar-thumb {
        background-color: ${colorFillSecondary};
        border-radius: 10px;
      }
    `,
    history: import_css.css`
      height: auto !important;
      background: ${colorBgContainer} !important;
      border-right: none !important;
      border-radius: 0 6px 6px 0;
      border: 1px solid ${colorBorder};
      border-bottom: 1px solid ${colorBorder} !important;
      color: ${colorIcon};
      :hover {
        color: ${colorText};
        background: ${colorBgContainer};
      }
    `,
    historyIcon: import_css.css`
      font-size: 20px;
    `
  };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  useStyle
});
