// @antv/l7-editor/lib/components/geojson-editor/editortool.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/geojson-editor/editortool.ts
var editortool_exports = {};
__export(editortool_exports, {
  provideCompletionItems: () => provideCompletionItems
});
module.exports = __toCommonJS(editortool_exports);
var turf = __toESM(require("@turf/turf"));
var import_lodash_es = __toESM(require("lodash-es"));
var monacoEditor = __toESM(require("monaco-editor/esm/vs/editor/editor.api"));
function provideCompletionItems(model, position, type) {
  const wordUntilPosition = model.getWordUntilPosition(position);
  const range = {
    startLineNumber: position.lineNumber,
    endLineNumber: position.lineNumber,
    startColumn: wordUntilPosition.startColumn,
    endColumn: wordUntilPosition.endColumn
  };
  let suggestions = [];
  if (type === "turf") {
    for (const key in turf) {
      const completionItem = {
        label: key,
        kind: monacoEditor.languages.CompletionItemKind.Function,
        documentation: turf[key].toString(),
        insertText: key,
        range
      };
      suggestions.push(completionItem);
    }
  } else {
    suggestions = import_lodash_es.default.chain(import_lodash_es.default).keys().map((key) => {
      return {
        label: key,
        kind: monacoEditor.languages.CompletionItemKind.Function,
        insertText: key,
        range
      };
    }).value();
  }
  return {
    suggestions
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  provideCompletionItems
});
