// @antv/l7-editor/lib/components/geojson-editor/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/geojson-editor/index.tsx
var geojson_editor_exports = {};
__export(geojson_editor_exports, {
  GeoJsonEditor: () => GeoJsonEditor
});
module.exports = __toCommonJS(geojson_editor_exports);
var import_ahooks = require("ahooks");
var monacoEditor = __toESM(require("monaco-editor/esm/vs/editor/editor.api"));
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_react_monaco_editor = __toESM(require("react-monaco-editor"));
var import_recoil = require("../../recoil");
var import_utils = require("../../utils");
var import_prettier_text = require("../../utils/prettier-text");
var import_editortool = require("./editortool");
var import_styles = __toESM(require("./styles"));
var GeoJsonEditor = (0, import_react.forwardRef)((props, ref) => {
  const { language = "json" } = props;
  const { theme, autoFitBounds } = (0, import_recoil.useGlobal)();
  const { editorText, setEditorText, saveEditorText, bboxAutoFit } = (0, import_recoil.useFeature)();
  const [scriptContent, setScriptContent] = (0, import_react.useState)("");
  const [container, setContainer] = (0, import_react.useState)(null);
  const { width = 0, height = 0 } = (0, import_ahooks.useSize)(container) ?? {};
  const styles = (0, import_styles.default)();
  const [geoJonText, setGeoJonText] = (0, import_react.useState)("");
  const debouncedGeoJsonText = (0, import_ahooks.useDebounce)(geoJonText, { wait: 2e3 });
  const { t } = (0, import_react_i18next.useTranslation)();
  monacoEditor.languages.registerDocumentFormattingEditProvider(language, {
    provideDocumentFormattingEdits: (model) => {
      return [
        {
          range: model.getFullModelRange(),
          text: (0, import_prettier_text.prettierText)({ content: model.getValue(), parser: language })
        }
      ];
    }
  });
  (0, import_ahooks.useMount)(() => {
    monacoEditor.editor.defineTheme("custome-theme", {
      base: "vs",
      inherit: true,
      rules: [
        { token: t("geojson_editor.index.tiaoShi"), foreground: "959595" },
        { token: t("geojson_editor.index.tongZhi"), foreground: "00b4ff" },
        { token: t("geojson_editor.index.jingGao"), foreground: "fff000" },
        { token: t("geojson_editor.index.cuoWu"), foreground: "ff0000" },
        { token: t("geojson_editor.index.bengKui"), foreground: "c30209" },
        { token: t("geojson_editor.index.xinXi"), foreground: "ffffff" }
      ],
      colors: {
        "editor.background": "#fafafa",
        "editorLineNumber.foreground": "#222222",
        "editor.lineHighlightBackground": "#f4f4f4"
      }
    });
    monacoEditor.languages.registerCompletionItemProvider(language, {
      provideCompletionItems: (model, position) => (0, import_editortool.provideCompletionItems)(model, position, "lodash")
    });
    monacoEditor.languages.registerCompletionItemProvider(language, {
      provideCompletionItems: (model, position) => (0, import_editortool.provideCompletionItems)(model, position, "turf")
    });
  });
  const monacoChange = (event) => {
    if (language === "json") {
      setEditorText(event);
      setGeoJonText(event);
      return;
    }
    setScriptContent == null ? void 0 : setScriptContent(event);
  };
  (0, import_react.useImperativeHandle)(
    ref,
    () => ({
      getData: () => (
        // eslint-disable-next-line no-async-promise-executor
        new Promise(async (resolve, reject) => {
          let geoData;
          const funcResult = new Function(scriptContent);
          if (funcResult()) {
            geoData = funcResult();
          } else {
            const evalResult = eval(scriptContent);
            geoData = (0, import_utils.isPromise)(evalResult) ? await evalResult : evalResult;
          }
          if (geoData) {
            resolve(geoData);
          } else {
            reject(t("geojson_editor.index.jiaoBenShuJuYou"));
          }
        })
      )
    }),
    [scriptContent, t]
  );
  (0, import_ahooks.useUpdateEffect)(() => {
    const features = saveEditorText();
    if (autoFitBounds) {
      bboxAutoFit(features);
    }
  }, [debouncedGeoJsonText]);
  const value = (0, import_react.useMemo)(() => {
    if (language === "javascript") {
      return {};
    }
    return { value: editorText };
  }, [language, editorText]);
  return /* @__PURE__ */ import_react.default.createElement("div", { ref: setContainer, className: styles.appEditor }, /* @__PURE__ */ import_react.default.createElement(
    import_react_monaco_editor.default,
    {
      width,
      height,
      language,
      ...value,
      onChange: monacoChange,
      theme: theme === "dark" ? "vs-dark" : "custome-theme",
      options: {
        selectOnLineNumbers: true,
        tabIndex: 2,
        tabSize: 2,
        folding: true,
        fontSize: 13,
        mouseStyle: "text",
        foldingStrategy: "indentation",
        scrollBeyondLastLine: false,
        foldingMaximumRegions: Number.MAX_SAFE_INTEGER,
        suggest: {
          showKeywords: true
        }
      }
    }
  ));
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  GeoJsonEditor
});
