// @antv/l7-editor/es/components/app-table/index.js
var _excluded = ["editable", "children", "dataIndex", "record", "inputType", "handleSave", "features", "scene", "isDraw"];
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _regeneratorRuntime() { "use strict"; /*! regenerator-runtime -- Copyright (c) 2014-present, Facebook, Inc. -- license (MIT): https://github.com/facebook/regenerator/blob/main/LICENSE */ _regeneratorRuntime = function _regeneratorRuntime() { return e; }; var t, e = {}, r = Object.prototype, n = r.hasOwnProperty, o = Object.defineProperty || function (t, e, r) { t[e] = r.value; }, i = "function" == typeof Symbol ? Symbol : {}, a = i.iterator || "@@iterator", c = i.asyncIterator || "@@asyncIterator", u = i.toStringTag || "@@toStringTag"; function define(t, e, r) { return Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }), t[e]; } try { define({}, ""); } catch (t) { define = function define(t, e, r) { return t[e] = r; }; } function wrap(t, e, r, n) { var i = e && e.prototype instanceof Generator ? e : Generator, a = Object.create(i.prototype), c = new Context(n || []); return o(a, "_invoke", { value: makeInvokeMethod(t, r, c) }), a; } function tryCatch(t, e, r) { try { return { type: "normal", arg: t.call(e, r) }; } catch (t) { return { type: "throw", arg: t }; } } e.wrap = wrap; var h = "suspendedStart", l = "suspendedYield", f = "executing", s = "completed", y = {}; function Generator() {} function GeneratorFunction() {} function GeneratorFunctionPrototype() {} var p = {}; define(p, a, function () { return this; }); var d = Object.getPrototypeOf, v = d && d(d(values([]))); v && v !== r && n.call(v, a) && (p = v); var g = GeneratorFunctionPrototype.prototype = Generator.prototype = Object.create(p); function defineIteratorMethods(t) { ["next", "throw", "return"].forEach(function (e) { define(t, e, function (t) { return this._invoke(e, t); }); }); } function AsyncIterator(t, e) { function invoke(r, o, i, a) { var c = tryCatch(t[r], t, o); if ("throw" !== c.type) { var u = c.arg, h = u.value; return h && "object" == _typeof(h) && n.call(h, "__await") ? e.resolve(h.__await).then(function (t) { invoke("next", t, i, a); }, function (t) { invoke("throw", t, i, a); }) : e.resolve(h).then(function (t) { u.value = t, i(u); }, function (t) { return invoke("throw", t, i, a); }); } a(c.arg); } var r; o(this, "_invoke", { value: function value(t, n) { function callInvokeWithMethodAndArg() { return new e(function (e, r) { invoke(t, n, e, r); }); } return r = r ? r.then(callInvokeWithMethodAndArg, callInvokeWithMethodAndArg) : callInvokeWithMethodAndArg(); } }); } function makeInvokeMethod(e, r, n) { var o = h; return function (i, a) { if (o === f) throw new Error("Generator is already running"); if (o === s) { if ("throw" === i) throw a; return { value: t, done: !0 }; } for (n.method = i, n.arg = a;;) { var c = n.delegate; if (c) { var u = maybeInvokeDelegate(c, n); if (u) { if (u === y) continue; return u; } } if ("next" === n.method) n.sent = n._sent = n.arg;else if ("throw" === n.method) { if (o === h) throw o = s, n.arg; n.dispatchException(n.arg); } else "return" === n.method && n.abrupt("return", n.arg); o = f; var p = tryCatch(e, r, n); if ("normal" === p.type) { if (o = n.done ? s : l, p.arg === y) continue; return { value: p.arg, done: n.done }; } "throw" === p.type && (o = s, n.method = "throw", n.arg = p.arg); } }; } function maybeInvokeDelegate(e, r) { var n = r.method, o = e.iterator[n]; if (o === t) return r.delegate = null, "throw" === n && e.iterator.return && (r.method = "return", r.arg = t, maybeInvokeDelegate(e, r), "throw" === r.method) || "return" !== n && (r.method = "throw", r.arg = new TypeError("The iterator does not provide a '" + n + "' method")), y; var i = tryCatch(o, e.iterator, r.arg); if ("throw" === i.type) return r.method = "throw", r.arg = i.arg, r.delegate = null, y; var a = i.arg; return a ? a.done ? (r[e.resultName] = a.value, r.next = e.nextLoc, "return" !== r.method && (r.method = "next", r.arg = t), r.delegate = null, y) : a : (r.method = "throw", r.arg = new TypeError("iterator result is not an object"), r.delegate = null, y); } function pushTryEntry(t) { var e = { tryLoc: t[0] }; 1 in t && (e.catchLoc = t[1]), 2 in t && (e.finallyLoc = t[2], e.afterLoc = t[3]), this.tryEntries.push(e); } function resetTryEntry(t) { var e = t.completion || {}; e.type = "normal", delete e.arg, t.completion = e; } function Context(t) { this.tryEntries = [{ tryLoc: "root" }], t.forEach(pushTryEntry, this), this.reset(!0); } function values(e) { if (e || "" === e) { var r = e[a]; if (r) return r.call(e); if ("function" == typeof e.next) return e; if (!isNaN(e.length)) { var o = -1, i = function next() { for (; ++o < e.length;) if (n.call(e, o)) return next.value = e[o], next.done = !1, next; return next.value = t, next.done = !0, next; }; return i.next = i; } } throw new TypeError(_typeof(e) + " is not iterable"); } return GeneratorFunction.prototype = GeneratorFunctionPrototype, o(g, "constructor", { value: GeneratorFunctionPrototype, configurable: !0 }), o(GeneratorFunctionPrototype, "constructor", { value: GeneratorFunction, configurable: !0 }), GeneratorFunction.displayName = define(GeneratorFunctionPrototype, u, "GeneratorFunction"), e.isGeneratorFunction = function (t) { var e = "function" == typeof t && t.constructor; return !!e && (e === GeneratorFunction || "GeneratorFunction" === (e.displayName || e.name)); }, e.mark = function (t) { return Object.setPrototypeOf ? Object.setPrototypeOf(t, GeneratorFunctionPrototype) : (t.__proto__ = GeneratorFunctionPrototype, define(t, u, "GeneratorFunction")), t.prototype = Object.create(g), t; }, e.awrap = function (t) { return { __await: t }; }, defineIteratorMethods(AsyncIterator.prototype), define(AsyncIterator.prototype, c, function () { return this; }), e.AsyncIterator = AsyncIterator, e.async = function (t, r, n, o, i) { void 0 === i && (i = Promise); var a = new AsyncIterator(wrap(t, r, n, o), i); return e.isGeneratorFunction(r) ? a : a.next().then(function (t) { return t.done ? t.value : a.next(); }); }, defineIteratorMethods(g), define(g, u, "Generator"), define(g, a, function () { return this; }), define(g, "toString", function () { return "[object Generator]"; }), e.keys = function (t) { var e = Object(t), r = []; for (var n in e) r.push(n); return r.reverse(), function next() { for (; r.length;) { var t = r.pop(); if (t in e) return next.value = t, next.done = !1, next; } return next.done = !0, next; }; }, e.values = values, Context.prototype = { constructor: Context, reset: function reset(e) { if (this.prev = 0, this.next = 0, this.sent = this._sent = t, this.done = !1, this.delegate = null, this.method = "next", this.arg = t, this.tryEntries.forEach(resetTryEntry), !e) for (var r in this) "t" === r.charAt(0) && n.call(this, r) && !isNaN(+r.slice(1)) && (this[r] = t); }, stop: function stop() { this.done = !0; var t = this.tryEntries[0].completion; if ("throw" === t.type) throw t.arg; return this.rval; }, dispatchException: function dispatchException(e) { if (this.done) throw e; var r = this; function handle(n, o) { return a.type = "throw", a.arg = e, r.next = n, o && (r.method = "next", r.arg = t), !!o; } for (var o = this.tryEntries.length - 1; o >= 0; --o) { var i = this.tryEntries[o], a = i.completion; if ("root" === i.tryLoc) return handle("end"); if (i.tryLoc <= this.prev) { var c = n.call(i, "catchLoc"), u = n.call(i, "finallyLoc"); if (c && u) { if (this.prev < i.catchLoc) return handle(i.catchLoc, !0); if (this.prev < i.finallyLoc) return handle(i.finallyLoc); } else if (c) { if (this.prev < i.catchLoc) return handle(i.catchLoc, !0); } else { if (!u) throw new Error("try statement without catch or finally"); if (this.prev < i.finallyLoc) return handle(i.finallyLoc); } } } }, abrupt: function abrupt(t, e) { for (var r = this.tryEntries.length - 1; r >= 0; --r) { var o = this.tryEntries[r]; if (o.tryLoc <= this.prev && n.call(o, "finallyLoc") && this.prev < o.finallyLoc) { var i = o; break; } } i && ("break" === t || "continue" === t) && i.tryLoc <= e && e <= i.finallyLoc && (i = null); var a = i ? i.completion : {}; return a.type = t, a.arg = e, i ? (this.method = "next", this.next = i.finallyLoc, y) : this.complete(a); }, complete: function complete(t, e) { if ("throw" === t.type) throw t.arg; return "break" === t.type || "continue" === t.type ? this.next = t.arg : "return" === t.type ? (this.rval = this.arg = t.arg, this.method = "return", this.next = "end") : "normal" === t.type && e && (this.next = e), y; }, finish: function finish(t) { for (var e = this.tryEntries.length - 1; e >= 0; --e) { var r = this.tryEntries[e]; if (r.finallyLoc === t) return this.complete(r.completion, r.afterLoc), resetTryEntry(r), y; } }, catch: function _catch(t) { for (var e = this.tryEntries.length - 1; e >= 0; --e) { var r = this.tryEntries[e]; if (r.tryLoc === t) { var n = r.completion; if ("throw" === n.type) { var o = n.arg; resetTryEntry(r); } return o; } } throw new Error("illegal catch attempt"); }, delegateYield: function delegateYield(e, r, n) { return this.delegate = { iterator: values(e), resultName: r, nextLoc: n }, "next" === this.method && (this.arg = t), y; } }, e; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { DeleteOutlined, PlusSquareOutlined } from '@ant-design/icons';
import { bbox, center, featureCollection } from '@turf/turf';
import { useSize } from 'ahooks';
import { Button, Empty, Form, Input, InputNumber, Popconfirm, Space, Table, Tooltip, Typography, message } from 'antd';
import classNames from 'classnames';
import { cloneDeep, isNull, isUndefined, uniqBy } from 'lodash-es';
import React, { useContext, useEffect, useMemo, useRef, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { FeatureKey } from "../../constants";
import { useFeature } from "../../recoil";
import { prettierText } from "../../utils/prettier-text";
import useStyle from "./styles";
var Text = Typography.Text;
var formatTableValue = function formatTableValue(value) {
  if (isNull(value) || isUndefined(value) || value === '') {
    return '-';
  }
  return value instanceof Object ? JSON.stringify(value) : /*#__PURE__*/React.createElement(Text, {
    ellipsis: {
      tooltip: String(value)
    }
  }, String(value));
};
var FormContext = /*#__PURE__*/React.createContext(null);
var EditableRow = function EditableRow(props) {
  var _Form$useForm = Form.useForm(),
    _Form$useForm2 = _slicedToArray(_Form$useForm, 1),
    form = _Form$useForm2[0];
  return /*#__PURE__*/React.createElement(Form, {
    form: form,
    component: false
  }, /*#__PURE__*/React.createElement(FormContext.Provider, {
    value: form
  }, /*#__PURE__*/React.createElement("tr", props)));
};
var EditableCell = function EditableCell(_ref) {
  var editable = _ref.editable,
    children = _ref.children,
    dataIndex = _ref.dataIndex,
    record = _ref.record,
    inputType = _ref.inputType,
    handleSave = _ref.handleSave,
    features = _ref.features,
    scene = _ref.scene,
    isDraw = _ref.isDraw,
    restProps = _objectWithoutProperties(_ref, _excluded);
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    editing = _useState2[0],
    setEditing = _useState2[1];
  var inputRef = useRef(null);
  var styles = useStyle();
  var form = useContext(FormContext);
  useEffect(function () {
    if (editing) {
      var _inputRef$current;
      (_inputRef$current = inputRef.current) === null || _inputRef$current === void 0 || _inputRef$current.focus();
    }
  }, [editing]);
  var toggleEdit = function toggleEdit() {
    if (scene && !isDraw) {
      setEditing(!editing);
      form === null || form === void 0 || form.setFieldsValue(inputType !== 'object' ? _defineProperty({}, dataIndex, record[dataIndex]) : _defineProperty({}, dataIndex, JSON.stringify(record[dataIndex])));
    }
  };
  var save = /*#__PURE__*/function () {
    var _ref4 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee() {
      var fieldValue, values, newValues;
      return _regeneratorRuntime().wrap(function _callee$(_context) {
        while (1) switch (_context.prev = _context.next) {
          case 0:
            _context.prev = 0;
            fieldValue = inputType !== 'object' ? _defineProperty({}, dataIndex, record[dataIndex]) : _defineProperty({}, dataIndex, JSON.stringify(record[dataIndex]));
            _context.next = 4;
            return form === null || form === void 0 ? void 0 : form.validateFields();
          case 4:
            values = _context.sent;
            if (JSON.stringify(values) !== JSON.stringify(fieldValue)) {
              newValues = inputType === 'object' ? _defineProperty({}, dataIndex, JSON.parse(values[dataIndex])) : values;
              toggleEdit();
              handleSave(_objectSpread(_objectSpread(_objectSpread({}, record), newValues), {}, {
                newValues: newValues
              }));
            } else {
              toggleEdit();
            }
            _context.next = 11;
            break;
          case 8:
            _context.prev = 8;
            _context.t0 = _context["catch"](0);
            console.log('Save failed:', _context.t0);
          case 11:
          case "end":
            return _context.stop();
        }
      }, _callee, null, [[0, 8]]);
    }));
    return function save() {
      return _ref4.apply(this, arguments);
    };
  }();
  var childNode = children;
  if (editable) {
    childNode = editing ? /*#__PURE__*/React.createElement(Form.Item, {
      style: {
        margin: 0
      },
      name: dataIndex
    }, inputType === 'number' ? /*#__PURE__*/React.createElement(InputNumber, {
      ref: inputRef,
      onPressEnter: save,
      onBlur: save
    }) : /*#__PURE__*/React.createElement(Input, {
      ref: inputRef,
      onPressEnter: save,
      onBlur: save
    })) : /*#__PURE__*/React.createElement("div", {
      className: !isDraw ? styles.editableCellValueWrap : '',
      style: {
        paddingRight: 24
      },
      onClick: toggleEdit
    }, children);
  }
  return /*#__PURE__*/React.createElement("td", restProps, childNode);
};
var components = {
  body: {
    row: EditableRow,
    cell: EditableCell
  }
};
export var AppTable = function AppTable() {
  var _useSize;
  var container = useRef(null);
  var styles = useStyle();
  var _ref8 = (_useSize = useSize(container)) !== null && _useSize !== void 0 ? _useSize : {},
    _ref8$height = _ref8.height,
    height = _ref8$height === void 0 ? 0 : _ref8$height;
  var _useFeature = useFeature(),
    saveEditorText = _useFeature.saveEditorText,
    isDraw = _useFeature.isDraw,
    scene = _useFeature.scene,
    features = _useFeature.features,
    resetFeatures = _useFeature.resetFeatures;
  var _useState3 = useState([]),
    _useState4 = _slicedToArray(_useState3, 2),
    newDataSource = _useState4[0],
    setNewDataSource = _useState4[1];
  var _useFeature2 = useFeature(),
    transformCoord = _useFeature2.transformCoord;
  var _useState5 = useState(undefined),
    _useState6 = _slicedToArray(_useState5, 2),
    addInputValue = _useState6[0],
    setAddInputValue = _useState6[1];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  useEffect(function () {
    var dataSource = features.map(function (item, index) {
      var properties = item.properties;
      return _objectSpread({
        __index: index + 1
      }, properties);
    });
    setNewDataSource(dataSource);
  }, [features]);
  var defaultColumns = useMemo(function () {
    var newColumns = [];
    var featureKeyList = Array.from(new Set(features.map(function (item) {
      var properties = item.properties;
      return Object.keys(properties);
    }).flat()));
    if (featureKeyList) {
      newColumns.push({
        title: t('app_table.index.xuHao'),
        dataIndex: '__index',
        key: "__index",
        width: 80,
        align: 'center',
        fixed: 'left',
        sorter: function sorter(a, b) {
          return a.__index - b.__index;
        }
      });
    }
    var delConfirm = function delConfirm(key) {
      var newFeatures = features.map(function (feature) {
        var properties = feature.properties;
        var newProperties = cloneDeep(properties);
        //@ts-ignore
        delete newProperties[key];
        return _objectSpread(_objectSpread({}, feature), {}, {
          properties: newProperties
        });
      });
      resetFeatures(newFeatures);
    };
    featureKeyList.forEach(function (key, index) {
      var options = uniqBy(newDataSource.map(function (item) {
        return item[key];
      }).map(function (value) {
        return {
          text: formatTableValue(value),
          value: value
        };
      }), 'text');
      newColumns.push({
        title: /*#__PURE__*/React.createElement("div", {
          className: styles.deleteColumns
        }, /*#__PURE__*/React.createElement(Tooltip, {
          title: key
        }, /*#__PURE__*/React.createElement(Text, {
          style: {
            wordWrap: 'break-word',
            whiteSpace: 'break-spaces'
          },
          ellipsis: {
            tooltip: key
          }
        }, key)), /*#__PURE__*/React.createElement(Popconfirm, {
          title: t('app_table.index.shanChuLie'),
          onConfirm: function onConfirm() {
            delConfirm(key);
          }
        }, /*#__PURE__*/React.createElement(Button, {
          type: "text",
          className: classNames([styles.delButton, 'delete'])
        }, /*#__PURE__*/React.createElement(DeleteOutlined, null)))),
        width: 200,
        align: 'center',
        dataIndex: key,
        key: "".concat(key).concat(index),
        editable: true,
        render: formatTableValue,
        filters: options.length ? options : undefined,
        onFilter: function onFilter(value, record) {
          var _record$key;
          return ((_record$key = record[key]) !== null && _record$key !== void 0 ? _record$key : '') === value;
        },
        filterSearch: true,
        sorter: !options.length ? function (a, b) {
          return (typeof a[key] === 'string' || !a[key] ? 0 : a[key]) - (typeof b[key] === 'string' || !b[key] ? 0 : b[key]);
        } : undefined
      });
    });
    var confirm = function confirm() {
      if (!addInputValue) {
        return;
      }
      var newFeatures = features.map(function (item) {
        var properties = item.properties;
        var newProperties = _objectSpread(_objectSpread({}, properties), {}, _defineProperty({}, addInputValue, ''));
        return _objectSpread(_objectSpread({}, item), {}, {
          properties: newProperties
        });
      });
      resetFeatures(newFeatures);
      setAddInputValue(undefined);
    };
    newColumns.push({
      title: /*#__PURE__*/React.createElement("div", {
        className: styles.addColumns
      }, /*#__PURE__*/React.createElement(Popconfirm, {
        placement: "bottom",
        icon: null,
        title: t('app_table.index.tianJiaLieZiDuan'),
        description: /*#__PURE__*/React.createElement("div", {
          style: {
            display: 'flex',
            alignItems: 'center'
          }
        }, /*#__PURE__*/React.createElement("div", {
          style: {
            width: 80
          }
        }, t('app_table.index.ziDuanMing')), /*#__PURE__*/React.createElement(Input, {
          placeholder: t('app_table.index.qingShuRuLieZiDuan'),
          value: addInputValue,
          onChange: function onChange(e) {
            setAddInputValue(e.target.value);
          }
        })),
        onConfirm: confirm,
        onCancel: function onCancel() {
          setAddInputValue(undefined);
        }
      }, /*#__PURE__*/React.createElement(Button, {
        type: "text",
        className: styles.addButton
      }, /*#__PURE__*/React.createElement(PlusSquareOutlined, null))), t('app_table.index.caoZuo')),
      key: 'action',
      width: 120,
      align: 'center',
      fixed: 'right',
      render: function render(_, record) {
        return /*#__PURE__*/React.createElement(Space, {
          size: "middle"
        }, /*#__PURE__*/React.createElement("a", {
          onClick: function onClick() {
            if (scene) {
              var targetFeature = features.find(function (item) {
                return item.properties[FeatureKey.Index] === record[FeatureKey.Index];
              });
              if (targetFeature) {
                var newBboxFit = transformCoord([targetFeature]);
                if (newBboxFit[0].geometry.type === 'Point') {
                  var content = center(newBboxFit[0]);
                  scene.setCenter(content.geometry.coordinates);
                } else {
                  var _content = bbox(newBboxFit[0]);
                  scene.fitBounds([[_content[0], _content[1]], [_content[2], _content[3]]]);
                }
              }
            }
          }
        }, t('app_table.index.dingWei')), /*#__PURE__*/React.createElement("a", {
          onClick: function onClick() {
            resetFeatures(features.filter(function (feature, index) {
              return index !== record[FeatureKey.Index];
            }));
            message.success(t('app_table.index.shuJuShanChuCheng'));
          },
          style: {
            color: '#ff0000'
          }
        }, t('app_table.index.shanChu')));
      }
    });
    return newColumns;
  }, [addInputValue, features, newDataSource, resetFeatures, scene, styles.addButton, styles.addColumns, styles.delButton, styles.deleteColumns, t, transformCoord]);
  var handleSave = function handleSave(row) {
    var newData = _toConsumableArray(newDataSource);
    var index = newData.findIndex(function (item) {
      return row.__index === (item === null || item === void 0 ? void 0 : item.__index);
    });
    var item = newData[index];
    newData.splice(index, 1, _objectSpread(_objectSpread({}, item), row));
    var __index = row.__index,
      newValues = row.newValues;
    var indexProperties = _objectSpread(_objectSpread({}, features[__index - 1].properties), newValues);
    var feature = _objectSpread(_objectSpread({}, features[__index - 1]), {}, {
      properties: indexProperties
    });
    features[index] = feature;
    saveEditorText(prettierText({
      content: featureCollection(features)
    }));
    setNewDataSource(newData);
  };
  var newColumns = defaultColumns.map(function (col) {
    if (!col.editable) {
      return col;
    }
    return _objectSpread(_objectSpread({}, col), {}, {
      onCell: function onCell(record) {
        return {
          record: record,
          editable: col.editable,
          dataIndex: col.dataIndex,
          title: col.title,
          inputType: _typeof(record[col.dataIndex]),
          newDataSource: newDataSource,
          features: features,
          scene: scene,
          isDraw: isDraw,
          handleSave: handleSave
        };
      }
    });
  });
  return /*#__PURE__*/React.createElement("div", {
    style: {
      width: '100%',
      height: '100%'
    },
    ref: container
  }, newDataSource !== null && newDataSource !== void 0 && newDataSource.length ? /*#__PURE__*/React.createElement(Table, {
    className: styles.tableContent,
    components: components,
    columns: newColumns,
    rowClassName: function rowClassName() {
      return 'editable-row';
    },
    dataSource: newDataSource,
    bordered: true,
    scroll: {
      y: height - 54,
      x: 'max-content'
    },
    size: "small",
    pagination: false,
    rowKey: '__index'
  }) : /*#__PURE__*/React.createElement(Empty, {
    description: t('app_table.index.dangQianShuJuWu'),
    style: {
      margin: '12px 0'
    }
  }));
};
