// @antv/l7-editor/es/components/app-header/btn/import-btn/file-upload.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
import { FileTextOutlined, QuestionCircleOutlined } from '@ant-design/icons';
import { feature, featureCollection } from '@turf/turf';
import { Form, Select, Tooltip, Upload, message } from 'antd';
import { cloneDeep } from 'lodash';
import React, { forwardRef, useImperativeHandle, useMemo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { parse } from 'wellknown';
import { FeatureCollectionVT } from "../../../../constants/variable-type";
import { isGeometry, isWkt, parserFileToSource } from "../../../../utils/upload";
import useStyle from "../../styles";
var Dragger = Upload.Dragger;
var FileUpload = /*#__PURE__*/forwardRef(function FileUpload(_ref, ref) {
  _objectDestructuringEmpty(_ref);
  var _useState = useState([]),
    _useState2 = _slicedToArray(_useState, 2),
    uploadData = _useState2[0],
    setUploadData = _useState2[1];
  var _useState3 = useState([]),
    _useState4 = _slicedToArray(_useState3, 2),
    fileList = _useState4[0],
    setFileList = _useState4[1];
  var _useState5 = useState([]),
    _useState6 = _slicedToArray(_useState5, 2),
    selectList = _useState6[0],
    setSelectList = _useState6[1];
  var _Form$useForm = Form.useForm(),
    _Form$useForm2 = _slicedToArray(_Form$useForm, 1),
    form = _Form$useForm2[0];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var styles = useStyle();
  var customRequest = function customRequest(uploadRequestOption) {
    var file = uploadRequestOption.file,
      onSuccess = uploadRequestOption.onSuccess,
      onError = uploadRequestOption.onError;
    parserFileToSource(file, t).then(function (dataSource) {
      if (!(dataSource !== null && dataSource !== void 0 && dataSource.columns)) {
        var _dataSource$data;
        if ((_dataSource$data = dataSource.data) !== null && _dataSource$data !== void 0 && (_dataSource$data = _dataSource$data.features) !== null && _dataSource$data !== void 0 && _dataSource$data.length) {
          var newData = uploadData;
          newData.push({
            id: dataSource.id,
            features: dataSource.data.features
          });
          setUploadData(newData);
        }
      } else {
        var wktValue = dataSource.data[0];
        var val = undefined;
        for (var _i = 0, _Object$keys = Object.keys(wktValue); _i < _Object$keys.length; _i++) {
          var key = _Object$keys[_i];
          var values = wktValue[key];
          if (isWkt(values)) {
            form.setFieldValue(dataSource.id, key);
            val = key;
          } else if (isGeometry(values)) {
            form.setFieldValue(dataSource.id, key);
            val = key;
          }
        }
        if (val) {
          var data = dataSource === null || dataSource === void 0 ? void 0 : dataSource.data.map(function (value) {
            //@ts-ignore
            return value[val];
          });
          var propertiesList = dataSource === null || dataSource === void 0 ? void 0 : dataSource.data.map(function (v) {
            var properties = cloneDeep(v);
            //@ts-ignore
            delete properties[val];
            return _objectSpread({}, properties);
          });
          var newGeoJson = [];
          if (isWkt(data[0])) {
            newGeoJson = data.map(function (v, index) {
              var geometry = parse(v);
              return feature(geometry, _objectSpread({}, propertiesList[index]));
            });
          } else if (isGeometry(data[0])) {
            newGeoJson = data.map(function (v, index) {
              return feature(JSON.parse(v), _objectSpread({}, propertiesList[index]));
            });
          }
          var newDates = uploadData;
          newDates.push({
            features: newGeoJson,
            id: dataSource === null || dataSource === void 0 ? void 0 : dataSource.id
          });
          setUploadData(newDates);
        }
        setSelectList(function (pre) {
          return [].concat(_toConsumableArray(pre), [dataSource]);
        });
      }
      // @ts-ignore
      onSuccess();
    }).catch(function () {
      onError();
      message.error(t('import_btn.file_upload.shuJuGeShiBu'));
    });
  };
  useImperativeHandle(ref, function () {
    return {
      getData: function getData() {
        return new Promise(function (resolve, reject) {
          if (!uploadData.length) {
            reject(t('import_btn.file_upload.qingTianJiaWenJian'));
          }
          var isErrorList = fileList.filter(function (item) {
            return item.status === 'error';
          });
          if (!!isErrorList.length) {
            reject(t('import_btn.file_upload.qingShanChuBaoCuo'));
          }
          var data = uploadData.map(function (item) {
            return item.features;
          }).flat();
          if (FeatureCollectionVT.check(featureCollection(data))) {
            resolve(featureCollection(data));
          } else {
            message.info(t('import_btn.file_upload.qingJianChaShuJu'));
          }
        });
      }
    };
  }, [fileList, t, uploadData]);
  var selectItem = useMemo(function () {
    if (selectList.length) {
      var data = selectList.map(function (item, index) {
        var options = item.columns.map(function (option) {
          return {
            value: option,
            label: option
          };
        });
        return /*#__PURE__*/React.createElement(Form.Item, {
          name: item.id,
          label: "".concat(t('import_btn.file_upload.wenJian')).concat(index + 1),
          key: item.id
        }, /*#__PURE__*/React.createElement(Select, {
          options: options,
          style: {
            width: 300
          },
          onChange: function onChange(e) {
            var newData1 = selectList.find(function (v) {
              return v.id === item.id;
            });
            var newData2 = newData1 === null || newData1 === void 0 ? void 0 : newData1.data.map(function (value) {
              return value[e];
            });
            var propertiesList = newData1 === null || newData1 === void 0 ? void 0 : newData1.data.map(function (v) {
              var properties = cloneDeep(v);
              delete properties[e];
              return _objectSpread({}, properties);
            });
            var newGeoJson = [];
            if (isWkt(newData2[0])) {
              newGeoJson = newData2.map(function (v, geojsonIndex) {
                var geometry = parse(v);
                return feature(geometry, _objectSpread({}, propertiesList[geojsonIndex]));
              });
            } else if (isGeometry(newData2[0])) {
              newGeoJson = newData2.map(function (v, geojsonIndex) {
                return feature(JSON.parse(v), _objectSpread({}, propertiesList[geojsonIndex]));
              });
            } else {
              message.error(t('import_btn.file_upload.ciZiDuanFeiDiLiZiDuan'));
            }
            var newDates = uploadData.filter(function (updateItem) {
              return updateItem.id !== (newData1 === null || newData1 === void 0 ? void 0 : newData1.id);
            });
            newDates.push({
              features: newGeoJson,
              id: newData1 === null || newData1 === void 0 ? void 0 : newData1.id
            });
            setUploadData(newDates);
          }
        }));
      });
      return data;
    }
  }, [selectList, t, uploadData]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Form, {
    form: form
  }, /*#__PURE__*/React.createElement(Form.Item, {
    name: "file",
    rules: [{
      required: true
    }],
    style: {
      marginTop: 16,
      marginBottom: 4
    }
  }, /*#__PURE__*/React.createElement(Dragger, {
    accept: ".json,.geojson,.kml,.wkt,.csv,.xlsx,.xls",
    customRequest: customRequest,
    multiple: true,
    onRemove: function onRemove(file) {
      var newData = uploadData.filter(function (item) {
        return item.id !== file.uid;
      });
      setUploadData(newData);
      var newSelect = selectList.filter(function (item) {
        return item.id !== file.uid;
      });
      setSelectList(newSelect);
      return true;
    },
    onChange: function onChange(file) {
      setFileList(file.fileList);
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.upload
  }, /*#__PURE__*/React.createElement("span", {
    style: {
      fontSize: 40
    }
  }, /*#__PURE__*/React.createElement(FileTextOutlined, null)), /*#__PURE__*/React.createElement("div", null, t('import_btn.file_upload.dianJiHuoJiangWenJian'))))), /*#__PURE__*/React.createElement("div", {
    style: {
      color: '#777'
    }
  }, t('import_btn.file_upload.jinZhiChiJS')), selectList.length ? /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex'
    }
  }, /*#__PURE__*/React.createElement("span", {
    className: styles.uploadTitle
  }, t('import_btn.file_upload.diLiZiDuanXuanZe')), /*#__PURE__*/React.createElement(Tooltip, {
    title: t('import_btn.file_upload.dangQianJinZhiChi')
  }, /*#__PURE__*/React.createElement(QuestionCircleOutlined, null))) : null, selectItem));
});
export default FileUpload;
