// @antv/l7-editor/es/components/app-header/btn/import-btn/lnglat-import-btn.js
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
import { featureCollection } from '@turf/turf';
import { Form, Input, Radio } from 'antd';
import React, { forwardRef, useImperativeHandle } from 'react';
import { useTranslation } from 'react-i18next';
import { useLnglat } from "../../../../recoil";
var LngLatImportBtn = /*#__PURE__*/forwardRef(function (_ref, ref) {
  _objectDestructuringEmpty(_ref);
  var _useLnglat = useLnglat(),
    lngLatText = _useLnglat.lngLatText,
    setLngLatText = _useLnglat.setLngLatText,
    importLngLatText = _useLnglat.importLngLatText,
    lngLatImportType = _useLnglat.lngLatImportType,
    setLngLatImportType = _useLnglat.setLngLatImportType;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var LngLatImportTypeOptions = [{
    label: t('import_btn.lnglat_import_btn.dian'),
    value: 'Point'
  }, {
    label: t('import_btn.lnglat_import_btn.xian'),
    value: 'LingString'
  }, {
    label: t('import_btn.lnglat_import_btn.mian'),
    value: 'Polygon'
  }];
  useImperativeHandle(ref, function () {
    return {
      getData: function getData() {
        return new Promise(function (resolve, reject) {
          if (!lngLatText) {
            reject(t('import_btn.lnglat_import_btn.qingShuRuJingWei'));
          }
          var data = importLngLatText(lngLatText);
          resolve(featureCollection(data));
          reject(t('import_btn.lnglat_import_btn.lNGLA'));
        });
      }
    };
  }, [lngLatText, importLngLatText, t]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Form, null, /*#__PURE__*/React.createElement(Form.Item, {
    style: {
      marginTop: 16
    },
    label: t('import_btn.lnglat_import_btn.shuJuLeiXing')
  }, /*#__PURE__*/React.createElement(Radio.Group, {
    value: lngLatImportType,
    buttonStyle: "solid",
    onChange: function onChange(e) {
      setLngLatImportType(e.target.value);
    }
  }, LngLatImportTypeOptions.map(function (item) {
    return /*#__PURE__*/React.createElement(Radio.Button, {
      key: item.value,
      value: item.value
    }, item.label);
  }))), /*#__PURE__*/React.createElement(Form.Item, {
    style: {
      marginTop: 16
    },
    label: t('import_btn.lnglat_import_btn.shuJuNeiRong')
  }, /*#__PURE__*/React.createElement(Input.TextArea, {
    placeholder: t('import_btn.lnglat_import_btn.qingShuRuLianXu'),
    rows: 10,
    onChange: function onChange(e) {
      setLngLatText(e.target.value);
    }
  }))));
});
export default LngLatImportBtn;
