// @antv/l7-editor/es/components/app-header/btn/import-btn/url-upload.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
import { Form, Input, Radio } from 'antd';
import React, { forwardRef, useImperativeHandle, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { getUrlFeatureCollection } from "../../../../utils";
var UrlUpload = /*#__PURE__*/forwardRef(function (_ref, ref) {
  _objectDestructuringEmpty(_ref);
  var _useState = useState(''),
    _useState2 = _slicedToArray(_useState, 2),
    inputValue = _useState2[0],
    setInputValue = _useState2[1];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var _useState3 = useState('GeoJSON'),
    _useState4 = _slicedToArray(_useState3, 2),
    radioValue = _useState4[0],
    setRadioValue = _useState4[1];
  useImperativeHandle(ref, function () {
    return {
      getData: function getData() {
        return new Promise(function (resolve, reject) {
          if (inputValue) {
            resolve(getUrlFeatureCollection(inputValue, radioValue, t));
            reject(t('import_btn.url_upload.shuJuGeShiCuo'));
          } else {
            reject(t('import_btn.url_upload.qingShuRuWenBen'));
          }
        });
      }
    };
  }, [inputValue, radioValue, t]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Form, {
    layout: 'vertical'
  }, /*#__PURE__*/React.createElement(Form.Item, {
    name: "urlType",
    label: t('import_btn.url_upload.shuJuLeiXing'),
    rules: [{
      required: true
    }],
    style: {
      marginTop: 16,
      marginBottom: 4
    }
  }, /*#__PURE__*/React.createElement(Radio.Group, {
    defaultValue: "GeoJSON",
    onChange: function onChange(e) {
      setRadioValue(e.target.value);
    }
  }, /*#__PURE__*/React.createElement(Radio.Button, {
    value: "GeoJSON"
  }, "GeoJSON"), /*#__PURE__*/React.createElement(Radio.Button, {
    value: "WKT"
  }, "WKT"), /*#__PURE__*/React.createElement(Radio.Button, {
    value: "KML"
  }, "KML"))), /*#__PURE__*/React.createElement(Form.Item, {
    name: "url",
    label: t('import_btn.url_upload.uRLDiZhi'),
    rules: [{
      required: true
    }],
    style: {
      marginTop: 16,
      marginBottom: 4
    }
  }, /*#__PURE__*/React.createElement(Input, {
    placeholder: "https://...",
    onChange: function onChange(e) {
      setInputValue(e.target.value);
      // checkWithRestData(e.target.value);
    }
  }))));
});
export default UrlUpload;
