// @antv/l7-editor/lib/constants/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/constants/index.ts
var constants_exports = {};
__export(constants_exports, {
  FeatureKey: () => FeatureKey,
  GOOGLE_TILE_MAP_ROUTER_URL: () => GOOGLE_TILE_MAP_ROUTER_URL,
  GOOGLE_TILE_MAP_URL: () => GOOGLE_TILE_MAP_URL,
  GaodeConfig: () => GaodeConfig,
  LayerId: () => LayerId,
  LayerZIndex: () => LayerZIndex,
  LocalStorageKey: () => LocalStorageKey,
  MapBoxConfig: () => MapBoxConfig,
  OfficeLayerEnum: () => OfficeLayerEnum,
  PrimaryColor: () => PrimaryColor,
  RightPanelWidthRange: () => RightPanelWidthRange,
  dingGroupImg: () => dingGroupImg
});
module.exports = __toCommonJS(constants_exports);
__reExport(constants_exports, require("./variable-type"), module.exports);
var RightPanelWidthRange = [20, 80];
var LocalStorageKey = /* @__PURE__ */ ((LocalStorageKey2) => {
  LocalStorageKey2["RightPanelWidth"] = "RightPanelWidth";
  LocalStorageKey2["MapOptions"] = "MapOptions";
  LocalStorageKey2["LayerColor"] = "LayerColor";
  LocalStorageKey2["HideEditor"] = "HideEditor";
  LocalStorageKey2["EditorText"] = "EditorText";
  LocalStorageKey2["PopupTrigger"] = "PopupTrigger";
  LocalStorageKey2["ActiveRightTabKey"] = "ActiveRightTabKey";
  LocalStorageKey2["AutoFitBounds"] = "AutoFitBounds";
  LocalStorageKey2["BaseMap"] = "BaseMap";
  LocalStorageKey2["officialLayers"] = "officialLayers";
  LocalStorageKey2["WktText"] = "WktText";
  LocalStorageKey2["Convert"] = "Convert";
  LocalStorageKey2["theme"] = "theme";
  LocalStorageKey2["cityHistory"] = "cityHistory";
  LocalStorageKey2["showTextLayer"] = "showTextLayer";
  LocalStorageKey2["textLayerFields"] = "textLayerFields";
  LocalStorageKey2["locale"] = "locale";
  LocalStorageKey2["firstOpening"] = "firstOpening";
  LocalStorageKey2["wasmPath"] = "wasmPath";
  LocalStorageKey2["customTiles"] = "customTiles";
  LocalStorageKey2["showDrawDistance"] = "showDrawDistance";
  LocalStorageKey2["showDrawArea"] = "showDrawArea";
  return LocalStorageKey2;
})(LocalStorageKey || {});
var LayerId = /* @__PURE__ */ ((LayerId2) => {
  LayerId2["PointLayer"] = "PointLayer";
  LayerId2["LineLayer"] = "LineLayer";
  LayerId2["PolygonLayer"] = "PolygonLayer";
  return LayerId2;
})(LayerId || {});
var dingGroupImg = "https://mdn.alipayobjects.com/huamei_baaa7a/afts/img/A*mvwjR5RwS9QAAAAAAAAAAAAADqSCAQ/original";
var PrimaryColor = "#1677ff";
var FeatureKey = {
  Index: Symbol("index"),
  DrawType: Symbol("drawType"),
  IsEdit: Symbol("isEdit")
};
var MapBoxConfig = {
  token: "pk.eyJ1IjoibGl1dmlnb25nenVvc2hpIiwiYSI6ImNsaGs2czBrcTBvczUzbnFzOHU0dzk2ZWQifQ.hVvTgcbg_Ym-VQz36psLRg",
  WebGLParams: {
    preserveDrawingBuffer: true
  }
};
var GaodeConfig = {
  token: "5ae4492ef912cbbc93034fea0e66ff2a",
  WebGLParams: {
    preserveDrawingBuffer: true
  }
};
var LayerZIndex = 100;
var GOOGLE_TILE_MAP_URL = "https://gac-geo.googlecnapps.cn/maps/vt?lyrs=s&gl=CN&x={x}&y={y}&z={z}";
var GOOGLE_TILE_MAP_ROUTER_URL = "https://gac-geo.googlecnapps.cn/maps/vt?lyrs=h&gl=CN&x={x}&y={y}&z={z}";
var OfficeLayerEnum = /* @__PURE__ */ ((OfficeLayerEnum2) => {
  OfficeLayerEnum2["GoogleSatellite"] = "googleSatellite";
  OfficeLayerEnum2["VectorMap"] = "vectorMap";
  return OfficeLayerEnum2;
})(OfficeLayerEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  FeatureKey,
  GOOGLE_TILE_MAP_ROUTER_URL,
  GOOGLE_TILE_MAP_URL,
  GaodeConfig,
  LayerId,
  LayerZIndex,
  LocalStorageKey,
  MapBoxConfig,
  OfficeLayerEnum,
  PrimaryColor,
  RightPanelWidthRange,
  dingGroupImg,
  ...require("./variable-type")
});
