// @antv/l7-editor/lib/constants/variable-type.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/constants/variable-type.ts
var variable_type_exports = {};
__export(variable_type_exports, {
  FeatureCollectionVT: () => FeatureCollectionVT,
  FeatureListVT: () => FeatureListVT,
  FeatureVT: () => FeatureVT,
  GeometryListVT: () => GeometryListVT,
  GeometryVT: () => GeometryVT,
  LngLatVT: () => LngLatVT
});
module.exports = __toCommonJS(variable_type_exports);
var import_variable_type = __toESM(require("variable-type"));
var GeometryVT = import_variable_type.default.shape({
  type: import_variable_type.default.in([
    "Point",
    "LineString",
    "Polygon",
    "MultiPoint",
    "MultiLineString",
    "MultiPolygon"
  ]),
  coordinates: import_variable_type.default.array
});
var GeometryListVT = import_variable_type.default.arrayOf(GeometryVT);
var FeatureVT = import_variable_type.default.shape({
  type: import_variable_type.default.in(["Feature"]),
  properties: import_variable_type.default.or([import_variable_type.default.object, import_variable_type.default.null, import_variable_type.default.undefined]),
  geometry: GeometryVT
});
var FeatureListVT = import_variable_type.default.arrayOf(FeatureVT);
var FeatureCollectionVT = import_variable_type.default.shape({
  type: import_variable_type.default.in(["FeatureCollection"]),
  features: import_variable_type.default.arrayOf(FeatureVT)
});
var LngLatVT = import_variable_type.default.arrayOf(
  import_variable_type.default.arrayOf(
    import_variable_type.default.and([
      import_variable_type.default.arrayOf(import_variable_type.default.number),
      import_variable_type.default.apply((position) => {
        if (position.length === 2) {
          const [lng, lat] = position;
          return lng >= -180 && lng <= 180 && lat <= 90 && lat >= -90;
        }
        return false;
      })
    ])
  )
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  FeatureCollectionVT,
  FeatureListVT,
  FeatureVT,
  GeometryListVT,
  GeometryVT,
  LngLatVT
});
