// @antv/l7-editor/es/components/map-control-group/auto-control/index.js
import { CustomControl } from '@antv/larkmap';
import { Tooltip } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { IconFont } from "../../iconfont";
import { useFeature } from "../../../recoil";
import useStyles from "../styles";
export var AutoControl = function AutoControl() {
  var styles = useStyles();
  var _useFeature = useFeature(),
    bboxAutoFit = _useFeature.bboxAutoFit;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  return /*#__PURE__*/React.createElement(CustomControl, {
    position: "bottomright"
  }, /*#__PURE__*/React.createElement(Tooltip, {
    title: t('auto_control.index.ziDongSuoFangZhi'),
    placement: "left"
  }, /*#__PURE__*/React.createElement("button", {
    type: "button",
    id: "l7-editor-auto",
    className: styles.L7EditorControl,
    onClick: function onClick() {
      bboxAutoFit();
    }
  }, /*#__PURE__*/React.createElement(IconFont, {
    type: "icon-suofang",
    className: styles.l7EditorIcon
  }))));
};
