// @antv/l7-editor/lib/components/map-control-group/export-image-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/export-image-control/index.tsx
var export_image_control_exports = {};
__export(export_image_control_exports, {
  ExportImage: () => ExportImage
});
module.exports = __toCommonJS(export_image_control_exports);
var import_larkmap = require("@antv/larkmap");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_utils = require(".././../../utils");
var import_styles = __toESM(require("./styles"));
var ExportImage = () => {
  const styles = (0, import_styles.default)();
  const [imageData, setImageData] = (0, import_react.useState)("");
  const [isModalOpen, setIsModalOpen] = (0, import_react.useState)(false);
  const { t } = (0, import_react_i18next.useTranslation)();
  const onExport = (value) => {
    setIsModalOpen(true);
    setImageData(value);
  };
  const handle = () => {
    setIsModalOpen(false);
    setImageData("");
  };
  const onOk = () => {
    setIsModalOpen(false);
    (0, import_utils.downloadImg)(imageData);
    setImageData("");
  };
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.ExportImageControl,
    {
      onExport,
      className: styles.l7ExportImg,
      imageType: "jpeg",
      position: "bottomright"
    }
  ), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Modal,
    {
      title: t("export-img-control_title"),
      open: isModalOpen,
      okText: t("btn.download_btn.xiaZai"),
      onOk,
      onCancel: handle
    },
    /* @__PURE__ */ import_react.default.createElement(import_antd.Image, { src: imageData })
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ExportImage
});
