// @antv/l7-editor/es/components/map-control-group/filter-form-list-control/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { DeleteOutlined, FieldBinaryOutlined, FieldStringOutlined, FilterFilled, FilterOutlined, PlusOutlined } from '@ant-design/icons';
import { CustomControl } from '@antv/larkmap';
import { Button, Form, Select, Tooltip } from 'antd';
import { cloneDeep, debounce, isEmpty } from 'lodash-es';
import React, { useMemo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { isEmptyFilter } from "../../../hooks/use-filter-features";
import { useFeature, useFilter } from "../../../recoil";
import useStyles from "../styles";
import NumberFilter from "./numberFilter";
import StringFilter from "./stringFilter";
import useStyle from "./styles";
var Option = Select.Option;
var FilterControl = function FilterControl() {
  var _useFeature = useFeature(),
    dataSource = _useFeature.dataSource;
  var _useFilter = useFilter(),
    setFilters = _useFilter.setFilters,
    filters = _useFilter.filters;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isVisible = _useState2[0],
    setIsVisible = _useState2[1];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var _Form$useForm = Form.useForm(),
    _Form$useForm2 = _slicedToArray(_Form$useForm, 1),
    form = _Form$useForm2[0];
  var styles = useStyle();
  var style = useStyles();
  var addVisible = useMemo(function () {
    return !dataSource.length;
  }, [dataSource]);
  var onValuesChange = function onValuesChange(_, all) {
    if (isEmpty(all.filterFromList)) {
      setFilters([]);
      return;
    }
    var newValue = all.filterFromList.filter(function (item) {
      return item;
    }).map(function (item) {
      var logic = item.logic,
        operator = item.operator;
      if (item.field) {
        var _JSON$parse = JSON.parse(item.field || ''),
          field = _JSON$parse.field,
          type = _JSON$parse.type;
        if (item.operator === 'BETWEEN') {
          if (item.min && item.max) {
            return {
              logic: logic,
              field: field,
              type: type,
              operator: operator,
              value: [item.min, item.max]
            };
          }
        }
        if ((item.operator === 'IN' || item.operator === 'NO_IN') && item.value) {
          return _objectSpread(_objectSpread({}, item), {}, {
            field: field,
            type: type,
            value: item.value.length ? item.value : undefined
          });
        }
        return _objectSpread(_objectSpread({}, item), {}, {
          field: field,
          type: type
        });
      } else {
        return _objectSpread(_objectSpread({}, item), {}, {
          field: undefined,
          type: undefined
        });
      }
    });
    setFilters(newValue);
  };
  var isFilterActive = useMemo(function () {
    return filters.filter(function (item) {
      return !isEmptyFilter(item);
    }).find(function (item) {
      return item;
    });
  }, [filters]);
  return /*#__PURE__*/React.createElement(CustomControl, {
    position: "bottomright",
    className: styles.l7FilterPanel,
    style: {
      display: 'flex'
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.l7Filter,
    style: {
      display: isVisible ? 'block' : 'none'
    }
  }, /*#__PURE__*/React.createElement(Form, {
    style: {
      width: '100%'
    },
    form: form,
    onValuesChange: debounce(onValuesChange, 500)
  }, /*#__PURE__*/React.createElement(Form.List, {
    name: "filterFromList"
  }, function (fields, _ref) {
    var add = _ref.add,
      remove = _ref.remove;
    return /*#__PURE__*/React.createElement(React.Fragment, null, fields.map(function (_ref2, index) {
      var _dataSource$, _dataSource$2;
      var name = _ref2.name;
      return /*#__PURE__*/React.createElement("div", {
        key: name,
        style: {
          display: 'flex',
          marginBottom: 8
        }
      }, /*#__PURE__*/React.createElement(Form.Item, {
        name: [name, 'logic'],
        initialValue: "and"
      }, /*#__PURE__*/React.createElement(Select, {
        style: {
          width: 70,
          marginRight: '8px'
        }
      }, /*#__PURE__*/React.createElement(Option, {
        value: "and"
      }, t('filter_form_list_control.index.bingQie')), /*#__PURE__*/React.createElement(Option, {
        value: "or"
      }, t('filter_form_list_control.index.huoZhe')))), /*#__PURE__*/React.createElement(Form.Item, {
        name: [name, 'field'],
        initialValue: JSON.stringify({
          field: (_dataSource$ = dataSource[0]) === null || _dataSource$ === void 0 ? void 0 : _dataSource$.field,
          type: (_dataSource$2 = dataSource[0]) === null || _dataSource$2 === void 0 ? void 0 : _dataSource$2.type
        })
      }, /*#__PURE__*/React.createElement(Select, {
        style: {
          width: 130,
          marginRight: '8px'
        },
        placeholder: t('filter_form_list_control.index.qingXuanZeZiDuan'),
        onChange: function onChange() {
          var newFilterFromList = cloneDeep(form.getFieldValue('filterFromList'));
          newFilterFromList.forEach(function (v, i) {
            if (index === i) {
              v.operator = undefined;
              v.value = undefined;
            }
          });
          form.setFieldValue('filterFromList', newFilterFromList);
          setFilters(newFilterFromList.map(function (item) {
            var _JSON$parse2 = JSON.parse(item.field),
              field = _JSON$parse2.field,
              type = _JSON$parse2.type;
            return _objectSpread(_objectSpread({}, item), {}, {
              field: field,
              type: type
            });
          }));
        }
      }, dataSource.map(function (_ref3) {
        var field = _ref3.field,
          type = _ref3.type;
        return /*#__PURE__*/React.createElement(Option, {
          key: field,
          value: JSON.stringify({
            field: field,
            type: type
          })
        }, /*#__PURE__*/React.createElement("i", {
          style: {
            fontSize: 20,
            marginRight: 8,
            color: '#999'
          }
        }, type === 'number' ? /*#__PURE__*/React.createElement(FieldBinaryOutlined, null) : /*#__PURE__*/React.createElement(FieldStringOutlined, null)), /*#__PURE__*/React.createElement(Tooltip, {
          title: field
        }, field));
      }))), /*#__PURE__*/React.createElement(Form.Item, {
        shouldUpdate: function shouldUpdate(prevValues, curValues) {
          return prevValues.field === curValues.field;
        },
        style: {
          margin: 0
        }
      }, function (_ref4) {
        var getFieldsValue = _ref4.getFieldsValue;
        var _getFieldsValue = getFieldsValue(),
          filterFromList = _getFieldsValue.filterFromList;
        if (filterFromList[index].field) {
          var _JSON$parse3 = JSON.parse(filterFromList[index].field),
            type = _JSON$parse3.type;
          if (type === 'number') {
            return /*#__PURE__*/React.createElement(NumberFilter, {
              name: name,
              index: index,
              form: form
            });
          }
          return /*#__PURE__*/React.createElement(StringFilter, {
            name: name,
            index: index,
            form: form
          });
        }
      }), /*#__PURE__*/React.createElement(Button, {
        type: "text",
        onClick: function onClick() {
          remove(name);
          setFilters(form.getFieldValue('filterFromList'));
        },
        icon: /*#__PURE__*/React.createElement(DeleteOutlined, null)
      }));
    }), /*#__PURE__*/React.createElement(Button, {
      type: "dashed",
      disabled: addVisible,
      onClick: function onClick() {
        add();
      },
      block: true,
      icon: /*#__PURE__*/React.createElement(PlusOutlined, null),
      style: {
        width: 500
      }
    }, t('filter_form_list_control.index.tianJiaShaiXuanTiao')));
  }))), /*#__PURE__*/React.createElement(Tooltip, {
    title: t('filter_form_list_control.index.shuJuGuoLu'),
    placement: "left"
  }, /*#__PURE__*/React.createElement("button", {
    type: "button",
    id: "l7-editor-filter",
    className: style.L7EditorControl,
    onClick: function onClick() {
      setIsVisible(!isVisible);
    }
  }, !isFilterActive ? /*#__PURE__*/React.createElement(FilterOutlined
  // className="l7-draw-icon"
  , {
    style: {
      lineHeight: '30px'
    }
  }) : /*#__PURE__*/React.createElement(FilterFilled, {
    style: {
      color: '#597ef7'
    }
  }))));
};
export default FilterControl;
