// @antv/l7-editor/es/components/map-control-group/filter-form-list-control/numberFilter.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { Form, InputNumber, Select } from 'antd';
import { cloneDeep } from 'lodash-es';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useFeature, useFilter } from "../../../recoil";
var NumberFilter = function NumberFilter(_ref) {
  var name = _ref.name,
    index = _ref.index,
    form = _ref.form;
  var _useFilter = useFilter(),
    setFilters = _useFilter.setFilters;
  var _useFeature = useFeature(),
    dataSource = _useFeature.dataSource;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var select = [{
    label: '>',
    value: '>'
  }, {
    label: '>=',
    value: '>='
  }, {
    label: '=',
    value: '='
  }, {
    label: '<=',
    value: '<='
  }, {
    label: '<',
    value: '<'
  }, {
    label: t('filter_form_list_control.numberFilter.quJian'),
    value: 'BETWEEN'
  }];
  return /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex'
    }
  }, /*#__PURE__*/React.createElement(Form.Item, {
    name: [name, 'operator']
  }, /*#__PURE__*/React.createElement(Select, {
    style: {
      width: '100px',
      marginRight: '8px'
    },
    placeholder: t('filter_form_list_control.numberFilter.qingXuanZeGuoLu'),
    options: select,
    onChange: function onChange() {
      var newFilterFromList = cloneDeep(form.getFieldValue('filterFromList'));
      newFilterFromList.forEach(function (v, i) {
        if (index === i) {
          v.value = undefined;
        }
      });
      form.setFieldValue('filterFromList', newFilterFromList);
      setFilters(newFilterFromList.map(function (item) {
        var _JSON$parse = JSON.parse(item.field),
          field = _JSON$parse.field,
          type = _JSON$parse.type;
        return _objectSpread(_objectSpread({}, item), {}, {
          field: field,
          type: type
        });
      }));
    }
  })), /*#__PURE__*/React.createElement(Form.Item, {
    style: {
      width: 150,
      marginBottom: 0
    },
    shouldUpdate: function shouldUpdate(prevValues, curValues) {
      return prevValues.operator === curValues.operator;
    }
  }, function (_ref2) {
    var _JSON$parse2, _DataList$min3, _DataList$max3;
    var getFieldsValue = _ref2.getFieldsValue;
    var _getFieldsValue = getFieldsValue(),
      filterFromList = _getFieldsValue.filterFromList;
    var fieldValue = (_JSON$parse2 = JSON.parse(filterFromList[index].field)) === null || _JSON$parse2 === void 0 ? void 0 : _JSON$parse2.field;
    var DataList = dataSource.find(function (item) {
      return (item === null || item === void 0 ? void 0 : item.field) === fieldValue;
    });
    if (filterFromList[index].operator === 'BETWEEN') {
      var _DataList$min, _DataList$max, _DataList$min2, _DataList$max2;
      return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(Form.Item, {
        name: [name, 'min'],
        style: {
          width: '70px'
        }
      }, /*#__PURE__*/React.createElement(InputNumber, {
        placeholder: t('filter_form_list_control.numberFilter.qingShuRuShaiXuan'),
        style: {
          width: '100%'
        },
        min: (_DataList$min = DataList === null || DataList === void 0 ? void 0 : DataList.min) !== null && _DataList$min !== void 0 ? _DataList$min : 0,
        max: (_DataList$max = DataList === null || DataList === void 0 ? void 0 : DataList.max) !== null && _DataList$max !== void 0 ? _DataList$max : 0
      })), /*#__PURE__*/React.createElement("span", null, " - "), /*#__PURE__*/React.createElement(Form.Item, {
        name: [name, 'max'],
        style: {
          width: '70px'
        }
      }, /*#__PURE__*/React.createElement(InputNumber, {
        placeholder: t('filter_form_list_control.numberFilter.qingShuRuShaiXuan'),
        style: {
          width: '100%'
        },
        min: (_DataList$min2 = DataList === null || DataList === void 0 ? void 0 : DataList.min) !== null && _DataList$min2 !== void 0 ? _DataList$min2 : 0,
        max: (_DataList$max2 = DataList === null || DataList === void 0 ? void 0 : DataList.max) !== null && _DataList$max2 !== void 0 ? _DataList$max2 : 0
      })));
    }
    return /*#__PURE__*/React.createElement(Form.Item, {
      name: [name, 'value']
    }, /*#__PURE__*/React.createElement(InputNumber, {
      placeholder: t('filter_form_list_control.numberFilter.qingShuRuShaiXuan'),
      style: {
        width: '100%'
      },
      min: (_DataList$min3 = DataList === null || DataList === void 0 ? void 0 : DataList.min) !== null && _DataList$min3 !== void 0 ? _DataList$min3 : 0,
      max: (_DataList$max3 = DataList === null || DataList === void 0 ? void 0 : DataList.max) !== null && _DataList$max3 !== void 0 ? _DataList$max3 : 0
    }));
  }));
};
export default NumberFilter;
