// @antv/l7-editor/es/components/map-control-group/filter-form-list-control/stringFilter.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { Form, Input, Select, Tooltip } from 'antd';
import { cloneDeep, uniq } from 'lodash-es';
import React from 'react';
import { useTranslation } from 'react-i18next';
import useFeature from "../../../recoil/feature";
import useFilter from "../../../recoil/filter";
var StringFilter = function StringFilter(_ref) {
  var name = _ref.name,
    index = _ref.index,
    form = _ref.form;
  var _useFilter = useFilter(),
    setFilters = _useFilter.setFilters;
  var _useFeature = useFeature(),
    dataSource = _useFeature.dataSource;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var select = [{
    label: t('filter_form_list_control.stringFilter.baoHan'),
    value: 'IN'
  }, {
    label: t('filter_form_list_control.stringFilter.buBaoHan'),
    value: 'NOT_IN'
  }, {
    label: t('filter_form_list_control.stringFilter.piPei'),
    value: 'LIKE'
  }, {
    label: t('filter_form_list_control.stringFilter.buPiPei'),
    value: 'NOT_LIKE'
  }];
  return /*#__PURE__*/React.createElement("div", {
    style: {
      display: 'flex'
    }
  }, /*#__PURE__*/React.createElement(Form.Item, {
    name: [name, 'operator'],
    style: {
      width: '100px',
      marginRight: '8px'
    }
  }, /*#__PURE__*/React.createElement(Select, {
    placeholder: t('filter_form_list_control.numberFilter.qingXuanZeGuoLu'),
    options: select,
    onChange: function onChange() {
      var newFilterFromList = cloneDeep(form.getFieldValue('filterFromList'));
      newFilterFromList.forEach(function (v, i) {
        if (index === i) {
          v.value = undefined;
        }
      });
      form.setFieldValue('filterFromList', newFilterFromList);
      setFilters(newFilterFromList.map(function (item) {
        var _JSON$parse = JSON.parse(item.field),
          field = _JSON$parse.field,
          type = _JSON$parse.type;
        return _objectSpread(_objectSpread({}, item), {}, {
          field: field,
          type: type
        });
      }));
    }
  })), /*#__PURE__*/React.createElement(Form.Item, {
    shouldUpdate: function shouldUpdate(prevValues, curValues) {
      return prevValues.operator !== curValues.operator;
    },
    style: {
      width: 150,
      margin: 0
    }
  }, function (_ref2) {
    var _JSON$parse2, _uniq;
    var getFieldsValue = _ref2.getFieldsValue;
    var _getFieldsValue = getFieldsValue(),
      filterFromList = _getFieldsValue.filterFromList;
    var fieldValue = (_JSON$parse2 = JSON.parse(filterFromList[index].field)) === null || _JSON$parse2 === void 0 ? void 0 : _JSON$parse2.field;
    var DataList = dataSource.find(function (item) {
      return (item === null || item === void 0 ? void 0 : item.field) === fieldValue;
    });
    var fieldType = filterFromList[index].operator;
    if (fieldType === 'LIKE' || fieldType === 'NOT_LIKE') {
      return /*#__PURE__*/React.createElement(Form.Item, {
        name: [name, 'value']
      }, /*#__PURE__*/React.createElement(Input, {
        placeholder: t('filter_form_list_control.numberFilter.qingShuRuShaiXuan'),
        style: {
          width: '100%'
        },
        allowClear: true
      }));
    }
    return /*#__PURE__*/React.createElement(Form.Item, {
      name: [name, 'value']
    }, /*#__PURE__*/React.createElement(Select, {
      placeholder: t('filter_form_list_control.stringFilter.qingXuanZeShaiXuan'),
      style: {
        width: '100%'
      },
      mode: "multiple",
      maxTagCount: 1
    }, ((_uniq = uniq(DataList === null || DataList === void 0 ? void 0 : DataList.value)) !== null && _uniq !== void 0 ? _uniq : []).filter(function (item) {
      return item !== 'undefined';
    }).map(function (item) {
      return /*#__PURE__*/React.createElement(Select.Option, {
        value: item,
        key: item
      }, /*#__PURE__*/React.createElement(Tooltip, {
        title: item
      }, item));
    })));
  }));
};
export default StringFilter;
