// @antv/l7-editor/lib/recoil/atomState.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/recoil/atomState.ts
var atomState_exports = {};
__export(atomState_exports, {
  activeTabState: () => activeTabState,
  autoFitBoundsState: () => autoFitBoundsState,
  baseMapState: () => baseMapState,
  cityHistoryState: () => cityHistoryState,
  convertState: () => convertState,
  customTilesState: () => customTilesState,
  editorTextState: () => editorTextState,
  featureState: () => featureState,
  filterState: () => filterState,
  hideEditorState: () => hideEditorState,
  isDrawState: () => isDrawState,
  layerColorState: () => layerColorState,
  lnglatTextState: () => lnglatTextState,
  lnglatTypeState: () => lnglatTypeState,
  localeState: () => localeState,
  mapOptionState: () => mapOptionState,
  officialLayersState: () => officialLayersState,
  popupTriggerState: () => popupTriggerState,
  rightWidthState: () => rightWidthState,
  savedTextState: () => savedTextState,
  sceneState: () => sceneState,
  showDrawAreaState: () => showDrawAreaState,
  showDrawDistanceState: () => showDrawDistanceState,
  showTextLayerState: () => showTextLayerState,
  textLayerFieldsState: () => textLayerFieldsState,
  themeState: () => themeState,
  wasmPathState: () => wasmPathState
});
module.exports = __toCommonJS(atomState_exports);
var import_localforage = __toESM(require("localforage"));
var import_recoil = require("recoil");
var import_constants = require("../constants");
var localStorageEffect = (key) => ({ setSelf, onSet }) => {
  const getValue = localStorage.getItem(key);
  if (getValue) {
    setSelf(JSON.parse(getValue));
  }
  onSet((newValue) => {
    if (newValue instanceof import_recoil.DefaultValue) {
      localStorage.removeItem(key);
    } else {
      localStorage.setItem(key, JSON.stringify(newValue));
    }
  });
};
var indexDBEffect = (key) => ({ setSelf, onSet }) => {
  import_localforage.default.getItem(key).then(function(getValue) {
    if (getValue) {
      setSelf(getValue);
    }
  });
  onSet((newValue) => {
    if (newValue instanceof import_recoil.DefaultValue) {
      import_localforage.default.removeItem(key);
    } else {
      import_localforage.default.setItem(key, newValue);
    }
  });
};
var filterState = (0, import_recoil.atom)({
  key: "filter",
  default: [],
  dangerouslyAllowMutability: true
});
var isDrawState = (0, import_recoil.atom)({
  key: "isDraw",
  default: false
});
var savedTextState = (0, import_recoil.atom)({
  key: "savedText",
  default: ""
});
var featureState = (0, import_recoil.atom)({
  key: "features",
  default: [],
  dangerouslyAllowMutability: true
});
var sceneState = (0, import_recoil.atom)({
  key: "scene",
  default: null,
  dangerouslyAllowMutability: true
});
var editorTextState = (0, import_recoil.atom)({
  key: "editorText",
  default: JSON.stringify({ type: "FeatureCollection", features: [] }, null, 2),
  effects: [indexDBEffect(import_constants.LocalStorageKey.EditorText)]
});
var lnglatTypeState = (0, import_recoil.atom)({
  key: "lnglatType"
});
var lnglatTextState = (0, import_recoil.atom)({
  key: "lnglatText",
  default: ""
});
var rightWidthState = (0, import_recoil.atom)({
  key: "rightPanelWidth",
  effects: [localStorageEffect(import_constants.LocalStorageKey.RightPanelWidth)]
});
var mapOptionState = (0, import_recoil.atom)({
  key: "mapOptions",
  effects: [localStorageEffect(import_constants.LocalStorageKey.MapOptions)]
});
var layerColorState = (0, import_recoil.atom)({
  key: "layerColor",
  effects: [localStorageEffect(import_constants.LocalStorageKey.LayerColor)]
});
var hideEditorState = (0, import_recoil.atom)({
  key: "hideEditor",
  default: false,
  effects: [localStorageEffect(import_constants.LocalStorageKey.HideEditor)]
});
var autoFitBoundsState = (0, import_recoil.atom)({
  key: "autoFitBounds",
  effects: [localStorageEffect(import_constants.LocalStorageKey.AutoFitBounds)]
});
var showDrawDistanceState = (0, import_recoil.atom)({
  key: "showDrawDistance",
  default: false,
  effects: [localStorageEffect(import_constants.LocalStorageKey.showDrawDistance)]
});
var showDrawAreaState = (0, import_recoil.atom)({
  key: "showDrawArea",
  default: false,
  effects: [localStorageEffect(import_constants.LocalStorageKey.showDrawArea)]
});
var popupTriggerState = (0, import_recoil.atom)({
  key: "popupTrigger",
  effects: [localStorageEffect(import_constants.LocalStorageKey.PopupTrigger)]
});
var baseMapState = (0, import_recoil.atom)({
  key: "baseMap",
  effects: [localStorageEffect(import_constants.LocalStorageKey.BaseMap)]
});
var activeTabState = (0, import_recoil.atom)({
  key: "activeTab",
  default: "geojson",
  effects: [localStorageEffect(import_constants.LocalStorageKey.ActiveRightTabKey)]
});
var officialLayersState = (0, import_recoil.atom)({
  key: "layerType",
  default: [],
  effects: [localStorageEffect(import_constants.LocalStorageKey.officialLayers)]
});
var convertState = (0, import_recoil.atom)({
  key: "coordConvert",
  default: "GCJ02",
  effects: [localStorageEffect(import_constants.LocalStorageKey.Convert)]
});
var themeState = (0, import_recoil.atom)({
  key: "theme",
  default: "light",
  effects: [localStorageEffect(import_constants.LocalStorageKey.theme)]
});
var cityHistoryState = (0, import_recoil.atom)({
  key: "cityHistory",
  default: [],
  effects: [localStorageEffect(import_constants.LocalStorageKey.cityHistory)]
});
var showTextLayerState = (0, import_recoil.atom)({
  key: "showTextLayer",
  default: true,
  effects: [localStorageEffect(import_constants.LocalStorageKey.showTextLayer)]
});
var textLayerFieldsState = (0, import_recoil.atom)({
  key: "textLayerFields",
  default: void 0,
  effects: [localStorageEffect(import_constants.LocalStorageKey.textLayerFields)]
});
var customTilesState = (0, import_recoil.atom)({
  key: "customTiles",
  default: [],
  effects: [localStorageEffect(import_constants.LocalStorageKey.customTiles)]
});
var localeState = (0, import_recoil.atom)({
  key: "locale",
  default: "zh-CN",
  effects: [localStorageEffect(import_constants.LocalStorageKey.locale)]
});
var wasmPathState = (0, import_recoil.atom)({
  key: "wasmPaths",
  effects: [localStorageEffect(import_constants.LocalStorageKey.wasmPath)]
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  activeTabState,
  autoFitBoundsState,
  baseMapState,
  cityHistoryState,
  convertState,
  customTilesState,
  editorTextState,
  featureState,
  filterState,
  hideEditorState,
  isDrawState,
  layerColorState,
  lnglatTextState,
  lnglatTypeState,
  localeState,
  mapOptionState,
  officialLayersState,
  popupTriggerState,
  rightWidthState,
  savedTextState,
  sceneState,
  showDrawAreaState,
  showDrawDistanceState,
  showTextLayerState,
  textLayerFieldsState,
  themeState,
  wasmPathState
});
