// @antv/l7-editor/lib/components/app-header/base-map/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-header/base-map/index.tsx
var base_map_exports = {};
__export(base_map_exports, {
  BaseMap: () => BaseMap
});
module.exports = __toCommonJS(base_map_exports);
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../../recoil");
var { Option } = import_antd.Select;
var BaseMap = () => {
  const {
    baseMap,
    setBaseMap,
    setMapOptions,
    mapOptions,
    theme,
    setCoordConvert
  } = (0, import_recoil.useGlobal)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const options = [
    {
      img: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*POJeRJzjdNAAAAAAAAAAAAAADjWqAQ/original",
      label: t("btn.setting_btn.gaoDe"),
      value: "Gaode"
    },
    {
      img: "https://mdn.alipayobjects.com/huamei_k6sfo0/afts/img/A*-G-sT4L_sNoAAAAAAAAAAAAADjWqAQ/original",
      label: "Mapbox",
      value: "Mapbox"
    }
  ];
  const handleChange = (e) => {
    setBaseMap(e);
    setMapOptions({
      ...mapOptions,
      style: theme === "dark" ? "dark" : "normal"
    });
    setCoordConvert(e === "Gaode" ? "GCJ02" : "WGS84");
    if (e !== baseMap) {
      location.reload();
    }
  };
  return /* @__PURE__ */ import_react.default.createElement(
    import_antd.Select,
    {
      style: { width: "120px" },
      bordered: false,
      defaultValue: baseMap,
      onChange: handleChange
    },
    options.map((item) => {
      return /* @__PURE__ */ import_react.default.createElement(Option, { key: item.value, value: item.value, label: item.label }, /* @__PURE__ */ import_react.default.createElement(import_antd.Space, null, /* @__PURE__ */ import_react.default.createElement("span", { style: { display: "flex" } }, /* @__PURE__ */ import_react.default.createElement("img", { src: item.img, alt: "", style: { height: 15, width: 15 } })), /* @__PURE__ */ import_react.default.createElement("span", null, item.label)));
    })
  );
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BaseMap
});
