// @antv/l7-editor/lib/components/app-map/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/app-map/index.tsx
var app_map_exports = {};
__export(app_map_exports, {
  AppMap: () => AppMap
});
module.exports = __toCommonJS(app_map_exports);
var import_larkmap = require("@antv/larkmap");
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_constants = require("../../constants");
var import_recoil = require("../../recoil");
var import_utils = require("../../utils");
var import_prettier_text = require("../../utils/prettier-text");
var import_styles = __toESM(require("./styles"));
var AppMap = ({ children }) => {
  const { mapOptions: baseMapOptions, baseMap } = (0, import_recoil.useGlobal)();
  const { saveEditorText, setScene } = (0, import_recoil.useFeature)();
  const styles = (0, import_styles.default)();
  const { t } = (0, import_react_i18next.useTranslation)();
  (0, import_ahooks.useMount)(async () => {
    const url = (0, import_utils.getParamsNew)("url");
    if (url) {
      try {
        const geoData = await (0, import_utils.getUrlFeatureCollection)(url, "GeoJSON", t);
        saveEditorText((0, import_prettier_text.prettierText)({ content: geoData }));
      } catch (e) {
        import_antd.message.error(`${e}`);
        saveEditorText(
          JSON.stringify({ type: "FeatureCollection", features: [] }, null, 2)
        );
      }
    }
  });
  const mapOptions = (0, import_react.useMemo)(() => {
    if (baseMap === "Mapbox") {
      return {
        ...import_constants.MapBoxConfig,
        ...baseMapOptions
      };
    }
    return {
      ...import_constants.GaodeConfig,
      ...baseMapOptions
    };
  }, [baseMap, baseMapOptions]);
  return /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.LarkMap,
    {
      className: styles.l7EditorMap,
      mapOptions,
      mapType: baseMap,
      onSceneLoaded: setScene,
      id: "l7-editor-map",
      logoVisible: false
    },
    children
  );
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AppMap
});
