// @antv/l7-editor/es/components/map-control-group/l7-option-control/index.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { CustomControl, useScene } from '@antv/larkmap';
import { Button, Input, Modal, Tooltip, message } from 'antd';
import Clipboard from 'clipboard';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { IconFont } from "../../../components/iconfont";
import useStyles from "../styles";
var TextArea = Input.TextArea;
export var L7MapOptionControl = function L7MapOptionControl() {
  var scene = useScene();
  var styles = useStyles();
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isModalOpen = _useState2[0],
    setIsModalOpen = _useState2[1];
  var _useState3 = useState(''),
    _useState4 = _slicedToArray(_useState3, 2),
    textValue = _useState4[0],
    setTextValue = _useState4[1];
  var showModal = function showModal() {
    setIsModalOpen(true);
    var optionJson = {
      center: [scene.getCenter().lng, scene.getCenter().lat],
      zoom: scene.getZoom(),
      pitch: Number(scene.getPitch().toFixed(6)),
      rotation: scene.getRotation()
    };
    setTextValue(JSON.stringify(optionJson, null, 2));
  };
  var handleOk = function handleOk() {
    var clipboard = new Clipboard("#l7-option-copy_btn", {
      text: function text() {
        return textValue;
      }
    });
    clipboard.on('success', function () {
      setIsModalOpen(false);
      message.success(t('layer_contextmenu_popup.fuZhiChengGong'));
      clipboard.destroy();
    });
  };
  var handleCancel = function handleCancel() {
    setIsModalOpen(false);
  };
  return /*#__PURE__*/React.createElement(CustomControl, {
    position: "bottomleft"
  }, /*#__PURE__*/React.createElement(Tooltip, {
    placement: "left",
    overlay: t('l7Options.huoqucanshu')
  }, /*#__PURE__*/React.createElement("button", {
    className: styles.L7EditorControl,
    id: "text-layer-control",
    onClick: showModal
  }, /*#__PURE__*/React.createElement(IconFont, {
    type: "icon-xitongzhuangtai"
  }))), /*#__PURE__*/React.createElement(Modal, {
    title: t('l7Options.dangqiancanshu'),
    open: isModalOpen,
    onCancel: handleCancel,
    width: 800,
    destroyOnClose: true,
    footer: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
      onClick: handleCancel
    }, t('btn.setting_btn.guanBi')), /*#__PURE__*/React.createElement(Button, {
      type: "primary",
      id: "l7-option-copy_btn",
      onClick: handleOk
    }, t('layer_contextmenu_popup.fuZhi')))
  }, /*#__PURE__*/React.createElement(TextArea, {
    rows: 10,
    value: textValue,
    onChange: function onChange(e) {
      setTextValue(e.target.value);
    }
  })));
};
