// @antv/l7-editor/es/components/layer-popup/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { CloseOutlined, PlusSquareOutlined } from '@ant-design/icons';
import { DrawCircle, DrawEvent, DrawLine, DrawPoint, DrawPolygon, DrawRect } from '@antv/l7-draw';
import { Popup, useLayerList, useScene } from '@antv/larkmap';
import { featureCollection } from '@turf/turf';
import { Button, Descriptions, Empty, Form, Input, InputNumber, Tooltip, Typography } from 'antd';
import { cloneDeep } from 'lodash-es';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { FeatureKey, LayerId } from "../../constants";
import { useDrawHelper } from "../../hooks";
import { useFeature, useGlobal } from "../../recoil";
import { getDrawStyle, isCircle, isRect } from "../../utils";
import { prettierText } from "../../utils/prettier-text";
// import './index.css';
import useStyle from "./styles";
var Paragraph = Typography.Paragraph;
export var LayerPopup = function LayerPopup() {
  var _targetFeature$proper;
  var _Form$useForm = Form.useForm(),
    _Form$useForm2 = _slicedToArray(_Form$useForm, 1),
    form = _Form$useForm2[0];
  var scene = useScene();
  var _useFeature = useFeature(),
    saveEditorText = _useFeature.saveEditorText,
    isDraw = _useFeature.isDraw,
    setIsDraw = _useFeature.setIsDraw,
    resetFeatures = _useFeature.resetFeatures,
    features = _useFeature.features,
    setFeatures = _useFeature.setFeatures,
    revertCoord = _useFeature.revertCoord,
    transformCoord = _useFeature.transformCoord;
  var _useGlobal = useGlobal(),
    layerColor = _useGlobal.layerColor,
    popupTrigger = _useGlobal.popupTrigger,
    showDrawDistance = _useGlobal.showDrawDistance,
    showDrawArea = _useGlobal.showDrawArea;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var helperText = useDrawHelper();
  var styles = useStyle();
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    addOpen = _useState2[0],
    setAddOpen = _useState2[1];
  var _useState3 = useState({
      lngLat: {
        lng: 0,
        lat: 0
      },
      visible: false,
      feature: null
    }),
    _useState4 = _slicedToArray(_useState3, 2),
    popupProps = _useState4[0],
    setPopupProps = _useState4[1];
  var _useState5 = useState({
      isInput: false,
      index: null
    }),
    _useState6 = _slicedToArray(_useState5, 2),
    tableClick = _useState6[0],
    setTableClick = _useState6[1];
  var _useState7 = useState({
      label: undefined,
      value: undefined
    }),
    _useState8 = _slicedToArray(_useState7, 2),
    addValue = _useState8[0],
    setAddValue = _useState8[1];
  var targetFeature = useMemo(function () {
    return features.find(function (feature) {
      var _feature$properties;
      return (
        // @ts-ignore
        ((_feature$properties = feature.properties) === null || _feature$properties === void 0 ? void 0 : _feature$properties[FeatureKey.Index]) === popupProps.featureIndex
      );
    });
  }, [features, popupProps]);
  var featureFields = Object.entries((_targetFeature$proper = targetFeature === null || targetFeature === void 0 ? void 0 : targetFeature.properties) !== null && _targetFeature$proper !== void 0 ? _targetFeature$proper : {});
  var allLayerList = useLayerList();
  var layerList = useMemo(function () {
    var layerIds = [LayerId.PointLayer, LayerId.LineLayer, LayerId.PolygonLayer];
    return allLayerList.filter(function (layer) {
      return layerIds.includes(layer.id);
    });
  }, [allLayerList]);
  var disabledEdit = function disabledEdit(feature) {
    if (feature) {
      var reg = RegExp(/Multi/);
      return reg.test(feature.geometry.type);
    }
    return false;
  };
  var onLayerClick = useCallback(function (e) {
    if (!isDraw) {
      var lngLat = e.lngLat,
        feature = e.feature;
      var featureIndex = feature.properties[FeatureKey.Index];
      var isIndex = popupProps.featureIndex === feature.properties[FeatureKey.Index];
      if (popupProps.visible && isIndex) {
        setPopupProps(function (oldPopupProps) {
          return _objectSpread(_objectSpread({}, oldPopupProps), {}, {
            visible: false,
            featureIndex: undefined,
            feature: null
          });
        });
        return;
      }
      setPopupProps({
        lngLat: lngLat,
        visible: true,
        featureIndex: featureIndex,
        feature: e.feature
      });
    }
  }, [setPopupProps, popupProps, isDraw]);
  var onLayerMouseenter = useCallback(function (e) {
    if (!isDraw) {
      var lngLat = e.lngLat,
        feature = e.feature;
      var featureIndex = feature.properties[FeatureKey.Index];
      setPopupProps({
        lngLat: lngLat,
        visible: true,
        featureIndex: featureIndex
      });
    }
  }, [setPopupProps, isDraw]);
  var onLayerMouseout = useCallback(function () {
    if (!isDraw) {
      setPopupProps(function (oldPopupProps) {
        return _objectSpread(_objectSpread({}, oldPopupProps), {}, {
          visible: false,
          featureIndex: undefined
        });
      });
    }
  }, [setPopupProps, isDraw]);
  var onEdit = function onEdit(feature) {
    setIsDraw(true);
    var newFeature = cloneDeep(feature);
    var newFeatures = features.filter(function (item) {
      var _feature$properties2;
      return (
        //@ts-ignore
        item.properties[FeatureKey.Index] !== ( //@ts-ignore
        (_feature$properties2 = feature.properties) === null || _feature$properties2 === void 0 ? void 0 : _feature$properties2[FeatureKey.Index])
      );
    });
    var index = features.findIndex(function (item) {
      var _feature$properties3;
      return (
        //@ts-ignore
        item.properties[FeatureKey.Index] === ( //@ts-ignore
        (_feature$properties3 = feature.properties) === null || _feature$properties3 === void 0 ? void 0 : _feature$properties3[FeatureKey.Index])
      );
    });
    var onEditFinish = function onEditFinish(selectFeature, draw) {
      if (!selectFeature) {
        var data = draw.getData();
        if (data.length) {
          newFeature = _objectSpread(_objectSpread({}, data[0]), {}, {
            properties: feature === null || feature === void 0 ? void 0 : feature.properties
          });
          var newTransformFeatures = revertCoord([newFeature])[0];
          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
          features[index] = newTransformFeatures;
        } else {
          features.splice(index, 1);
        }
        draw.destroy();
        setIsDraw(false);
        resetFeatures(features);
      }
    };
    var options = {
      initialData: [newFeature],
      maxCount: 1,
      style: getDrawStyle(layerColor),
      helper: helperText,
      distanceOptions: showDrawDistance ? {} : undefined,
      areaOptions: showDrawArea ? {} : undefined
    };
    var draw;
    var _transformCoord = transformCoord([newFeature]),
      _transformCoord2 = _slicedToArray(_transformCoord, 1),
      originFeature = _transformCoord2[0];
    var type = originFeature === null || originFeature === void 0 ? void 0 : originFeature.geometry.type;
    if (type === 'Point') {
      draw = new DrawPoint(scene, options);
    } else if (type === 'LineString') {
      draw = new DrawLine(scene, options);
    } else if (type === 'Polygon' && isRect(originFeature)) {
      draw = new DrawRect(scene, options);
    } else if (type === 'Polygon' && isCircle(originFeature)) {
      draw = new DrawCircle(scene, options);
    } else {
      draw = new DrawPolygon(scene, options);
    }
    draw.enable();
    draw.setActiveFeature(draw.getData()[0]);
    setFeatures(newFeatures);
    draw.on(DrawEvent.Select, function (selectFeature) {
      return onEditFinish(selectFeature, draw);
    });
    setPopupProps({
      visible: false,
      featureIndex: undefined
    });
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  var onLayerDblClick = function onLayerDblClick(e) {
    var feature = e.feature;
    if (!disabledEdit(feature) && !isDraw) {
      onEdit(feature);
    }
  };
  useEffect(function () {
    layerList.forEach(function (layer) {
      return layer.on('dblclick', onLayerDblClick);
    });
    return function () {
      layerList.forEach(function (layer) {
        return layer.off('dblclick', onLayerDblClick);
      });
    };
  }, [onLayerDblClick, layerList, popupTrigger, scene]);
  useEffect(function () {
    var layerEvent = {
      click: [{
        event: 'click',
        click: onLayerClick
      }],
      hover: [{
        event: 'mouseenter',
        click: onLayerMouseenter
      }, {
        event: 'mouseout',
        click: onLayerMouseout
      }]
    };
    //@ts-ignore
    layerEvent[popupTrigger].forEach(function (e) {
      layerList.forEach(function (layer) {
        return layer.on(e.event, e.click);
      });
    });
    return function () {
      //@ts-ignore
      layerEvent[popupTrigger].forEach(function (e) {
        layerList.forEach(function (layer) {
          return layer.off(e.event, e.click);
        });
      });
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [onLayerClick, onLayerMouseenter, layerList, popupTrigger, scene]);
  var save = function save(key, value) {
    var formValue = form.getFieldValue('input');
    if (value !== formValue) {
      var properties = _objectSpread(_objectSpread({}, popupProps.feature.properties), {}, _defineProperty({}, key, formValue));
      var feature = _objectSpread(_objectSpread({}, popupProps.feature), {}, {
        properties: properties
      });
      setPopupProps(function (event) {
        return _objectSpread(_objectSpread({}, event), {}, {
          feature: feature
        });
      });
      var index = features.findIndex(function (item) {
        var _feature$properties4;
        return (
          //@ts-ignore
          item.properties[FeatureKey.Index] === ( //@ts-ignore
          (_feature$properties4 = feature.properties) === null || _feature$properties4 === void 0 ? void 0 : _feature$properties4[FeatureKey.Index])
        );
      });
      features[index] = feature;
      saveEditorText(prettierText({
        content: featureCollection(features)
      }));
    }
    setTableClick({
      isInput: false,
      index: null
    });
  };
  var addBlur = function addBlur() {
    if (addValue.label && addValue.value) {
      var properties = _objectSpread(_objectSpread({}, popupProps.feature.properties), {}, _defineProperty({}, addValue.label, addValue.value));
      var feature = _objectSpread(_objectSpread({}, popupProps.feature), {}, {
        properties: properties
      });
      setPopupProps(function (event) {
        return _objectSpread(_objectSpread({}, event), {}, {
          feature: feature
        });
      });
      var index = features.findIndex(function (item) {
        var _feature$properties5;
        return (
          //@ts-ignore
          item.properties[FeatureKey.Index] === ( //@ts-ignore
          (_feature$properties5 = feature.properties) === null || _feature$properties5 === void 0 ? void 0 : _feature$properties5[FeatureKey.Index])
        );
      });
      features[index] = feature;
      saveEditorText(prettierText({
        content: featureCollection(features)
      }));
      setAddValue({
        label: undefined,
        value: undefined
      });
      setAddOpen(!addOpen);
    }
  };
  var popupTable = /*#__PURE__*/React.createElement("div", null, featureFields.length || addOpen ? /*#__PURE__*/React.createElement("div", {
    className: styles.layerPopupInfo,
    onWheel: function onWheel(e) {
      e.stopPropagation();
    }
  }, /*#__PURE__*/React.createElement(Descriptions, {
    size: "small",
    bordered: true,
    column: 1
  }, featureFields.map(function (_ref, index) {
    var _ref2 = _slicedToArray(_ref, 2),
      key = _ref2[0],
      value = _ref2[1];
    if (!(value instanceof Object)) {
      return /*#__PURE__*/React.createElement(Descriptions.Item, {
        label: key,
        key: key
      }, /*#__PURE__*/React.createElement(Paragraph, {
        copyable: {
          text: "".concat(value)
        },
        style: {
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between'
        }
      }, tableClick.isInput && tableClick.index === index ? /*#__PURE__*/React.createElement(Form, {
        form: form,
        className: styles.form
      }, /*#__PURE__*/React.createElement(Form.Item, {
        name: "input"
      }, typeof value === 'number' ? /*#__PURE__*/React.createElement(InputNumber, {
        autoFocus: true,
        onPressEnter: function onPressEnter() {
          return save(key, value);
        },
        onBlur: function onBlur() {
          return save(key, value);
        }
      }) : /*#__PURE__*/React.createElement(Input, {
        autoFocus: true,
        onPressEnter: function onPressEnter() {
          return save(key, value);
        },
        onBlur: function onBlur() {
          return save(key, value);
        }
      }))) : /*#__PURE__*/React.createElement("div", {
        style: {
          width: '100%'
        },
        onClick: function onClick() {
          setTableClick({
            isInput: !tableClick.isInput,
            index: index
          });
          setAddOpen(false);
          form.setFieldsValue({
            input: value
          });
        }
      }, value ? String(value) : '-')));
    }
    return null;
  }), addOpen && /*#__PURE__*/React.createElement(Descriptions.Item, {
    contentStyle: {
      paddingRight: 8
    },
    label: /*#__PURE__*/React.createElement(Input, {
      size: "small",
      onChange: function onChange(e) {
        setAddValue(function (prevState) {
          return _objectSpread(_objectSpread({}, prevState), {}, {
            label: e.target.value
          });
        });
      },
      onBlur: addBlur
    })
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.addField
  }, /*#__PURE__*/React.createElement(Input, {
    size: "small",
    onChange: function onChange(e) {
      setAddValue(function (prevState) {
        return _objectSpread(_objectSpread({}, prevState), {}, {
          value: e.target.value
        });
      });
    },
    onBlur: addBlur
  }), /*#__PURE__*/React.createElement(Button, {
    type: "text",
    className: styles.addBut,
    icon: /*#__PURE__*/React.createElement(CloseOutlined, null),
    onClick: function onClick() {
      setAddOpen(false);
    }
  }))))) : /*#__PURE__*/React.createElement(Empty, {
    description: t('layer_popup.index.dangQianYuanSuWu'),
    style: {
      margin: '12px 0'
    }
  }), /*#__PURE__*/React.createElement("div", {
    style: {
      marginTop: 10
    }
  }, !addOpen && popupTrigger === 'click' && /*#__PURE__*/React.createElement(Button, {
    type: "dashed",
    block: true,
    style: {
      marginBottom: 6
    },
    onClick: function onClick() {
      setAddOpen(!addOpen);
    }
  }, /*#__PURE__*/React.createElement(PlusSquareOutlined, null), t('layer_popup.index.tianJiaZiDuan'))));
  return /*#__PURE__*/React.createElement(React.Fragment, null, popupProps.visible && typeof popupProps.featureIndex === 'number' && targetFeature && /*#__PURE__*/React.createElement(Popup, {
    className: styles.layerPopupContent,
    lngLat: popupProps.lngLat,
    closeButton: false,
    offsets: [0, 10],
    followCursor: popupTrigger === 'hover',
    closeOnClick: true,
    onClose: function onClose() {
      setAddOpen(false);
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.layerPopup,
    onClick: function onClick(e) {
      e.stopPropagation();
    }
  }, popupTable, /*#__PURE__*/React.createElement("div", {
    className: styles.layerPopupBtnGroup
  }, popupTrigger === 'click' && /*#__PURE__*/React.createElement(Tooltip, {
    title: disabledEdit(popupProps.feature) ? t('layer_popup.index.mULTI') : ''
  }, /*#__PURE__*/React.createElement(Button, {
    size: "small",
    type: "link",
    onClick: function onClick() {
      return onEdit(popupProps.feature);
    },
    disabled: disabledEdit(popupProps.feature)
  }, t('layer_popup.index.gengGaiHuiZhi'))), popupTrigger === 'click' && /*#__PURE__*/React.createElement(Button, {
    size: "small",
    type: "link",
    danger: true,
    onClick: function onClick() {
      resetFeatures(features.filter(function (_, index) {
        return index !== popupProps.featureIndex;
      }));
      setPopupProps({
        visible: false,
        featureIndex: undefined
      });
    }
  }, t('app_table.index.shanChu'))))));
};
