// @antv/l7-editor/lib/components/layer-contextmenu-popup/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/layer-contextmenu-popup/index.tsx
var layer_contextmenu_popup_exports = {};
__export(layer_contextmenu_popup_exports, {
  LayerContextmenuPopup: () => LayerContextmenuPopup
});
module.exports = __toCommonJS(layer_contextmenu_popup_exports);
var import_larkmap = require("@antv/larkmap");
var import_turf = require("@turf/turf");
var import_react = __toESM(require("react"));
var import_constants = require("../../constants");
var import_recoil = require("../../recoil");
var import_wkt = require("../../utils/wkt");
var import_code_block = __toESM(require("./code-block"));
var import_styles = __toESM(require("./styles"));
var LayerContextmenuPopup = () => {
  const scene = (0, import_larkmap.useScene)();
  const { isDraw, features } = (0, import_recoil.useFeature)();
  const styles = (0, import_styles.default)();
  const [markerProps, setMarkerProps] = (0, import_react.useState)({
    lngLat: {
      lng: 0,
      lat: 0
    },
    visible: false,
    feature: null
  });
  const targetFeature = (0, import_react.useMemo)(() => {
    return features.find(
      (feature) => {
        var _a;
        return (
          // @ts-ignore
          ((_a = feature.properties) == null ? void 0 : _a[import_constants.FeatureKey.Index]) === markerProps.featureIndex
        );
      }
    );
  }, [features, markerProps]);
  const allLayerList = (0, import_larkmap.useLayerList)();
  const layerList = (0, import_react.useMemo)(() => {
    const layerIds = [
      import_constants.LayerId.PointLayer,
      import_constants.LayerId.LineLayer,
      import_constants.LayerId.PolygonLayer
    ];
    return allLayerList.filter((layer) => {
      return layerIds.includes(layer.id);
    });
  }, [allLayerList]);
  const onLayerClick = (0, import_react.useCallback)(
    (e) => {
      if (!isDraw) {
        const { lngLat, feature } = e;
        const featureIndex = feature.properties[import_constants.FeatureKey.Index];
        const isIndex = markerProps.featureIndex === feature.properties[import_constants.FeatureKey.Index];
        if (markerProps.visible && isIndex) {
          setMarkerProps((oldPopupProps) => {
            return {
              ...oldPopupProps,
              visible: false,
              featureIndex: void 0,
              feature: null
            };
          });
          return;
        }
        setMarkerProps({
          lngLat,
          visible: true,
          featureIndex,
          feature: e.feature
        });
      }
    },
    [setMarkerProps, markerProps, isDraw]
  );
  const mapRightMenuClose = () => {
    const timeOut = setTimeout(() => {
      if (timeOut) {
        clearTimeout(timeOut);
        setMarkerProps({
          lngLat: {
            lng: 0,
            lat: 0
          },
          visible: false,
          feature: null
        });
      }
    }, 0);
  };
  (0, import_react.useEffect)(() => {
    layerList.forEach((layer) => layer.on("contextmenu", onLayerClick));
    scene.on("click", mapRightMenuClose);
    return () => {
      layerList.forEach((layer) => layer.off("contextmenu", onLayerClick));
      scene.off("click", mapRightMenuClose);
    };
  }, [onLayerClick, layerList, scene]);
  const copyTypeList = (0, import_react.useMemo)(() => {
    if (!markerProps.feature) {
      return [];
    }
    const index = markerProps.feature.properties[import_constants.FeatureKey.Index];
    const featureCollectionData = (0, import_turf.featureCollection)([features[index]]);
    return [
      {
        copyType: "GeoJSON",
        text: JSON.stringify(featureCollectionData)
      },
      {
        copyType: "WKT",
        text: (0, import_wkt.GeoJSON2Wkt)(featureCollectionData)
      }
    ];
  }, [features, markerProps.feature]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, markerProps.visible && typeof markerProps.featureIndex === "number" && targetFeature && markerProps.lngLat && /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.Marker,
    {
      lngLat: markerProps.lngLat,
      anchor: "top-left",
      offsets: [0, 10]
    },
    /* @__PURE__ */ import_react.default.createElement("div", { className: styles.layerPopupContent }, copyTypeList.map((item) => {
      return /* @__PURE__ */ import_react.default.createElement("div", { className: styles.layerPopupCopyRow, key: item.copyType }, /* @__PURE__ */ import_react.default.createElement(import_code_block.default, { copyType: item.copyType, text: item.text }));
    }))
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LayerContextmenuPopup
});
