// @antv/l7-editor/lib/components/layer-contextmenu-popup/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/layer-contextmenu-popup/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var { useToken } = import_antd.theme;
var useStyle = () => {
  const { token } = useToken();
  const { colorBgContainer, colorText, colorFillSecondary } = token;
  return {
    layerPopupContent: import_css.css`
      background: ${colorBgContainer} !important;
      .l7-popup-tip {
        border-top-color: ${colorBgContainer} !important;
      }
      .l7-popup-content {
        background: ${colorBgContainer} !important;
      }
      padding: 4px;
      border-radius: 8px;
    `,
    layerPopupCopyRow: import_css.css`
      color: ${colorText};
      font-size: 13px;
      border-radius: 4px;
      padding: 8px;
      &:last-child {
        margin-bottom: 0;
      }
      &:hover {
        background: ${colorFillSecondary};
      }
    `,
    layerPopupCopyButton: import_css.css`
      padding: 0;
      height: auto;
    `
  };
};
var styles_default = useStyle;
