// @antv/l7-editor/lib/components/map-control-group/location-search-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/location-search-control/index.tsx
var location_search_control_exports = {};
__export(location_search_control_exports, {
  default: () => location_search_control_default
});
module.exports = __toCommonJS(location_search_control_exports);
var import_larkmap = require("@antv/larkmap");
var import_turf = require("@turf/turf");
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_color = __toESM(require("color"));
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../../recoil");
var import_change_image_color = require("../../../utils/change-image-color");
var import_location_search = require("./location-search");
var LocationSearchControl = import_react.default.memo(() => {
  const scene = (0, import_larkmap.useScene)();
  const [selectLocation, setSelectLocation] = (0, import_react.useState)();
  const [locationText, setLocationText] = (0, import_react.useState)("");
  const { features, resetFeatures } = (0, import_recoil.useFeature)();
  const { layerColor } = (0, import_recoil.useGlobal)();
  const [colorImg, setColorImg] = (0, import_react.useState)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const syncMapCenter = (0, import_react.useCallback)(() => {
    if (scene) {
      const { lng, lat } = scene.getCenter();
      setLocationText(`${lng},${lat}`);
    }
  }, [scene]);
  (0, import_react.useEffect)(() => {
    syncMapCenter();
    scene == null ? void 0 : scene.on("moveend", syncMapCenter);
    scene == null ? void 0 : scene.on("zoomend", syncMapCenter);
    return () => {
      scene == null ? void 0 : scene.off("moveend", syncMapCenter);
      scene == null ? void 0 : scene.off("zoomend", syncMapCenter);
    };
  }, [scene, syncMapCenter]);
  (0, import_ahooks.useAsyncEffect)(async () => {
    const newLayerColor = (0, import_color.default)(layerColor).rgb().object();
    setColorImg(await (0, import_change_image_color.getPointImage)(newLayerColor, { x: 100, y: 100 }));
  }, [layerColor]);
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "lefttop" }, /* @__PURE__ */ import_react.default.createElement("div", { id: "l7-editor-citySelect", style: { display: "flex" } }, /* @__PURE__ */ import_react.default.createElement("div", null, /* @__PURE__ */ import_react.default.createElement("div", null, /* @__PURE__ */ import_react.default.createElement("div", null, /* @__PURE__ */ import_react.default.createElement(
    import_location_search.LocationSearch,
    {
      getPopupContainer: () => document.querySelector(".larkmap"),
      allowClear: true,
      value: selectLocation == null ? void 0 : selectLocation.name,
      searchParams: {
        key: "98d10f05a2da96697313a2ce35ebf1a2",
        location: locationText
      },
      onChange: (_, item) => {
        if (item) {
          const currentZoom = scene.getZoom();
          scene.setZoomAndCenter(
            currentZoom > 16 ? currentZoom : 16,
            [item.longitude, item.latitude]
          );
        }
        setSelectLocation(item);
      }
    }
  )))))), selectLocation && /* @__PURE__ */ import_react.default.createElement(
    import_larkmap.Marker,
    {
      lngLat: {
        lng: selectLocation.longitude,
        lat: selectLocation.latitude
      }
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_antd.Popover,
      {
        content: /* @__PURE__ */ import_react.default.createElement("div", null, /* @__PURE__ */ import_react.default.createElement(
          import_antd.Button,
          {
            type: "primary",
            onClick: () => {
              const { longitude, latitude } = selectLocation;
              resetFeatures([
                ...features,
                (0, import_turf.point)([longitude, latitude], selectLocation)
              ]);
              setSelectLocation(void 0);
            }
          },
          t("location_search_control.index.tianJiaZhiShuJu")
        ), /* @__PURE__ */ import_react.default.createElement(
          import_antd.Button,
          {
            danger: true,
            style: { marginLeft: 8 },
            onClick: () => {
              setSelectLocation(void 0);
            }
          },
          t("app_header.constants.qingChu")
        ))
      },
      /* @__PURE__ */ import_react.default.createElement("img", { style: { width: 40, height: 40 }, src: colorImg == null ? void 0 : colorImg.src })
    )
  ));
});
var location_search_control_default = LocationSearchControl;
