// @antv/l7-editor/lib/components/map-control-group/location-search-control/location-search.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/location-search-control/location-search.tsx
var location_search_exports = {};
__export(location_search_exports, {
  LocationSearch: () => LocationSearch
});
module.exports = __toCommonJS(location_search_exports);
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_styles = __toESM(require("./styles"));
var import_utils = require("./utils");
var { Option } = import_antd.Select;
var LocationSearch = ({
  searchParams,
  showDistrict,
  showAddress,
  onSearchFinish,
  onChange,
  ...selectProps
}) => {
  const styles = (0, import_styles.default)();
  const [options, setOptions] = (0, import_react.useState)([]);
  const { t } = (0, import_react_i18next.useTranslation)();
  (0, import_react.useEffect)(() => {
    onSearchFinish == null ? void 0 : onSearchFinish(options);
  }, [onSearchFinish, options]);
  const { run: onSearch } = (0, import_ahooks.useDebounceFn)(
    async (searchText) => {
      if (!searchText) {
        setOptions([]);
        return;
      }
      const url = (0, import_utils.urlStringify)(
        "https://restapi.amap.com/v3/assistant/inputtips",
        {
          ...searchParams,
          keywords: [...(searchParams.keywords ?? "").split("|"), searchText].filter((item) => !!item).join("|")
        }
      );
      const res = await (await fetch(url)).json();
      setOptions(
        ((res == null ? void 0 : res.tips) ?? []).filter((item) => item.location && item.location.length).map((item) => {
          const [lon, lat] = item.location.split(",");
          item.longitude = +lon;
          item.latitude = +lat;
          return item;
        })
      );
    },
    {
      wait: 1e3
    }
  );
  const onLocationChange = (0, import_react.useCallback)(
    (name) => {
      const targetOption = name && options.find((option) => option.name === name);
      onChange == null ? void 0 : onChange(name || void 0, targetOption || void 0);
    },
    [onChange, options]
  );
  return /* @__PURE__ */ import_react.default.createElement(
    import_antd.Select,
    {
      className: styles.locationSearch,
      onSearch,
      onChange: onLocationChange,
      placeholder: t("location_search_control.location_search.qingShuRuYaoSou"),
      optionLabelProp: "label",
      ...selectProps
    },
    options.map((option) => {
      const tip = `${showDistrict ? option.district : ""}${showAddress ? option.address : ""}`;
      return /* @__PURE__ */ import_react.default.createElement(Option, { key: option.id, value: option.name, label: option.name }, /* @__PURE__ */ import_react.default.createElement("div", { title: option.name, className: styles.locationSearchName }, option.name), tip && /* @__PURE__ */ import_react.default.createElement("div", { title: tip, className: styles.locationSearchTip }, tip));
    })
  );
};
LocationSearch.defaultProps = {
  showSearch: true,
  allowClear: true,
  filterOption: false,
  defaultActiveFirstOption: false,
  showAddress: true,
  showDistrict: true
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LocationSearch
});
