// @antv/l7-editor/lib/pages/components/editor.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/pages/components/editor.tsx
var editor_exports = {};
__export(editor_exports, {
  Editor: () => Editor
});
module.exports = __toCommonJS(editor_exports);
var import_ahooks = require("ahooks");
var import_antd = require("antd");
var import_classnames = __toESM(require("classnames"));
var import_localforage = __toESM(require("localforage"));
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_components = require("../../components");
var import_constants = require("../../constants");
var import_locales = require("../../locales");
var import_recoil = require("../../recoil");
var import_styles = __toESM(require("./styles"));
var Editor = (props) => {
  const { onFeatureChange } = props;
  const { i18n, t } = (0, import_react_i18next.useTranslation)();
  const { theme, mapOptions, setMapOptions, locale } = (0, import_recoil.useGlobal)();
  const styles = (0, import_styles.default)();
  const { saveEditorText, bboxAutoFit, scene } = (0, import_recoil.useFeature)();
  (0, import_ahooks.useUpdateEffect)(() => {
    if (theme === "dark") {
      setMapOptions({ ...mapOptions, style: "dark" });
    } else {
      setMapOptions({ ...mapOptions, style: "normal" });
    }
  }, [theme]);
  (0, import_react.useEffect)(() => {
    i18n.changeLanguage(locale);
  }, []);
  const antdLocale = (0, import_react.useMemo)(
    // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
    () => {
      var _a;
      return (_a = import_locales.LangList.find((lang) => lang.lang === locale)) == null ? void 0 : _a.antd;
    },
    [locale]
  );
  (0, import_ahooks.useAsyncEffect)(async () => {
    const newEditorText = await import_localforage.default.getItem(
      import_constants.LocalStorageKey.EditorText
    );
    if (newEditorText && scene && !props.features) {
      try {
        const newFeatures = JSON.parse(newEditorText).features;
        bboxAutoFit(newFeatures);
      } catch (error) {
        import_antd.message.error(t("import_btn.file_upload.qingJianChaShuJu"));
      }
    } else if (scene && props.features) {
      bboxAutoFit();
    }
  }, [scene]);
  (0, import_ahooks.useAsyncEffect)(async () => {
    const newEditorText = await import_localforage.default.getItem(
      import_constants.LocalStorageKey.EditorText
    );
    if (newEditorText && !props.features) {
      saveEditorText(newEditorText);
    }
  }, []);
  return /* @__PURE__ */ import_react.default.createElement(
    import_antd.ConfigProvider,
    {
      locale: antdLocale,
      theme: {
        algorithm: theme === "dark" ? import_antd.theme.darkAlgorithm : import_antd.theme.defaultAlgorithm
      }
    },
    /* @__PURE__ */ import_react.default.createElement(
      "div",
      {
        className: (0, import_classnames.default)([styles.l7Editor, "l7-editor"]),
        id: "l7-editor"
      },
      /* @__PURE__ */ import_react.default.createElement(import_components.AppHeader, { toolbar: props.toolbar }),
      /* @__PURE__ */ import_react.default.createElement(
        import_components.ResizePanel,
        {
          onFeatureChange: (e) => {
            if (onFeatureChange) {
              onFeatureChange(e);
            }
          },
          left: /* @__PURE__ */ import_react.default.createElement(import_components.AppMap, null, /* @__PURE__ */ import_react.default.createElement(import_components.MapControlGroup, { mapControl: props.mapControl }), /* @__PURE__ */ import_react.default.createElement(import_components.LayerList, null), /* @__PURE__ */ import_react.default.createElement(import_components.LayerPopup, null), /* @__PURE__ */ import_react.default.createElement(import_components.LayerContextmenuPopup, null)),
          right: /* @__PURE__ */ import_react.default.createElement(import_components.MapContent, { tabItems: props == null ? void 0 : props.tabItems, features: props == null ? void 0 : props.features })
        }
      )
    )
  );
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Editor
});
