// @antv/l7-editor/es/components/map-content/index.js
import { GlobalOutlined, TableOutlined } from '@ant-design/icons';
import { Select, Tabs } from 'antd';
import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useFeature, useGlobal } from "../../recoil";
import { AppTable } from "../app-table";
import { GeoJsonEditor } from "../geojson-editor";
import { IconFont } from "../iconfont";
import { WktEditor } from "../wkt-editor";
import useStyle from "./styles";
export var MapContent = function MapContent(_ref) {
  var features = _ref.features,
    tabItems = _ref.tabItems;
  var _useGlobal = useGlobal(),
    activeTab = _useGlobal.activeTab,
    setActiveTab = _useGlobal.setActiveTab,
    coordConvert = _useGlobal.coordConvert,
    setCoordConvert = _useGlobal.setCoordConvert;
  var _useFeature = useFeature(),
    saveEditorText = _useFeature.saveEditorText;
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var styles = useStyle();
  useEffect(function () {
    if (features) {
      saveEditorText(JSON.stringify({
        type: 'FeatureCollection',
        features: features
      }, null, 2));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [features]);
  var defaultTabItems = [{
    key: 'geojson',
    icon: /*#__PURE__*/React.createElement(IconFont, {
      type: "icon-json"
    }),
    label: 'GeoJSON',
    children: /*#__PURE__*/React.createElement(GeoJsonEditor, null)
  }, {
    key: 'wkt',
    icon: /*#__PURE__*/React.createElement(GlobalOutlined, null),
    label: 'WKT',
    children: /*#__PURE__*/React.createElement(WktEditor, null)
  }, {
    key: 'table',
    icon: /*#__PURE__*/React.createElement(TableOutlined, null),
    label: t('app_header.constants.biaoGe'),
    children: /*#__PURE__*/React.createElement(AppTable, null)
  }];
  return /*#__PURE__*/React.createElement("div", {
    className: styles.mapContent,
    id: "l7-editor-panel"
  }, /*#__PURE__*/React.createElement("div", {
    className: styles.mapContentSelect
  }, /*#__PURE__*/React.createElement("span", {
    className: styles.mapContentSelectLabel
  }, t('map_content.index.zuoBiaoXi')), /*#__PURE__*/React.createElement(Select, {
    value: coordConvert,
    options: [{
      label: 'GCJ02',
      value: 'GCJ02'
    }, {
      label: 'WGS84',
      value: 'WGS84'
    }],
    onChange: setCoordConvert
  })), /*#__PURE__*/React.createElement(Tabs
  //@ts-ignore
  , {
    activeKey: activeTab === 'code' ? 'geojson' : activeTab,
    className: styles.mapContentRight,
    defaultActiveKey: "geojson",
    items: tabItems ? tabItems : defaultTabItems,
    onChange: function onChange(e) {
      setActiveTab(e);
    }
  }));
};
