// @antv/l7-editor/lib/components/map-content/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-content/index.tsx
var map_content_exports = {};
__export(map_content_exports, {
  MapContent: () => MapContent
});
module.exports = __toCommonJS(map_content_exports);
var import_icons = require("@ant-design/icons");
var import_antd = require("antd");
var import_react = __toESM(require("react"));
var import_react_i18next = require("react-i18next");
var import_recoil = require("../../recoil");
var import_app_table = require("../app-table");
var import_geojson_editor = require("../geojson-editor");
var import_iconfont = require("../iconfont");
var import_wkt_editor = require("../wkt-editor");
var import_styles = __toESM(require("./styles"));
var MapContent = ({
  features,
  tabItems
}) => {
  const { activeTab, setActiveTab, coordConvert, setCoordConvert } = (0, import_recoil.useGlobal)();
  const { saveEditorText } = (0, import_recoil.useFeature)();
  const { t } = (0, import_react_i18next.useTranslation)();
  const styles = (0, import_styles.default)();
  (0, import_react.useEffect)(() => {
    if (features) {
      saveEditorText(
        JSON.stringify({ type: "FeatureCollection", features }, null, 2)
      );
    }
  }, [features]);
  const defaultTabItems = [
    {
      key: "geojson",
      icon: /* @__PURE__ */ import_react.default.createElement(import_iconfont.IconFont, { type: "icon-json" }),
      label: "GeoJSON",
      children: /* @__PURE__ */ import_react.default.createElement(import_geojson_editor.GeoJsonEditor, null)
    },
    {
      key: "wkt",
      icon: /* @__PURE__ */ import_react.default.createElement(import_icons.GlobalOutlined, null),
      label: "WKT",
      children: /* @__PURE__ */ import_react.default.createElement(import_wkt_editor.WktEditor, null)
    },
    {
      key: "table",
      icon: /* @__PURE__ */ import_react.default.createElement(import_icons.TableOutlined, null),
      label: t("app_header.constants.biaoGe"),
      children: /* @__PURE__ */ import_react.default.createElement(import_app_table.AppTable, null)
    }
  ];
  return /* @__PURE__ */ import_react.default.createElement("div", { className: styles.mapContent, id: "l7-editor-panel" }, /* @__PURE__ */ import_react.default.createElement("div", { className: styles.mapContentSelect }, /* @__PURE__ */ import_react.default.createElement("span", { className: styles.mapContentSelectLabel }, t("map_content.index.zuoBiaoXi")), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Select,
    {
      value: coordConvert,
      options: [
        { label: "GCJ02", value: "GCJ02" },
        { label: "WGS84", value: "WGS84" }
      ],
      onChange: setCoordConvert
    }
  )), /* @__PURE__ */ import_react.default.createElement(
    import_antd.Tabs,
    {
      activeKey: activeTab === "code" ? "geojson" : activeTab,
      className: styles.mapContentRight,
      defaultActiveKey: "geojson",
      items: tabItems ? tabItems : defaultTabItems,
      onChange: (e) => {
        setActiveTab(e);
      }
    }
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MapContent
});
