// @antv/l7-editor/lib/components/map-content/styles.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-content/styles.ts
var styles_exports = {};
__export(styles_exports, {
  default: () => styles_default
});
module.exports = __toCommonJS(styles_exports);
var import_css = require("@emotion/css");
var import_antd = require("antd");
var useStyle = () => {
  const { useToken } = import_antd.theme;
  const { token } = useToken();
  const { colorBgContainer, colorPrimary, colorTextTertiary } = token;
  return {
    mapContent: import_css.css`
      display: flex;
      width: 100%;
      height: 100%;
      .ant-tabs {
        width: 100%;
        .ant-tabs-nav {
          margin: 0 !important;
        }

        .ant-tabs-content-holder {
          height: calc(100% - 46px);
        }

        .ant-tabs-tabpane,
        .ant-tabs-content {
          height: 100%;
        }
        .ant-tabs-tab-active .ant-tabs-tab-btn .anticon {
          color: ${colorPrimary};
        }

        .ant-tabs-nav-list {
          margin-left: 12px;
        }
      }
    `,
    mapContentSelect: import_css.css`
      position: absolute;
      right: 0;
      z-index: 1;
      margin: 7px 12px;
      .ant-select-selector {
        border-width: 0px !important;
      }
    `,
    mapContentSelectLabel: import_css.css`
      color: ${colorTextTertiary};
      font-size: 13px;
    `,
    mapContentRight: import_css.css`
      height: 100%;
      width: calc(100% - 50px);
      background-color: ${colorBgContainer};
    `
  };
};
var styles_default = useStyle;
