// @antv/l7-editor/lib/components/map-control-group/draw-control/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/map-control-group/draw-control/index.tsx
var draw_control_exports = {};
__export(draw_control_exports, {
  default: () => draw_control_default
});
module.exports = __toCommonJS(draw_control_exports);
var import_l7_draw = require("@antv/l7-draw");
var import_larkmap = require("@antv/larkmap");
var import_lodash_es = require("lodash-es");
var import_react = __toESM(require("react"));
var import_constants = require("../../../constants");
var import_hooks = require("../../../hooks");
var import_recoil = require("../../../recoil");
var import_utils = require("../../../utils");
var import_styles = __toESM(require("./styles"));
var DrawControl = () => {
  const helperText = (0, import_hooks.useDrawHelper)();
  const scene = (0, import_larkmap.useScene)();
  const styles = (0, import_styles.default)();
  const [drawControl, setDrawControl] = (0, import_react.useState)(null);
  const { setIsDraw, resetFeatures, features, revertCoord } = (0, import_recoil.useFeature)();
  const { layerColor, showDrawDistance, showDrawArea } = (0, import_recoil.useGlobal)();
  const editFeature = (0, import_react.useMemo)(
    () => (0, import_lodash_es.cloneDeep)(
      features.find(
        // @ts-ignore
        (feature) => feature.properties[import_constants.FeatureKey.IsEdit]
      )
    ),
    [features]
  );
  (0, import_react.useEffect)(() => {
    var _a;
    const option = {
      distanceOptions: showDrawDistance ? {} : void 0,
      areaOptions: showDrawArea ? {} : void 0
    };
    let newDrawControl;
    if (scene) {
      newDrawControl = new import_l7_draw.DrawControl(scene, {
        position: "topleft",
        drawConfig: {
          point: true,
          line: option,
          polygon: option,
          rect: option,
          circle: option
        },
        commonDrawOptions: {
          maxCount: 1,
          style: (0, import_utils.getDrawStyle)(layerColor),
          //@ts-ignore
          helper: helperText
        }
      });
      setDrawControl(newDrawControl);
      scene.addControl(newDrawControl);
      newDrawControl.on(import_l7_draw.ControlEvent.DrawChange, (newType) => {
        setIsDraw(!!newType);
      });
      const drawDom = document.querySelector(".l7-draw-control");
      drawDom.style.marginTop = 0;
      (_a = document.querySelector("#l7-draw-content")) == null ? void 0 : _a.appendChild(drawDom);
    }
    return () => {
      if (newDrawControl) {
        scene.removeControl(newDrawControl);
      }
    };
  }, [scene, layerColor, showDrawDistance, showDrawArea]);
  const onDrawAdd = (0, import_react.useCallback)(
    (drawType, newFeature) => {
      newFeature.properties = {
        [import_constants.FeatureKey.DrawType]: drawType
      };
      drawControl == null ? void 0 : drawControl.clearDrawData();
      drawControl == null ? void 0 : drawControl.setActiveType(null);
      const newFeatures = revertCoord([newFeature]);
      resetFeatures([...features, ...newFeatures]);
    },
    [drawControl, revertCoord, resetFeatures, features]
  );
  const onDrawEdit = (0, import_react.useCallback)(
    (drawType, feature) => {
      const index = feature.properties[import_constants.FeatureKey.Index];
      const newFeatures = [...features];
      newFeatures[index].properties[import_constants.FeatureKey.IsEdit] = false;
      newFeatures[index].geometry = feature.geometry;
      drawControl == null ? void 0 : drawControl.clearDrawData();
      drawControl == null ? void 0 : drawControl.setActiveType(null);
      const revertFeatures = revertCoord(newFeatures);
      resetFeatures([...revertFeatures]);
    },
    [features, drawControl, revertCoord, resetFeatures]
  );
  (0, import_react.useEffect)(() => {
    const drawTypeList = [
      "point",
      "line",
      "polygon",
      "rect",
      "circle"
    ];
    const onDrawAddMap = (0, import_lodash_es.fromPairs)(
      drawTypeList.map((drawType) => [
        drawType,
        {
          onAdd: (newFeature) => onDrawAdd(drawType, newFeature),
          onEdit: (newFeature) => onDrawEdit(drawType, newFeature)
        }
      ])
    );
    drawTypeList.forEach((drawType) => {
      const draw = drawControl == null ? void 0 : drawControl.getTypeDraw(drawType);
      draw == null ? void 0 : draw.on(import_l7_draw.DrawEvent.Add, onDrawAddMap[drawType].onAdd);
      draw == null ? void 0 : draw.on(import_l7_draw.DrawEvent.Edit, onDrawAddMap[drawType].onEdit);
    });
    return () => {
      drawTypeList.forEach((drawType) => {
        const draw = drawControl == null ? void 0 : drawControl.getTypeDraw(drawType);
        draw == null ? void 0 : draw.off(import_l7_draw.DrawEvent.Add, onDrawAddMap[drawType].onAdd);
        draw == null ? void 0 : draw.off(import_l7_draw.DrawEvent.Edit, onDrawAddMap[drawType].onEdit);
      });
    };
  }, [drawControl, onDrawAdd]);
  (0, import_react.useEffect)(() => {
    var _a;
    if (editFeature && drawControl) {
      const drawType = (_a = editFeature.properties) == null ? void 0 : _a[import_constants.FeatureKey.DrawType];
      editFeature.properties.isActive = true;
      drawControl.setActiveType(drawType);
      const targetDraw = drawControl.getTypeDraw(drawType);
      targetDraw == null ? void 0 : targetDraw.setData([editFeature]);
    }
  }, [editFeature, drawControl]);
  return /* @__PURE__ */ import_react.default.createElement(import_larkmap.CustomControl, { position: "topright" }, /* @__PURE__ */ import_react.default.createElement("div", { style: { display: "flex" }, id: "l7-editor-draw" }, /* @__PURE__ */ import_react.default.createElement("div", { className: styles.l7DrawControl }, /* @__PURE__ */ import_react.default.createElement("div", { id: "l7-draw-content" }))));
};
var draw_control_default = DrawControl;
