// @antv/l7-editor/es/recoil/atomState.js
import localforage from 'localforage';
import { atom, DefaultValue } from 'recoil';
import { LocalStorageKey } from "../constants";
var localStorageEffect = function localStorageEffect(key) {
  return function (_ref) {
    var setSelf = _ref.setSelf,
      onSet = _ref.onSet;
    var getValue = localStorage.getItem(key);
    if (getValue) {
      setSelf(JSON.parse(getValue));
    }
    onSet(function (newValue) {
      if (newValue instanceof DefaultValue) {
        localStorage.removeItem(key);
      } else {
        localStorage.setItem(key, JSON.stringify(newValue));
      }
    });
  };
};
var indexDBEffect = function indexDBEffect(key) {
  return function (_ref2) {
    var setSelf = _ref2.setSelf,
      onSet = _ref2.onSet;
    localforage.getItem(key).then(function (getValue) {
      if (getValue) {
        setSelf(getValue);
      }
    });
    onSet(function (newValue) {
      if (newValue instanceof DefaultValue) {
        localforage.removeItem(key);
      } else {
        localforage.setItem(key, newValue);
      }
    });
  };
};
var filterState = atom({
  key: 'filter',
  default: [],
  dangerouslyAllowMutability: true
});
var isDrawState = atom({
  key: 'isDraw',
  default: false
});
var savedTextState = atom({
  key: 'savedText',
  default: ''
});
var featureState = atom({
  key: 'features',
  default: [],
  dangerouslyAllowMutability: true
});
var sceneState = atom({
  key: 'scene',
  default: null,
  dangerouslyAllowMutability: true
});
var editorTextState = atom({
  key: 'editorText',
  default: JSON.stringify({
    type: 'FeatureCollection',
    features: []
  }, null, 2),
  effects: [indexDBEffect(LocalStorageKey.EditorText)]
});
var lnglatTypeState = atom({
  key: 'lnglatType'
});
var lnglatTextState = atom({
  key: 'lnglatText',
  default: ''
});
var rightWidthState = atom({
  key: 'rightPanelWidth',
  effects: [localStorageEffect(LocalStorageKey.RightPanelWidth)]
});
var mapOptionState = atom({
  key: 'mapOptions',
  effects: [localStorageEffect(LocalStorageKey.MapOptions)]
});
var layerColorState = atom({
  key: 'layerColor',
  effects: [localStorageEffect(LocalStorageKey.LayerColor)]
});
var hideEditorState = atom({
  key: 'hideEditor',
  default: false,
  effects: [localStorageEffect(LocalStorageKey.HideEditor)]
});
var autoFitBoundsState = atom({
  key: 'autoFitBounds',
  effects: [localStorageEffect(LocalStorageKey.AutoFitBounds)]
});
var showDrawDistanceState = atom({
  key: 'showDrawDistance',
  default: false,
  effects: [localStorageEffect(LocalStorageKey.showDrawDistance)]
});
var showDrawAreaState = atom({
  key: 'showDrawArea',
  default: false,
  effects: [localStorageEffect(LocalStorageKey.showDrawArea)]
});
var popupTriggerState = atom({
  key: 'popupTrigger',
  effects: [localStorageEffect(LocalStorageKey.PopupTrigger)]
});
var baseMapState = atom({
  key: 'baseMap',
  effects: [localStorageEffect(LocalStorageKey.BaseMap)]
});
var activeTabState = atom({
  key: 'activeTab',
  default: 'geojson',
  effects: [localStorageEffect(LocalStorageKey.ActiveRightTabKey)]
});
var officialLayersState = atom({
  key: 'layerType',
  default: [],
  effects: [localStorageEffect(LocalStorageKey.officialLayers)]
});
var convertState = atom({
  key: 'coordConvert',
  default: 'GCJ02',
  effects: [localStorageEffect(LocalStorageKey.Convert)]
});
var themeState = atom({
  key: 'theme',
  default: 'light',
  effects: [localStorageEffect(LocalStorageKey.theme)]
});
var cityHistoryState = atom({
  key: 'cityHistory',
  default: [],
  effects: [localStorageEffect(LocalStorageKey.cityHistory)]
});
var showTextLayerState = atom({
  key: 'showTextLayer',
  default: true,
  effects: [localStorageEffect(LocalStorageKey.showTextLayer)]
});
var textLayerFieldsState = atom({
  key: 'textLayerFields',
  default: undefined,
  effects: [localStorageEffect(LocalStorageKey.textLayerFields)]
});
var customTilesState = atom({
  key: 'customTiles',
  default: [],
  effects: [localStorageEffect(LocalStorageKey.customTiles)]
});
var localeState = atom({
  key: 'locale',
  default: 'zh-CN',
  effects: [localStorageEffect(LocalStorageKey.locale)]
});
var wasmPathState = atom({
  key: 'wasmPaths',
  effects: [localStorageEffect(LocalStorageKey.wasmPath)]
});
export { activeTabState, autoFitBoundsState, baseMapState, cityHistoryState, convertState, customTilesState, editorTextState, featureState, filterState, hideEditorState, isDrawState, layerColorState, lnglatTextState, lnglatTypeState, localeState, mapOptionState, officialLayersState, popupTriggerState, rightWidthState, savedTextState, sceneState, showDrawAreaState, showDrawDistanceState, showTextLayerState, textLayerFieldsState, themeState, wasmPathState };
