// @antv/l7-editor/es/recoil/feature.js
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { bbox, featureCollection, getType } from '@turf/turf';
import { cloneDeep, flatMap, max, min } from 'lodash-es';
import { useCallback, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useRecoilState } from 'recoil';
import { FeatureKey } from "../constants";
import { gcj02towgs84, transformFeatures, wgs84togcj02 } from "../utils";
import { prettierText } from "../utils/prettier-text";
import { editorTextState, featureState, isDrawState, savedTextState, sceneState } from "./atomState";
import useGlobal from "./global";
export default function useFeature() {
  var _useGlobal = useGlobal(),
    baseMap = _useGlobal.baseMap,
    coordConvert = _useGlobal.coordConvert;
  var _useRecoilState = useRecoilState(editorTextState),
    _useRecoilState2 = _slicedToArray(_useRecoilState, 2),
    editorText = _useRecoilState2[0],
    setEditorText = _useRecoilState2[1];
  var _useRecoilState3 = useRecoilState(savedTextState),
    _useRecoilState4 = _slicedToArray(_useRecoilState3, 2),
    savedText = _useRecoilState4[0],
    setSavedText = _useRecoilState4[1];
  var _useRecoilState5 = useRecoilState(featureState),
    _useRecoilState6 = _slicedToArray(_useRecoilState5, 2),
    features = _useRecoilState6[0],
    _setFeatures = _useRecoilState6[1];
  var _useRecoilState7 = useRecoilState(isDrawState),
    _useRecoilState8 = _slicedToArray(_useRecoilState7, 2),
    isDraw = _useRecoilState8[0],
    setIsDraw = _useRecoilState8[1];
  var _useRecoilState9 = useRecoilState(sceneState),
    _useRecoilState10 = _slicedToArray(_useRecoilState9, 2),
    scene = _useRecoilState10[0],
    setScene = _useRecoilState10[1];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var savable = useMemo(function () {
    return editorText !== savedText;
  }, [editorText, savedText]);
  var fc = useMemo(function () {
    return featureCollection(features);
  }, [features]);
  var setFeatures = function setFeatures(f) {
    _setFeatures(
    // @ts-ignore
    cloneDeep(f).map(function (feature, featureIndex) {
      var _feature$properties;
      feature.properties = _objectSpread(_objectSpread({}, feature.properties), {}, _defineProperty({}, FeatureKey.Index, featureIndex));
      // @ts-ignore
      if (!((_feature$properties = feature.properties) !== null && _feature$properties !== void 0 && _feature$properties[FeatureKey.DrawType])) {
        // @ts-ignore
        feature.properties[FeatureKey.DrawType] = function () {
          var type = getType(feature);
          if (/Point/.test(type)) {
            return 'point';
          }
          if (/LineString/.test(type)) {
            return 'line';
          }
          return 'polygon';
        }();
      }
      return feature;
    }));
  };
  var saveEditorText = function saveEditorText(value) {
    var emptyFeatures = JSON.stringify({
      type: 'FeatureCollection',
      features: []
    }, null, 2);
    var newFeatures = [];
    if (editorText || value) {
      newFeatures = transformFeatures(value !== null && value !== void 0 ? value : editorText, t);
      if (value) {
        setEditorText(value);
      }
      setSavedText(value !== null && value !== void 0 ? value : editorText);
      setFeatures(newFeatures);
    } else {
      setEditorText(emptyFeatures);
      setSavedText(emptyFeatures);
    }
    return newFeatures;
  };
  var resetFeatures = function resetFeatures(newFeatures) {
    var newText = prettierText({
      content: featureCollection(newFeatures)
    });
    setEditorText(newText);
    setSavedText(newText);
    setFeatures(newFeatures);
  };
  var dataSource = useMemo(function () {
    var data = features.map(function (item, index) {
      var properties = item.properties;
      return _objectSpread({
        __index: index + 1
      }, properties);
    });
    var featureKeyList = [];
    Array.from(new Set(flatMap(features.map(function (_ref) {
      var properties = _ref.properties;
      return Object.keys(properties);
    })))).forEach(function (field) {
      var type = _typeof(data[0][field]);
      if (type === 'string' || type === 'boolean') {
        var value = data.map(function (item) {
          return String(item[field]);
        });
        featureKeyList.push({
          type: 'string',
          field: field,
          value: value
        });
      } else if (type === 'number') {
        var _value = data.map(function (item) {
          return item[field];
        });
        featureKeyList.push({
          type: type,
          field: field,
          min: min(_value),
          max: max(_value)
        });
      }
    });
    return featureKeyList;
  }, [features]);
  var transformCoord = useCallback(function (newFeatures) {
    var data = _toConsumableArray(newFeatures);
    if (coordConvert === 'WGS84' && baseMap === 'Gaode') {
      data = data.map(function (item) {
        return wgs84togcj02(cloneDeep(item));
      });
    } else if (coordConvert === 'GCJ02' && baseMap === 'Mapbox') {
      data = data.map(function (item) {
        return gcj02towgs84(cloneDeep(item));
      });
    }
    return data;
  }, [baseMap, coordConvert]);
  var revertCoord = useCallback(function (newFeatures) {
    var data = _toConsumableArray(newFeatures);
    if (coordConvert === 'WGS84' && baseMap === 'Gaode') {
      data = data.map(function (item) {
        return gcj02towgs84(cloneDeep(item));
      });
    } else if (coordConvert === 'GCJ02' && baseMap === 'Mapbox') {
      data = data.map(function (item) {
        return wgs84togcj02(cloneDeep(item));
      });
    }
    return data;
  }, [baseMap, coordConvert]);
  var bboxAutoFit = useCallback(function (currentFeatures) {
    var realFeatures = currentFeatures !== null && currentFeatures !== void 0 ? currentFeatures : features;
    if (scene && realFeatures.length) {
      var _bbox = bbox(featureCollection(transformCoord(realFeatures))),
        _bbox2 = _slicedToArray(_bbox, 4),
        lng1 = _bbox2[0],
        lat1 = _bbox2[1],
        lng2 = _bbox2[2],
        lat2 = _bbox2[3];
      scene.fitBounds([[lng1, lat1], [lng2, lat2]]);
    }
  }, [features, scene, transformCoord]);
  return {
    editorText: editorText,
    setEditorText: setEditorText,
    savedText: savedText,
    setSavedText: setSavedText,
    features: features,
    setFeatures: setFeatures,
    savable: savable,
    saveEditorText: saveEditorText,
    resetFeatures: resetFeatures,
    dataSource: dataSource,
    bboxAutoFit: bboxAutoFit,
    setIsDraw: setIsDraw,
    isDraw: isDraw,
    scene: scene,
    setScene: setScene,
    fc: fc,
    transformCoord: transformCoord,
    revertCoord: revertCoord
  };
}
