// @antv/l7-editor/es/recoil/global.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { useRecoilState } from 'recoil';
import { activeTabState, showDrawAreaState, autoFitBoundsState, baseMapState, cityHistoryState, convertState, customTilesState, showDrawDistanceState, hideEditorState, layerColorState, localeState, mapOptionState, officialLayersState, popupTriggerState, rightWidthState, showTextLayerState, textLayerFieldsState, themeState, wasmPathState } from "./atomState";
export default function useGlobal() {
  var _useRecoilState = useRecoilState(rightWidthState),
    _useRecoilState2 = _slicedToArray(_useRecoilState, 2),
    rightPanelWidth = _useRecoilState2[0],
    setRightWidth = _useRecoilState2[1];
  var _useRecoilState3 = useRecoilState(mapOptionState),
    _useRecoilState4 = _slicedToArray(_useRecoilState3, 2),
    mapOptions = _useRecoilState4[0],
    setMapOptions = _useRecoilState4[1];
  var _useRecoilState5 = useRecoilState(layerColorState),
    _useRecoilState6 = _slicedToArray(_useRecoilState5, 2),
    layerColor = _useRecoilState6[0],
    setLayerColor = _useRecoilState6[1];
  var _useRecoilState7 = useRecoilState(hideEditorState),
    _useRecoilState8 = _slicedToArray(_useRecoilState7, 2),
    hideEditor = _useRecoilState8[0],
    setHideEditor = _useRecoilState8[1];
  var _useRecoilState9 = useRecoilState(autoFitBoundsState),
    _useRecoilState10 = _slicedToArray(_useRecoilState9, 2),
    autoFitBounds = _useRecoilState10[0],
    setAutoFitBounds = _useRecoilState10[1];
  var _useRecoilState11 = useRecoilState(popupTriggerState),
    _useRecoilState12 = _slicedToArray(_useRecoilState11, 2),
    popupTrigger = _useRecoilState12[0],
    setPopupTrigger = _useRecoilState12[1];
  var _useRecoilState13 = useRecoilState(baseMapState),
    _useRecoilState14 = _slicedToArray(_useRecoilState13, 2),
    baseMap = _useRecoilState14[0],
    setBaseMap = _useRecoilState14[1];
  var _useRecoilState15 = useRecoilState(activeTabState),
    _useRecoilState16 = _slicedToArray(_useRecoilState15, 2),
    activeTab = _useRecoilState16[0],
    setActiveTab = _useRecoilState16[1];
  var _useRecoilState17 = useRecoilState(officialLayersState),
    _useRecoilState18 = _slicedToArray(_useRecoilState17, 2),
    layerType = _useRecoilState18[0],
    setLayerType = _useRecoilState18[1];
  var _useRecoilState19 = useRecoilState(convertState),
    _useRecoilState20 = _slicedToArray(_useRecoilState19, 2),
    coordConvert = _useRecoilState20[0],
    setCoordConvert = _useRecoilState20[1];
  var _useRecoilState21 = useRecoilState(themeState),
    _useRecoilState22 = _slicedToArray(_useRecoilState21, 2),
    theme = _useRecoilState22[0],
    setTheme = _useRecoilState22[1];
  var _useRecoilState23 = useRecoilState(cityHistoryState),
    _useRecoilState24 = _slicedToArray(_useRecoilState23, 2),
    cityHistory = _useRecoilState24[0],
    setCityHistory = _useRecoilState24[1];
  var _useRecoilState25 = useRecoilState(showTextLayerState),
    _useRecoilState26 = _slicedToArray(_useRecoilState25, 2),
    showTextLayer = _useRecoilState26[0],
    setShowTextLayer = _useRecoilState26[1];
  var _useRecoilState27 = useRecoilState(textLayerFieldsState),
    _useRecoilState28 = _slicedToArray(_useRecoilState27, 2),
    textLayerFields = _useRecoilState28[0],
    setTextLayerFields = _useRecoilState28[1];
  var _useRecoilState29 = useRecoilState(customTilesState),
    _useRecoilState30 = _slicedToArray(_useRecoilState29, 2),
    customTiles = _useRecoilState30[0],
    setCustomTiles = _useRecoilState30[1];
  var _useRecoilState31 = useRecoilState(localeState),
    _useRecoilState32 = _slicedToArray(_useRecoilState31, 2),
    locale = _useRecoilState32[0],
    setLocale = _useRecoilState32[1];
  var _useRecoilState33 = useRecoilState(wasmPathState),
    _useRecoilState34 = _slicedToArray(_useRecoilState33, 2),
    wasmPath = _useRecoilState34[0],
    setWasmPath = _useRecoilState34[1];
  var _useRecoilState35 = useRecoilState(showDrawDistanceState),
    _useRecoilState36 = _slicedToArray(_useRecoilState35, 2),
    showDrawDistance = _useRecoilState36[0],
    setShowDrawDistance = _useRecoilState36[1];
  var _useRecoilState37 = useRecoilState(showDrawAreaState),
    _useRecoilState38 = _slicedToArray(_useRecoilState37, 2),
    showDrawArea = _useRecoilState38[0],
    setShowDrawArea = _useRecoilState38[1];
  return {
    rightPanelWidth: rightPanelWidth,
    setRightWidth: setRightWidth,
    mapOptions: mapOptions,
    setMapOptions: setMapOptions,
    layerColor: layerColor,
    setLayerColor: setLayerColor,
    hideEditor: hideEditor,
    setHideEditor: setHideEditor,
    popupTrigger: popupTrigger,
    setPopupTrigger: setPopupTrigger,
    baseMap: baseMap,
    setBaseMap: setBaseMap,
    autoFitBounds: autoFitBounds,
    setAutoFitBounds: setAutoFitBounds,
    activeTab: activeTab,
    setActiveTab: setActiveTab,
    layerType: layerType,
    setLayerType: setLayerType,
    coordConvert: coordConvert,
    setCoordConvert: setCoordConvert,
    theme: theme,
    setTheme: setTheme,
    cityHistory: cityHistory,
    setCityHistory: setCityHistory,
    showTextLayer: showTextLayer,
    setShowTextLayer: setShowTextLayer,
    textLayerFields: textLayerFields,
    setTextLayerFields: setTextLayerFields,
    customTiles: customTiles,
    setCustomTiles: setCustomTiles,
    locale: locale,
    setLocale: setLocale,
    wasmPath: wasmPath,
    setWasmPath: setWasmPath,
    showDrawDistance: showDrawDistance,
    setShowDrawDistance: setShowDrawDistance,
    showDrawArea: showDrawArea,
    setShowDrawArea: setShowDrawArea
  };
}
