// @antv/l7-editor/es/recoil/lnglat.js
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { lineString, point, polygon } from '@turf/turf';
import { first, isEqual, last } from 'lodash-es';
import { useTranslation } from 'react-i18next';
import { useRecoilState } from 'recoil';
import { LngLatVT } from "../constants";
import { lnglatTextState, lnglatTypeState } from "./atomState";
export default function useLnglat() {
  var _useRecoilState = useRecoilState(lnglatTypeState),
    _useRecoilState2 = _slicedToArray(_useRecoilState, 2),
    lngLatImportType = _useRecoilState2[0],
    setLngLatImportType = _useRecoilState2[1];
  var _useTranslation = useTranslation(),
    t = _useTranslation.t;
  var _useRecoilState3 = useRecoilState(lnglatTextState),
    _useRecoilState4 = _slicedToArray(_useRecoilState3, 2),
    lngLatText = _useRecoilState4[0],
    setLngLatText = _useRecoilState4[1];
  var importLngLatText = function importLngLatText(text) {
    var featurePositionList = text.split('\n').filter(function (item) {
      return item;
    }).map(function (item) {
      return item.split(';').filter(function (str) {
        return str;
      }).map(function (str) {
        return str.split(',').filter(function (num) {
          return num;
        }).map(function (num) {
          return +num;
        });
      });
    });
    if (LngLatVT.check(featurePositionList)) {
      var newFeatures = [];
      if (lngLatImportType === 'Point') {
        newFeatures.push.apply(newFeatures, _toConsumableArray(featurePositionList.flat().map(function (position) {
          return point(position);
        })));
      } else if (lngLatImportType === 'LingString') {
        newFeatures.push.apply(newFeatures, _toConsumableArray(featurePositionList.map(function (positions) {
          return lineString(positions);
        })));
      } else {
        newFeatures.push.apply(newFeatures, _toConsumableArray(featurePositionList.map(function (positions) {
          if (!isEqual(first(positions), last(positions))) {
            positions.push(positions[0]);
          }
          return polygon([positions]);
        })));
      }
      return newFeatures;
    } else {
      throw new Error(t('import_btn.lnglat_import_btn.lNGLA'));
    }
  };
  return {
    lngLatImportType: lngLatImportType,
    setLngLatImportType: setLngLatImportType,
    lngLatText: lngLatText,
    setLngLatText: setLngLatText,
    importLngLatText: importLngLatText
  };
}
