// @antv/l7-editor/es/components/resize-panel/index.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { CaretRightOutlined } from '@ant-design/icons';
import { useSize } from 'ahooks';
import classNames from 'classnames';
import { Resizable } from 're-resizable';
import React, { useEffect, useMemo, useState } from 'react';
import { RightPanelWidthRange } from "../../constants";
import { useFeature, useGlobal } from "../../recoil";
import useStyle from "./styles";
export var ResizePanel = function ResizePanel(_ref) {
  var _useSize;
  var left = _ref.left,
    right = _ref.right,
    onFeatureChange = _ref.onFeatureChange;
  var _useGlobal = useGlobal(),
    hideEditor = _useGlobal.hideEditor,
    setHideEditor = _useGlobal.setHideEditor,
    rightPanelWidth = _useGlobal.rightPanelWidth,
    setRightWidth = _useGlobal.setRightWidth;
  var _useState = useState(null),
    _useState2 = _slicedToArray(_useState, 2),
    resizePanel = _useState2[0],
    setResizePanel = _useState2[1];
  var _RightPanelWidthRange = _slicedToArray(RightPanelWidthRange, 2),
    minRightWidth = _RightPanelWidthRange[0],
    maxRightWidth = _RightPanelWidthRange[1];
  var _ref2 = (_useSize = useSize(resizePanel)) !== null && _useSize !== void 0 ? _useSize : {},
    _ref2$width = _ref2.width,
    containerWidth = _ref2$width === void 0 ? 0 : _ref2$width;
  var _useFeature = useFeature(),
    features = _useFeature.features;
  var styles = useStyle();
  var onResize = function onResize(event) {
    var _resizePanel$getBound;
    var _ref3 = (_resizePanel$getBound = resizePanel === null || resizePanel === void 0 ? void 0 : resizePanel.getBoundingClientRect()) !== null && _resizePanel$getBound !== void 0 ? _resizePanel$getBound : {},
      _ref3$left = _ref3.left,
      leftWidth = _ref3$left === void 0 ? 0 : _ref3$left;
    var newRightPanelWidth = 100 * (1 - (event.clientX - leftWidth) / containerWidth);
    if (newRightPanelWidth < minRightWidth) {
      newRightPanelWidth = minRightWidth;
    }
    if (newRightPanelWidth > maxRightWidth) {
      newRightPanelWidth = maxRightWidth;
    }
    setRightWidth(newRightPanelWidth);
  };
  var calcRightWidth = useMemo(function () {
    return hideEditor ? 0 : rightPanelWidth;
  }, [hideEditor, rightPanelWidth]);
  useEffect(function () {
    onFeatureChange(features);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [features]);
  return /*#__PURE__*/React.createElement("div", {
    className: classNames([styles.resizePanel, 'l7-editor-content']),
    ref: function ref(_ref4) {
      setResizePanel(_ref4);
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: classNames([styles.resizePanelLeft, 'l7-editor-content__left']),
    style: {
      width: "".concat(100 - calcRightWidth, "%")
    }
  }, left), /*#__PURE__*/React.createElement(Resizable, {
    enable: {
      left: true
    },
    style: {
      marginLeft: "".concat(100 - calcRightWidth, "%")
    },
    size: {
      width: "".concat(calcRightWidth, "%"),
      height: '100%'
    },
    minWidth: "".concat(minRightWidth / 100 * containerWidth, "px"),
    maxWidth: "".concat(maxRightWidth / 100 * containerWidth, "px"),
    className: classNames([styles.resizePanelRight, 'l7-editor-content__right']),
    handleClasses: {
      left: styles.resizePanelDragLine
    },
    handleStyles: {
      left: {
        width: 6,
        left: -3
      }
    },
    defaultSize: {
      width: "".concat(calcRightWidth, "%"),
      height: '100%'
    },
    onResize: onResize
  }, right, /*#__PURE__*/React.createElement("div", {
    className: styles.resizePanelToggleBtn,
    onClick: function onClick() {
      setHideEditor(!hideEditor);
    }
  }, /*#__PURE__*/React.createElement(CaretRightOutlined, {
    style: {
      transform: hideEditor ? 'rotate(-180deg)' : undefined
    }
  }))));
};
