// @antv/l7-editor/es/components/resize-panel/styles.js
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css } from '@emotion/css';
import { theme } from 'antd';
var useStyle = function useStyle() {
  var useToken = theme.useToken;
  var _useToken = useToken(),
    token = _useToken.token;
  var colorBgContainer = token.colorBgContainer,
    boxShadow = token.boxShadow,
    colorBorderSecondary = token.colorBorderSecondary,
    colorInfoHover = token.colorInfoHover,
    colorIcon = token.colorIcon;
  return {
    resizePanel: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n      width: 100%;\n      height: calc(100% - 50px);\n      display: flex;\n      position: relative;\n    "]))),
    resizePanelLeft: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n      position: absolute;\n      left: 0;\n      top: 0;\n      bottom: 0;\n      overflow: hidden;\n    "]))),
    resizePanelRight: css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n      position: relative;\n      border-left: 1px solid ", ";\n      box-shadow: ", ";\n    "])), colorBorderSecondary, boxShadow),
    resizePanelDragLine: css(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n      transition: all 0.5s;\n      cursor: move !important;\n      z-index: 100;\n\n      &:hover,\n      &:focus {\n        background-color: ", ";\n      }\n    "])), colorInfoHover),
    resizePanelToggleBtn: css(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n      height: 60px;\n      width: 20px;\n      display: flex;\n      align-items: center;\n      justify-content: center;\n      background: ", ";\n      position: absolute;\n      top: calc(50% - 30px);\n      left: -20px;\n      border-top-left-radius: 4px;\n      border-bottom-left-radius: 4px;\n      cursor: pointer;\n      z-index: 2;\n      .anticon {\n        transition: all 0.2s;\n      }\n      color: ", ";\n    "])), colorBgContainer, colorIcon)
  };
};
export default useStyle;
